---
title: "MID study humann3"
author: "Irina Velsko"
date: "`r format(Sys.time(), '%b %d, %Y')`"
output:
    github_document:
        fig_width: 8
        fig_height: 6
        toc: true
---


```{r load_libraries, echo = F, message = F}
library(knitr)
library(decontam)
library(data.table)
library(vegan)
library(ape)
library(mixOmics)
library(janitor)
library(rstatix)
library(variancePartition)
library(pander)
library(tidyverse)
library(gplots)
library(ggrepel)
library(viridis)
library(cowplot)
library(RColorBrewer)
opts_chunk$set(echo=F, warning=F, message=F, dpi=150)
```

Set the working directory of the notebook to the parent directory of the 
notebook itself.  

```{r setup}
knitr::opts_knit$set(root.dir = normalizePath(".."))
```


# humann3 tables 
```{r load_data}
## load the species and genus tables generated with humann3
humann3_path_full <- fread("./05-results.backup/pathabundance_joined_cpm.tsv")
humann3_path_full <- as_tibble(humann3_path_full)
humann3_GFs_full <- fread("./05-results.backup/genefamilies_joined_cpm_ur90rxn.tsv")
humann3_GFs_full <- as_tibble(humann3_GFs_full)

# clean the file names
humann3_GFs_full <- rename(humann3_GFs_full, GeneFamily = `# Gene Family`)
colnames(humann3_GFs_full) <- gsub(".unmapped_Abundance-RPKs","", colnames(humann3_GFs_full))
colnames(humann3_GFs_full) <- gsub(".SG1","", colnames(humann3_GFs_full))

humann3_path_full <- rename(humann3_path_full, Pathway = `# Pathway`)
colnames(humann3_path_full) <- gsub(".unmapped_Abundance","", colnames(humann3_path_full))
colnames(humann3_path_full) <- gsub(".SG1","", colnames(humann3_path_full))

# remove unmapped and ungrouped reads
humann3_path <- humann3_path_full %>% filter(!str_detect(Pathway, "UNMAPPED|UNINTEGRATED"))
humann3_GFs <- humann3_GFs_full %>% filter(!str_detect(GeneFamily, "UNMAPPED|UNGROUPED"))

```

```{r}
# load the metadata file
load("./05-results.backup/Metadata_tables.RData")

```


```{r}
poor_samples_mpa3 <- fread("./05-results.backup/cuperdec_mpa3_poor_samples.tsv") %>%
  pull(Sample)

```


```{r plot_pca_fxn}

# add functions for:
# making the input table for PCA?

# plotting PCA with colored dots
plot_pca <- function(df, pc1, pc2, color_group, shape_group, size_group) {
    metadata_group_colors <- get(paste(color_group, "_colors", sep = ""))
    metadata_group_shapes <- get(paste(shape_group, "_shapes", sep = ""))
    metadata_group_size <- get(paste(size_group, "_size", sep = ""))

    exp_var <- paste0(round(df$prop_expl_var$X * 100, 2), "%")
    df_X <- df$variates$X %>%
              as.data.frame() %>%
              rownames_to_column("Library_ID") %>%
              inner_join(full_metadata, by = "Library_ID") %>%
              inner_join(., strep_group_info, by = "Library_ID")
    
    color_group = df_X[[color_group]]
    shape_group = df_X[[shape_group]]
    size_group = df_X[[size_group]]
                      
                       if (pc1 == 'PC1') {
                           pc1 <- df_X$PC1
                           exp_var_pc1 <- exp_var[1]
                           xaxis <- c("PC1")
                       }  else if (pc1 == 'PC2') {
                           pc1 <- df_X$PC2
                           exp_var_pc1 <- exp_var[2]
                           xaxis <- c("PC2")
                       } else if (pc1 == 'PC3') {
                           pc1 <- df_X$PC3
                           exp_var_pc1 <- exp_var[3]
                           xaxis <- c("PC3")
                       }

                       if (pc2 == 'PC1') {
                           pc2 <- df_X$PC1
                           exp_var_pc2 <- exp_var[1]
                           yaxis <- c("PC1")
                       }  else if (pc2 == 'PC2') {
                           pc2 <- df_X$PC2
                           exp_var_pc2 <- exp_var[2]
                           yaxis <- c("PC2")
                       } else if (pc2 == 'PC3') {
                           pc2 <- df_X$PC3
                           exp_var_pc2 <- exp_var[3]
                           yaxis <- c("PC3")
                       }

    pca_plot <- ggplot(df_X, aes(pc1, pc2, colour = color_group, shape = shape_group, size = size_group)) +
                        geom_point() +
                        scale_colour_manual(values = metadata_group_colors) +
                        scale_shape_manual(values = metadata_group_shapes) +
                        scale_size_manual(values = metadata_group_size) +
                        # stat_ellipse() +
                        xlab(paste(xaxis, " - ", exp_var_pc1)) +
                        ylab(paste(yaxis, " - ", exp_var_pc2)) +
                        theme_minimal(base_size = 14) +
                        theme(legend.position = "top") +
                        theme(legend.title = element_blank())

    return(pca_plot)
}

```

```{r pca_names_fxn}
# this function plots the PCA with the sample names instead of points
plot_pca_names <- function(df, pc1, pc2, metadata_group) {
    metadata_group_colors <- get(paste(full_metadata_group, "_colors", sep = ""))

    exp_var <- paste0(round(df$prop_expl_var$X * 100, 2), "%")
    df_X <- df$variates$X %>%
              as.data.frame() %>%
              rownames_to_column("Library_ID") %>%
              inner_join(full_metadata, by = "Library_ID") %>%
              inner_join(., strep_group_info, by = "Library_ID")
    
    metadata_group = df_X[[metadata_group]]
    
                      # select the correct PC column and explained variance for PC1
                       if (pc1 == 'PC1') {
                           pc1 <- df_X$PC1
                           exp_var_pc1 <- exp_var[1]
                           xaxis <- c("PC1")
                       }  else if (pc1 == 'PC2') {
                           pc1 <- df_X$PC2
                           exp_var_pc1 <- exp_var[2]
                           xaxis <- c("PC2")
                       } else if (pc1 == 'PC3') {
                           pc1 <- df_X$PC3
                           exp_var_pc1 <- exp_var[3]
                           xaxis <- c("PC3")
                       }

                       if (pc2 == 'PC1') {
                           pc2 <- df_X$PC1
                           exp_var_pc2 <- exp_var[1]
                           yaxis <- c("PC1")
                       }  else if (pc2 == 'PC2') {
                           pc2 <- df_X$PC2
                           exp_var_pc2 <- exp_var[2]
                           yaxis <- c("PC2")
                       } else if (pc2 == 'PC3') {
                           pc2 <- df_X$PC3
                           exp_var_pc2 <- exp_var[3]
                           yaxis <- c("PC3")
                       }

    pca_plot_names <- ggplot(df_X, aes(pc1, pc2, colour = metadata_group)) +
                        geom_text(aes(label = df_X$Library_ID), size = 2) +
                        scale_colour_manual(values = metadata_group_colors) +
                        # stat_ellipse() +
                        xlab(paste(xaxis, " - ", exp_var_pc1)) +
                        ylab(paste(yaxis, " - ", exp_var_pc2)) +
                        theme_minimal(base_size = 14) +
                        theme(legend.position = "top") +
                        theme(legend.title = element_blank())

    return(pca_plot_names)
}

```

```{r plot_bi}
# this function plots the PCA with biplots to look at the categories with strongest loadings
plot_pca_bi <- function(df, pc1, pc2, metadata_group, columntitle, pathways = c("TRUE","FALSE")) {
    metadata_group_colors <- get(paste(metadata_group, "_colors", sep = ""))
    metadata_group_shapes <- get(paste(metadata_group, "_shapes", sep = ""))
   
    arrow_pc <- enquo(columntitle)
    
    exp_var <- paste0(round(df$prop_expl_var$X * 100, 2), "%") # explained variance for x- and y-labels
    
    # select only the PCs from the PCA and add metadata
    df_X <- df$variates$X %>%
              as.data.frame() %>%
              rownames_to_column("Library_ID") %>%
              inner_join(full_metadata, by = "Library_ID") %>%
              inner_join(., strep_group_info, by = "Library_ID")
    
    metadata_group = df_X[[metadata_group]]
    
    corr_lam <- df$sdev[c("PC1", "PC2", "PC3")] * sqrt(nrow(df_X))

    df_X <- df_X %>%
      mutate(PC1 = PC1 / corr_lam[1],
             PC2 = PC2 / corr_lam[2], 
             PC3 = PC3 / corr_lam[3])

    # select the correct PC column and explained variance for PC1
    if (pc1 == 'PC1') {
        Pc1 <- df_X$PC1
        exp_var_pc1 <- exp_var[1]
        xaxis <- c("PC1")
    } else if (pc1 == 'PC2') {
        Pc1 <- df_X$PC2
        exp_var_pc1 <- exp_var[2]
        xaxis <- c("PC2")
    } else if (pc1 == 'PC3') {
        Pc1 <- df_X$PC3
        exp_var_pc1 <- exp_var[3]
        xaxis <- c("PC3")
   }
    
    # select the correct PC column and explained variance for PC2
    if (pc2 == 'PC1') {
        Pc2 <- df_X$PC1
        exp_var_pc2 <- exp_var[1]
        yaxis <- c("PC1")
 } else if (pc2 == 'PC2') {
       Pc2 <- df_X$PC2
       exp_var_pc2 <- exp_var[2]
       yaxis <- c("PC2")
    } else if (pc2 == 'PC3') {
       Pc2 <- df_X$PC3
       exp_var_pc2 <- exp_var[3]
       yaxis <- c("PC3")
   }

    # Identify the 10 pathways that have highest positive and negative loadings in the selected PC
    pws_10 <- df$loadings$X %>%
      as.data.frame(.) %>%
      rownames_to_column(var = "Function") %>%
      top_n(10, !!arrow_pc)
   if (pathways == T) {
      pws_10 <- pws_10 %>%
        rename(pathway = Function) %>%
        mutate(Function = sapply(pathway, function(f) {
                                  unlist(str_split(f, ":"))[1]
                           })) 
    }

    neg_10 <- df$loadings$X %>%
      as.data.frame(.) %>%
      rownames_to_column(var = "Function") %>%
      top_n(-10, !!arrow_pc)
    if (pathways == T) {
      neg_10 <- neg_10 %>%
      rename(pathway = Function) %>%
       mutate(Function = sapply(pathway, function(f) {
                                  unlist(str_split(f, ":"))[1]
                           })) 
    }


    pca_plot_bi <- ggplot(df_X, aes(x = Pc1, y = Pc2, colour = metadata_group)) +
      geom_point(size = 2, aes(shape = metadata_group) )+
      geom_segment(data = pws_10,
                   aes(xend = get(paste(pc1)), yend = get(paste(pc2))),
                   x = 0, y = 0, colour = "black",
                   size = 0.5,
                   arrow = arrow(length = unit(0.03, "npc"))) +
      geom_label_repel(data = pws_10,
                   aes(x = get(paste(pc1)), y = get(paste(pc2)), label = Function),
                   size = 1.5, colour = "grey20", label.padding = 0.2, max.overlaps = 12) +
      geom_segment(data = neg_10,
                   aes(xend = get(paste(pc1)), yend = get(paste(pc2))),
                   x = 0, y = 0, colour = "grey50",
                   size = 0.5,
                   arrow = arrow(length = unit(0.03, "npc"))) +
      geom_label_repel(data = neg_10,
                   aes(x = get(paste(pc1)), y = get(paste(pc2)), label = Function),
                   size = 1.5, colour = "grey20", label.padding = 0.2, max.overlaps = 12) +
      labs(x = paste(xaxis, " - ", exp_var_pc1),
           y = paste(yaxis, " - ", exp_var_pc2)) +
      scale_colour_manual(values = metadata_group_colors) +
      scale_shape_manual(values = metadata_group_shapes) +
      theme_minimal() + theme(text = element_text(size = 14)) +
      theme(legend.position = "top")

    return(pca_plot_bi)
}

```

```{r pca_plot_cont_fxn}
# for continuous data
plot_pca_cont <- function(df, pc1, pc2, color_group, shape_group, ncomps, title_text) {

    pca.list <- mixOmics::pca(df, ncomp = ncomps, logratio = 'CLR')

    exp_var <- paste0(round(pca.list$prop_expl_var$X * 100, 2), "%")
    df_X <- pca.list$variates$X %>%
              as.data.frame() %>%
              rownames_to_column("Library_ID") %>%
              inner_join(full_metadata, by = "Library_ID") %>%
              inner_join(., strep_group_info, by = "Library_ID") %>%
              inner_join(., strep_group_info, by = "Library_ID")
    # %>%
              # full_join(., full_source_results %>%
              #             rename(Library_ID = SampleID),  by = "Library_ID")
    
    color_group = df_X[[color_group]]
    shape_group = df_X[[shape_group]]

    if (pc1 == 'PC1') {
        pc1 <- df_X$PC1
        exp_var_pc1 <- exp_var[1]
        xaxis <- c("PC1")
    }  else if (pc1 == 'PC2') {
        pc1 <- df_X$PC2
        exp_var_pc1 <- exp_var[2]
        xaxis <- c("PC2")
    } else if (pc1 == 'PC3') {
        pc1 <- df_X$PC3
        exp_var_pc1 <- exp_var[3]
        xaxis <- c("PC3")
    }

    if (pc2 == 'PC1') {
        pc2 <- df_X$PC1
        exp_var_pc2 <- exp_var[1]
        yaxis <- c("PC1")
    }  else if (pc2 == 'PC2') {
        pc2 <- df_X$PC2
        exp_var_pc2 <- exp_var[2]
        yaxis <- c("PC2")
    } else if (pc2 == 'PC3') {
        pc2 <- df_X$PC3
        exp_var_pc2 <- exp_var[3]
        yaxis <- c("PC3")
    }

    pca_plot <- ggplot(df_X, aes(pc1, pc2, colour = color_group, shape = shape_group)) +
     geom_point(size = 2) +
     scale_colour_viridis_c(option = "C") +
     scale_shape_manual(values = c(16, 17, 15, 18, 14,9,13,24)) +
     # stat_ellipse() +
     xlab(paste(xaxis, " - ", exp_var_pc1)) +
     ylab(paste(yaxis, " - ", exp_var_pc2)) +
     theme_minimal(base_size = 16) +
     theme(text = element_text(size=16)) +
     theme(legend.title = element_blank(),
           legend.key.size = unit(2,"mm"),
           legend.text = element_text(size = 6)) + 
     theme(legend.position = "top") +
     ggtitle(title_text) + theme(plot.title = element_text(size = 10))

    return(pca_plot)
}


```

```{r pca_plot_mid_fxn}
# plot PCA with colored dots 
plot_pca_mid <- function(df, pc1, pc2, color_group, shape_group, ncomps, title_text) {

    pca.list <- mixOmics::pca(df, ncomp = ncomps, logratio = 'CLR')

    exp_var <- paste0(round(pca.list$prop_expl_var$X * 100, 2), "%")
    df_X <- pca.list$variates$X %>%
              as.data.frame() %>%
              rownames_to_column("Library_ID") %>%
              inner_join(full_metadata, by = "Library_ID") %>%
              inner_join(., strep_group_info, by = "Library_ID")
    
    color_group = df_X[[color_group]]
    shape_group = df_X[[shape_group]]

    if (pc1 == 'PC1') {
        pc1 <- df_X$PC1
        exp_var_pc1 <- exp_var[1]
        xaxis <- c("PC1")
    }  else if (pc1 == 'PC2') {
        pc1 <- df_X$PC2
        exp_var_pc1 <- exp_var[2]
        xaxis <- c("PC2")
    } else if (pc1 == 'PC3') {
        pc1 <- df_X$PC3
        exp_var_pc1 <- exp_var[3]
        xaxis <- c("PC3")
    }

    if (pc2 == 'PC1') {
        pc2 <- df_X$PC1
        exp_var_pc2 <- exp_var[1]
        yaxis <- c("PC1")
    }  else if (pc2 == 'PC2') {
        pc2 <- df_X$PC2
        exp_var_pc2 <- exp_var[2]
        yaxis <- c("PC2")
    } else if (pc2 == 'PC3') {
        pc2 <- df_X$PC3
        exp_var_pc2 <- exp_var[3]
        yaxis <- c("PC3")
    }

    pca_plot <- ggplot(df_X, aes(pc1, pc2, colour = color_group, shape = shape_group)) +
     geom_point(size = 2) +
     scale_colour_manual(values = c("#5A167E","#E95562","#FED395","#252525","#311068","#C83E73","#FEA873")) + # 
     scale_shape_manual(values = c(16, 17, 15, 18)) +
     # stat_ellipse() +
     xlab(paste(xaxis, " - ", exp_var_pc1)) +
     ylab(paste(yaxis, " - ", exp_var_pc2)) +
     theme_minimal(base_size = 16) +
     theme(text = element_text(size=16)) +
     theme(legend.title = element_blank(),
           legend.key.size = unit(2,"mm"),
           legend.text = element_text(size = 6)) + # 
     theme(legend.position = "top") +
     ggtitle(title_text) + theme(plot.title = element_text(size = 10))

    return(pca_plot)
}

```

# Pathway analysis

```{r}
# mpa3 species table for Strep group proportions?
# read in the species table
mpa3_sp <- fread("./05-results.backup/mpa3_abundance_table.tsv") %>%
  select(-NCBI_tax_id)

# clean the file names
colnames(mpa3_sp) <- gsub(".metaphlan3","", colnames(mpa3_sp))
colnames(mpa3_sp) <- gsub(".SG1","", colnames(mpa3_sp))

# filter to have only species
mpa3_sp <- mpa3_sp %>% 
  filter(str_detect(clade_name, "s__")) %>%
  separate(clade_name, into = c("K","P","C","O","F","G","Species"), sep = "\\|") %>%
  select(-c("K","P","C","O","F","G")) %>%
  mutate(Species = str_replace_all(Species, "s__",""),
         Species = str_replace_all(Species, "_"," ")) %>%
  # these samples have no metadata
  select(-c("MID025.A0101","MID033.A0101","MID052.A0101","MID056.A0101","MID065.A0101","MID068.A0101","MID076.A0101","MID078.A0101")) %>%
  arrange(Species)

mpa3_contaminants <- fread("./05-results.backup/contaminant_species_mid_mpa3.tsv", sep = "\t") %>%
  bind_rows(., fread("./05-results.backup/contaminant_species_kilteasheen_mpa3.tsv", sep = "\t")) %>%
  bind_rows(., fread("./05-results.backup/contaminant_species_iberia_mpa3.tsv", sep = "\t")) %>%
  unique()

# remove contaminant species from table
mpa3_species_decontam <- mpa3_sp %>%
  anti_join(., mpa3_contaminants)

outliers_mpa3 <- c("EXB059.A2101","EXB059.A2501","EXB015.A3301","EXB034.A2701","EXB059.A2201","EXB059.A2301","EXB059.A2401","LIB058.A0103","LIB058.A0106","LIB058.A0104")
poor_samples_mpa3 <- fread("./05-results.backup/cuperdec_mpa3_poor_samples.tsv") %>%
  pull(Sample)


```


```{r load_strep_groups}
strep_groups <- fread("./00-documentation.backup/25-streptococcus_cladegroup_amylasegroup_database.tsv") %>%
  select(Taxon, Clade_Consensus) %>%
  rename(Species = Taxon) %>%
  mutate(Species = str_replace_all(Species, "_", " ")) %>%
  rename(Strep_group = Clade_Consensus) %>%
  mutate(Strep_group = fct_relevel(Strep_group, "sanguinis","mitis","salivarius","anginosus","bovis","pyogenic","mutans","unknown","NA"))

# get the proportion of sanguinis and anginosis for each sample
strep_group_info <- mpa3_species_decontam %>%
  filter(str_detect(Species, "Streptococcus")) %>%
  # join with the Strep grup table for plotting
  inner_join(., strep_groups) %>%
  select(-Species) %>%
  select(Strep_group, everything()) %>%
  # transpose
  gather("Library_ID","Counts", 2:ncol(.)) %>%
  group_by(Library_ID, Strep_group) %>%
  summarize(Reads = sum(Counts)) %>%
  spread(Strep_group, Reads) %>%
  # convert table to proportions
  adorn_percentages(denominator = "row") %>%
  # arrange by sanguinis and for the ones w/o sanguinis, by anginosis
  arrange(desc(sanguinis))

```



```{r strep_groups}
# load the MALT RefSeq taxonomy tables that have been decontaminated 
# and the column headers are cleaned up
# this is to get the Strep groups for coloring the PCA plot
load("./05-results.backup/Taxonomy_tables.RData")

# list poor samples from cuperdec
poor_samples_rs <- fread("./05-results.backup/cuperdec_poor_samples.tsv") %>%
  rename(Library_ID = Sample) %>%
  filter(str_detect(Library_ID, "MID")) %>%
  pull(Library_ID)

# we'll use the prevalence-filtered dataset
malt_rs <- all_species_decontam_prev %>%
  # select(matches("Species|MID|CMB")) %>%
  adorn_totals(where = "col")  %>%
  filter(Total >  0) %>%
  select(-Total) %>%
  select(-poor_samples_rs)

strep_groups <- fread("./00-documentation.backup/dRep_strep_groups.tsv") %>%
  # remove the extrac column with the old dRep
  select(-Clade_Consensus) %>%
  # make a factor for plotting
  mutate(dRep = fct_relevel(dRep, "Sanguinis","Mitis","Salivarius","Anginosus","Bovis","Pyogenic","Mutans","Other_oral","Other","Unknown"))

strep_rs <- malt_rs %>%
  filter(str_detect(Species, "Streptococcus")) %>%
  # join with the Strep grup table for plotting
  inner_join(., strep_groups) %>%
  select(Species, dRep, everything()) %>%
  # convert to long-format
  gather("Library_ID","Counts", 3:ncol(.)) %>%
  # mutate(Library_ID = fct_relevel(Library_ID, sample_order_rs)) %>%
  dplyr::group_by(Library_ID, dRep) %>%
  summarize(Count_sum = sum(Counts)) %>%
  ungroup() %>%
  group_by(Library_ID) %>%
  mutate(Percent = Count_sum / sum(Count_sum)) %>%
  select(Library_ID, dRep, Percent) %>%
  pivot_wider(names_from = dRep, values_from = Percent)

```

```{r species_pc1_pct}
# get the percent of species driving PC1+ and PC1- plotting from the MetaPhlan3 table
# Does the abundance of those species also affect plotting of the pathway PCA?

# select the species with strongest loadings in PC1 + and - directions
pc1_mpa3 <- fread("./05-results.backup/mpa3_pc1_species_list.tsv") 

pc1pws <- pc1_mpa3 %>%
  filter(PC1 > 0) %>%
  pull(Taxon)

pc1neg <- pc1_mpa3 %>%
  filter(PC1 < 0) %>%
  pull(Taxon)

# calculate percent of species in PC1 positice top 10 in each sample
pc_species_pws_pc1 <- mpa3_species_decontam %>%
  select(-one_of(outliers_mpa3)) %>%
  select(-one_of(poor_samples_mpa3)) %>%
  select(-matches("EXB|LIB|CSN|CSD|ARS|CSS|CSL")) %>%
  filter(Species %in% pc1pws) %>%
  # convert to long-format
  pivot_longer(!Species, names_to = "Library_ID", values_to = "Counts") %>%
  dplyr::group_by(Library_ID) %>%
  summarize(Count_sum = sum(Counts)) %>%
  ungroup() %>%
  full_join(., mpa3_species_decontam %>%
  select(-one_of(outliers_mpa3)) %>%
  select(-one_of(poor_samples_mpa3)) %>%
  select(-matches("EXB|LIB|CSN|CSD|ARS|CSS|CSL")) %>%
              pivot_longer(!Species, names_to = "Library_ID", values_to = "Counts") %>%
              dplyr::group_by(Library_ID) %>%
              summarize(Total = sum(Counts)) %>%
              select(Library_ID, Total)) %>%
  mutate(Percent_pws_pc1 = Count_sum / Total * 100) %>%
  select(Library_ID, Percent_pws_pc1)

# calculate percent of species in PC1 negative top 10 in each sample
pc_species_neg_pc1 <- mpa3_species_decontam %>%
  select(-one_of(outliers_mpa3)) %>%
  select(-one_of(poor_samples_mpa3)) %>%
  select(-matches("EXB|LIB|CSN|CSD|ARS|CSS|CSL")) %>%
  filter(Species %in% pc1neg) %>%
  # convert to long-format
  pivot_longer(!Species, names_to = "Library_ID", values_to = "Counts") %>%
  dplyr::group_by(Library_ID) %>%
  summarize(Count_sum = sum(Counts)) %>%
  ungroup() %>%
  full_join(., mpa3_species_decontam %>%
  select(-one_of(outliers_mpa3)) %>%
  select(-one_of(poor_samples_mpa3)) %>%
  select(-matches("EXB|LIB|CSN|CSD|ARS|CSS|CSL")) %>%
              pivot_longer(!Species, names_to = "Library_ID", values_to = "Counts") %>%
              dplyr::group_by(Library_ID) %>%
              summarize(Total = sum(Counts)) %>%
              select(Library_ID, Total)) %>%
  mutate(Percent_neg_pc1 = Count_sum / Total * 100) %>%
  select(Library_ID, Percent_neg_pc1)

# add both to the metadata table below

```

```{r species_pc2_pct}
# get the percent of species driving PC1+ and PC1- plotting from the MetaPhlan3 table
# Does the abundance of those species also affect plotting of the pathway PCA?

# select the species with strongest loadings in PC2 + and - directions
pc2_mpa3 <- fread("./05-results.backup/mpa3_pc2_species_list.tsv") 

pc2pws <- pc2_mpa3 %>%
  filter(PC2 > 0) %>%
  pull(Taxon)

pc2neg <- pc2_mpa3 %>%
  filter(PC2 < 0) %>%
  pull(Taxon)

# calculate percent of species in PC2 positice top 10 in each sample
pc_species_pws_pc2 <- mpa3_species_decontam %>%
  select(-one_of(outliers_mpa3)) %>%
  select(-one_of(poor_samples_mpa3)) %>%
  select(-matches("EXB|LIB|CSN|CSD|ARS|CSS|CSL")) %>%
  filter(Species %in% pc2pws) %>%
  # convert to long-format
  pivot_longer(!Species, names_to = "Library_ID", values_to = "Counts") %>%
  dplyr::group_by(Library_ID) %>%
  summarize(Count_sum = sum(Counts)) %>%
  ungroup() %>%
  full_join(., mpa3_species_decontam %>%
  select(-one_of(outliers_mpa3)) %>%
  select(-one_of(poor_samples_mpa3)) %>%
  select(-matches("EXB|LIB|CSN|CSD|ARS|CSS|CSL")) %>%
              pivot_longer(!Species, names_to = "Library_ID", values_to = "Counts") %>%
              dplyr::group_by(Library_ID) %>%
              summarize(Total = sum(Counts)) %>%
              select(Library_ID, Total)) %>%
  mutate(Percent_pws_pc2 = Count_sum / Total * 100) %>%
  select(Library_ID, Percent_pws_pc2)

# calculate percent of species in PC2 negative top 10 in each sample
pc_species_neg_pc2 <- mpa3_species_decontam %>%
  select(-one_of(outliers_mpa3)) %>%
  select(-one_of(poor_samples_mpa3)) %>%
  select(-matches("EXB|LIB|CSN|CSD|ARS|CSS|CSL")) %>%
  filter(Species %in% pc2neg) %>%
  # convert to long-format
  pivot_longer(!Species, names_to = "Library_ID", values_to = "Counts") %>%
  dplyr::group_by(Library_ID) %>%
  summarize(Count_sum = sum(Counts)) %>%
  ungroup() %>%
  full_join(., mpa3_species_decontam %>%
  select(-one_of(outliers_mpa3)) %>%
  select(-one_of(poor_samples_mpa3)) %>%
  select(-matches("EXB|LIB|CSN|CSD|ARS|CSS|CSL")) %>%
              pivot_longer(!Species, names_to = "Library_ID", values_to = "Counts") %>%
              dplyr::group_by(Library_ID) %>%
              summarize(Total = sum(Counts)) %>%
              select(Library_ID, Total)) %>%
  mutate(Percent_neg_pc2 = Count_sum / Total * 100) %>%
  select(Library_ID, Percent_neg_pc2)

# add both to the metadata table below

```

```{r}
# read in the MALT mapped/unmapped gc and read length info from `MID_malt_map_unmap_rl_gc.Rmd`
malt_map_unmap_rl_gc <- fread("./05-results.backup/rl_gc_sub_data.tsv")
colnames(malt_map_unmap_rl_gc) <- gsub("_map","_malt_map", colnames(malt_map_unmap_rl_gc))
colnames(malt_map_unmap_rl_gc) <- gsub("_unmap","_malt_unmap", colnames(malt_map_unmap_rl_gc))

```

```{r}
full_metadata <- full_metadata %>%
  mutate(`DNA_molecules_per_library_x10^6_log` = log10(`DNA_molecules_per_library_x10^6`)) %>%
  full_join(., malt_map_unmap_rl_gc) %>%
  full_join(., strep_rs, by = "Library_ID") %>%
  full_join(., pc_species_pws_pc1, by = "Library_ID") %>%
  full_join(., pc_species_neg_pc1, by = "Library_ID") %>%
  full_join(., pc_species_pws_pc2, by = "Library_ID") %>%
  full_join(., pc_species_neg_pc2, by = "Library_ID")

```


```{r pca_paths}
# create the input matrix
humann3_path.decontam_noblanks_presence_more_30 <- humann3_path %>%
  filter(!str_detect(Pathway, "\\|")) %>%
  # no full_metadata, remove these
  select(-c("MID025.A0101","MID033.A0101","MID052.A0101","MID056.A0101","MID065.A0101","MID068.A0101","MID076.A0101","MID078.A0101")) %>%
  # remove poorly prserved samples
  select(-c("MID024.A0101","MID063.A0101","MID092.A0101")) %>%
  mutate_if(is.numeric, ~1 * (. > 0)) %>%
  gather("Library_ID","Counts",2:ncol(.)) %>%
  group_by(Pathway) %>%
  summarize(Percent_presence = sum(Counts)/length(Library_ID)*100) %>%
  # select the pathways that are present in at least 1/3 of at least 2 groups
  ungroup() %>%
  # pathways present in > 1/3 of samples 
  filter(Percent_presence >= 33.33) %>%
  # pull out these pathways to keep
  select(Pathway)

humann3_path_l1 <- humann3_path %>%
  filter(!str_detect(Pathway, "\\|")) %>%
  # no full_metadata, remove these
  select(-c("MID025.A0101","MID033.A0101","MID052.A0101","MID056.A0101","MID065.A0101","MID068.A0101","MID076.A0101","MID078.A0101")) %>%
  # remove poorly prserved samples
  select(-c("MID024.A0101","MID063.A0101","MID092.A0101")) %>%
  inner_join(., humann3_path.decontam_noblanks_presence_more_30, by = "Pathway") %>%
  gather("Library_ID","Counts",2:ncol(.)) %>%
  mutate(Counts = Counts + 1) %>%
  spread(Pathway,Counts) %>%
  column_to_rownames("Library_ID")

# check the number of components to retain by tuning the PCA
tune.humann3_path_pca <- tune.pca(humann3_path_l1, logratio = 'CLR')

# perform a PCA to see how the data cluster
humann3_path_l1.pca <- pca(humann3_path_l1, ncomp = 3, logratio = 'CLR')

# plot the PCA
plot_pca(humann3_path_l1.pca, "PC1", "PC2", "Site_code", "Site_code", "Site_code")

```

```{r pca_paths_noblanks}

humann3_path_l1 <- humann3_path %>%
  filter(!str_detect(Pathway, "\\|")) %>%
  # no full_metadata, remove these
  select(-c("MID025.A0101","MID033.A0101","MID052.A0101","MID056.A0101","MID065.A0101","MID068.A0101","MID076.A0101","MID078.A0101")) %>%
  # remove poorly prserved samples
  select(-c("MID024.A0101","MID063.A0101","MID092.A0101")) %>%
  select(-matches("EXB|LIB")) %>%
  # inner_join(., humann3_path.decontam_noblanks_presence_more_30, by = "Pathway") %>%
  gather("Library_ID","Counts",2:ncol(.)) %>%
  mutate(Counts = Counts + 1) %>%
  spread(Pathway,Counts) %>%
  column_to_rownames("Library_ID")

# check the number of components to retain by tuning the PCA
tune.humann3_path_pca <- tune.pca(humann3_path_l1, logratio = 'CLR')

# perform a PCA to see how the data cluster
humann3_path_l1.pca <- pca(humann3_path_l1, ncomp = 3, logratio = 'CLR')

# set pipe notch colors 
Pipenotch_colors = c("#5A167E","#E95562")
Pipenotch_shapes = c(16,17)
# Pipenotch_size = c(4,4)

# plot the PCA
plot_pca(humann3_path_l1.pca, "PC1", "PC2", "Pipenotch", "Pipenotch", "Site_code")

```


```{r pca_paths_mid_pc12}
humann3_path_l1 <- humann3_path %>%
  filter(!str_detect(Pathway, "\\|")) %>%
  # no full_metadata, remove these
  select(-c("MID025.A0101","MID033.A0101","MID052.A0101","MID056.A0101","MID065.A0101","MID068.A0101","MID076.A0101","MID078.A0101")) %>%
  # remove poorly prserved samples
  select(-c("MID024.A0101","MID063.A0101","MID092.A0101")) %>%
  select(-matches("EXB|LIB")) %>%
  # inner_join(., humann3_path.decontam_noblanks_presence_more_30, by = "Pathway") %>%
  gather("Library_ID","Counts",2:ncol(.)) %>%
  mutate(Counts = Counts + 1) %>%
  spread(Pathway,Counts) %>%
  column_to_rownames("Library_ID")

# prepare to run a PCA
# check the number of components to retain by tuning the PCA
mixOmics::tune.pca(humann3_path_l1, logratio = 'CLR')


humann3_all_otu.pca <- mixOmics::pca(humann3_path_l1, ncomp = 2, logratio = 'CLR')
humann3_all_pca_values <- humann3_all_otu.pca$variates$X %>%
  as.data.frame() %>%
  rownames_to_column("Library_ID") %>%
  inner_join(., full_metadata, by = "Library_ID") %>%
  left_join(., strep_group_info, by = "Library_ID")

# ggplot(humann3_all_pca_values, aes(PC1, PC2, color = Site_code, shape = Time_period)) +
#   geom_point(size = 2) +
#   # geom_text(aes(label = Library_ID), size = 3) +
#   theme_minimal() +
#   scale_color_manual(values = Site_code_colors) +
#      theme(legend.position = "top")

plot_pca(humann3_path_l1.pca, "PC1", "PC2", "Time_period", "Time_period", "Time_period")
plot_pca(humann3_path_l1.pca, "PC1", "PC2", "Site_code", "Site_code", "Time_period") 


# ggsave("./06-publication/supplemental_figures/SXX8/site_code_beta.pdf", plot = site_code_beta, device = "pdf",
#        scale = 1, width = 6, height = 5, units = c("in"), dpi = 300)

# perform and plot a PCA to see how the data cluster
# by sequencing depth 
plot_pca_cont(humann3_path_l1, "PC1", "PC2", "No_reads_total", "Site_code", 3, "Total reads into MetaPhlAn3")

# by average GC content
plot_pca_cont(humann3_path_l1, "PC1", "PC2", "gc_avg", "Site_code", 3, "Avg. GC content (%)")

# by average read length
plot_pca_cont(humann3_path_l1, "PC1", "PC2", "seq_len_avg", "Site_code", 3, "Avg. read length (bp)")

# by Strep group proportion - sanguinis
plot_pca_cont(humann3_path_l1, "PC1", "PC2", "sanguinis", "Site_code", 3, "Proportion of Sanguinis group")

# by Strep group proportion - anginosus
plot_pca_cont(humann3_path_l1, "PC1", "PC2", "mitis", "Site_code", 3, "Proportion of Mitis group")

# by average read length
rl <- plot_pca_cont(humann3_path_l1, "PC1", "PC2", "seq_len_avg", "Site_code", 3, "Avg. read length (bp)")
rl

```

```{r pca_species_mid_pc12}
humann3_path_l1 <- humann3_path %>%
  filter(!str_detect(Pathway, "\\|")) %>%
  # no full_metadata, remove these
  select(-c("MID025.A0101","MID033.A0101","MID052.A0101","MID056.A0101","MID065.A0101","MID068.A0101","MID076.A0101","MID078.A0101")) %>%
  # remove poorly prserved samples
  select(-c("MID024.A0101","MID063.A0101","MID092.A0101")) %>%
  select(-matches("EXB|LIB")) %>%
  # inner_join(., humann3_path.decontam_noblanks_presence_more_30, by = "Pathway") %>%
  gather("Library_ID","Counts",2:ncol(.)) %>%
  mutate(Counts = Counts + 1) %>%
  spread(Pathway,Counts) %>%
  column_to_rownames("Library_ID")

# prepare to run a PCA
# check the number of components to retain by tuning the PCA
mixOmics::tune.pca(humann3_path_l1, logratio = 'CLR')


humann3_path.pca <- mixOmics::pca(humann3_path_l1, ncomp = 3, logratio = 'CLR')
malt_pca_values <- humann3_path.pca$variates$X %>%
  as.data.frame() %>%
  rownames_to_column("Library_ID") %>%
  inner_join(., full_metadata, by = "Library_ID")

ggplot(malt_pca_values, aes(PC1, PC2, color = Region, shape = Region)) +
  geom_text(aes(label = Library_ID), size = 3) +
  theme_minimal()

# perform and plot a PCA to see how the data cluster
# by pipe notch presence
mid_pipe_pc12 <- plot_pca_mid(humann3_path_l1, "PC1", "PC2", "Pipenotch", "Pipenotch", 3, "Pipe notch")
mid_pipe_pc12
# ggsave("~/Desktop/MID_pca_mid_pipe_pc12.pdf", plot = mid_pipe_pc12, device = "pdf",
#        scale = 1, width = 9, height = 5, units = c("in"), dpi = 300)

# by sex
plot_pca_mid(humann3_path_l1, "PC1", "PC2", "Sex", "Pipenotch", 3, "Sex")

# by age
plot_pca_mid(humann3_path_l1, "PC1", "PC2", "Age", "Pipenotch", 3, "Age")

# by presence of periodontal disease
plot_pca_mid(humann3_path_l1, "PC1", "PC2", "Periodontal_disease_present", "Pipenotch", 3, "Periodontal disease present")

# by presence of caries
plot_pca_mid(humann3_path_l1, "PC1", "PC2", "Caries_present", "Pipenotch", 3, "Caries present")

# by ante-mortem tooth loss
plot_pca_mid(humann3_path_l1, "PC1", "PC2", "AMTL", "Pipenotch", 3, "AMTL")

# by supragingival calculus score
plot_pca_cont(humann3_path_l1, "PC1", "PC2", "calc_sup_SI_score", "Pipenotch", 3, "Sup calc SI score")

# by subgingival calculus score
plot_pca_cont(humann3_path_l1, "PC1", "PC2", "Calc_sub_SI_score", "Pipenotch", 3, "Sub calc SI score")

# by presence of perioapical lesions
plot_pca_mid(humann3_path_l1, "PC1", "PC2", "Periapical_lesions", "Pipenotch", 3, "Periapical lesions")

# by max periodontal disease score
plot_pca_cont(humann3_path_l1, "PC1", "PC2", "Max_Perio_Score", "Pipenotch", 3, "Max perio score")

# by periodontal affect score
plot_pca_cont(humann3_path_l1, "PC1", "PC2", "Peridontal_affect_score", "Pipenotch", 3, "Perio affect score")

# by % antemortem tooth loss
plot_pca_cont(humann3_path_l1, "PC1", "PC2", "%AMTL", "Pipenotch", 3, " % AMTL")

# by % positions w/periapical lesions
plot_pca_cont(humann3_path_l1, "PC1", "PC2", "%_positions_perioapical", "Pipenotch", 3, "% positions periapical lesions")

# by % teeth with caries
plot_pca_cont(humann3_path_l1, "PC1", "PC2", "%teeth_with_caries", "Pipenotch", 3, "% teeth w/ caries")

# by % teeth with calculus
plot_pca_cont(humann3_path_l1, "PC1", "PC2", "%teeth_with_calc", "Pipenotch", 3, "% teeth w/ calc")

# by % positions with periodontal disease
plot_pca_cont(humann3_path_l1, "PC1", "PC2", "%_positions_with_Periodontal", "Pipenotch", 3, "% positions w/ PD")

# by minimum number of pipenotches 
mid_minpipe_pc12 <- plot_pca_cont(humann3_path_l1, "PC1", "PC2", "Min_no_Pipe_notches", "Pipenotch", 3, "Min # pipe notches")
mid_minpipe_pc12
# ggsave("~/Desktop/MID_pca_mid_minpipe_pc12.pdf", plot = mid_minpipe_pc12, device = "pdf",
#        scale = 1, width = 9, height = 5, units = c("in"), dpi = 300)

# by sequencing depth 
mid_seqdepth_pc12 <- plot_pca_cont(humann3_path_l1, "PC1", "PC2", "No_reads_total", "Pipenotch", 3, "Total reads into MetaPhlAn3")
mid_seqdepth_pc12
# ggsave("~/Desktop/MID_pca_mid_seqdepth_pc12.pdf", plot = mid_seqdepth_pc12, device = "pdf",
#        scale = 1, width = 9, height = 5, units = c("in"), dpi = 300)

# by sampled tooth site(s)
mid_tooth_pc12 <- plot_pca_mid(humann3_path_l1, "PC1", "PC2", "Tooth_location", "Pipenotch", 3, "Tooth location")
mid_tooth_pc12
# ggsave("~/Desktop/MID_pca_mid_tooth_pc12.pdf", plot = mid_tooth_pc12, device = "pdf",
#        scale = 1, width = 9, height = 5, units = c("in"), dpi = 300)

# by sampled tooth site(s)
plot_pca_mid(humann3_path_l1, "PC1", "PC2", "alt_tooth_location", "Pipenotch", 3, "Tooth location")

# by mg of extracted calculus
plot_pca_cont(humann3_path_l1, "PC1", "PC2", "Calculus_extracted_mg", "Pipenotch", 3, "Weight of extracted calculus")

# by [DNA] after extraction per mg of calculus extracted
plot_pca_cont(humann3_path_l1, "PC1", "PC2", "extract_conc_ng/mg", "Pipenotch", 3, "DNA extract concentration (ng/mg)")

# by [DNA] after extraction per uL of eluate
mid_extract_pc12 <- plot_pca_cont(humann3_path_l1, "PC1", "PC2", "extract_conc_ng/uL", "Pipenotch", 3, "DNA extract concentration (ng/uL)")
mid_extract_pc12
# ggsave("~/Desktop/MID_pca_mid_extract_pc12.pdf", plot = mid_extract_pc12, device = "pdf",
#        scale = 1, width = 9, height = 5, units = c("in"), dpi = 300)

# by soil type
plot_pca_mid(humann3_path_l1, "PC1", "PC2", "Soil", "Pipenotch", 3, "Soil type")

# by average GC content
plot_pca_cont(humann3_path_l1, "PC1", "PC2", "gc_avg", "Time_period", 3, "Avg. GC content (%)")

# by average read length
plot_pca_cont(humann3_path_l1, "PC1", "PC2", "seq_len_avg", "Time_period", 3, "Avg. read length (bp)")

# by Strep group proportion from MALT RefSeq - Sanguinis
plot_pca_cont(humann3_path_l1, "PC1", "PC2", "Sanguinis", "Time_period", 3, "Proportion of Sanguinis group")

# by Strep group proportion from MALT RefSeq - Anginosus
plot_pca_cont(humann3_path_l1, "PC1", "PC2", "Anginosus", "Time_period", 3, "Proportion of Anginosus group")

# by PC1+ species percent
plot_pca_cont(humann3_path_l1, "PC1", "PC2", "Percent_pws_pc1", "Time_period", 3, "Proportion of PC1+ species")

# by by PC1- sepcies percent
plot_pca_cont(humann3_path_l1, "PC1", "PC2", "Percent_neg_pc1", "Time_period", 3, "Proportion of PC1- species")

# by PC2+ species percent
plot_pca_cont(humann3_path_l1, "PC1", "PC2", "Percent_pws_pc2", "Time_period", 3, "Proportion of PC2+ species")

# by by PC2- sepcies percent
plot_pca_cont(humann3_path_l1, "PC1", "PC2", "Percent_neg_pc2", "Time_period", 3, "Proportion of PC2- species")


```

Now let's look at which pathways are driving separation of the sample groups.
Which of the pathways have the strongest loadings in PC1 and PC2?
```{r pca_path_biplot}
# which paths distinguish sample types when HMP and JAE are included?
pca_EG_biplot_pc1 <- plot_pca_bi(humann3_path.pca, "PC1", "PC2", "Pipenotch", PC1, pathways = T)
pca_EG_biplot_pc1

# ggsave("./05-results.backup/presentation_pdfs/pca_EG_bi_noyanomami_paths.pdf", plot = pca_EG_biplot_pc1, device = "pdf",
#        scale = 1, width = 7, height = 5, units = c("in"), dpi = 300)

# what are those pathways?
pandoc.table(pca_EG_biplot_pc1$plot_env$pws_10 %>% select(-Function) %>% arrange(desc(PC1)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC1 positive loadings")

pandoc.table(pca_EG_biplot_pc1$plot_env$neg_10 %>% select(-Function) %>% arrange(PC1),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC1 negative loadings")

# make a table of the pathways to save, to use again later in another R notebook
humann3_pathway_biplot_list <- pca_EG_biplot_pc1$plot_env$pws_10 %>% select(-Function) %>% arrange(desc(PC1)) %>% select(pathway) %>% mutate(Direction = "PC1+")
humann3_pathway_biplot_list <- humann3_pathway_biplot_list %>%
  bind_rows(pca_EG_biplot_pc1$plot_env$neg_10 %>% select(-Function) %>% arrange(desc(PC1)) %>% select(pathway)%>% mutate(Direction = "PC1-"))

# which paths distinguish sample types when HMP and JAE are included?
pca_EG_biplot_pc2 <- plot_pca_bi(humann3_path.pca, "PC1", "PC2", "Pipenotch", PC2, pathways = T)
pca_EG_biplot_pc2

# what are those pathways?
pandoc.table(pca_EG_biplot_pc2$plot_env$pws_10 %>% select(-Function) %>% arrange(desc(PC2)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC2 positive loadings")

pandoc.table(pca_EG_biplot_pc2$plot_env$neg_10 %>% select(-Function) %>% arrange(PC2),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC2 negative loadings")

humann3_pathway_biplot_list <- humann3_pathway_biplot_list %>%
  bind_rows(pca_EG_biplot_pc2$plot_env$pws_10 %>% select(-Function) %>% arrange(desc(PC2)) %>% select(pathway) %>% mutate(Direction = "PC2+"))
humann3_pathway_biplot_list <- humann3_pathway_biplot_list %>%
  bind_rows(pca_EG_biplot_pc2$plot_env$neg_10 %>% select(-Function) %>% arrange(desc(PC2)) %>% select(pathway) %>% mutate(Direction = "PC2-"))


fwrite(humann3_pathway_biplot_list, file = "./05-results.backup/humann3_pathway_biplot_list.tsv", quote = F, sep = "\t")

```

## Canonical correlation analysis
```{r pca_species_mid}
# read in the speies correlations to add here later
mpa3_sp_pc_corrs <- fread("./05-results.backup/mid_species_pca_values.tsv") %>%
  select(Library_ID, PC1, PC2) %>%
  rename(PC1_sp = PC1,
         PC2_sp = PC2)

humann3_path_l1 <- humann3_path %>%
  filter(!str_detect(Pathway, "\\|")) %>%
  # no full_metadata, remove these
  select(-c("MID025.A0101","MID033.A0101","MID052.A0101","MID056.A0101","MID065.A0101","MID068.A0101","MID076.A0101","MID078.A0101")) %>%
  select(-matches("EXB|LIB")) %>%
  # inner_join(., humann3_path.decontam_noblanks_presence_more_30, by = "Pathway") %>%
  gather("Library_ID","Counts",2:ncol(.)) %>%
  mutate(Counts = Counts + 1) %>%
  spread(Pathway,Counts) %>%
  column_to_rownames("Library_ID")

# prepare to run a PCA
# check the number of components to retain by tuning the PCA
mixOmics::tune.pca(humann3_path_l1, logratio = 'CLR')

humann3_path_l1.pca <- mixOmics::pca(humann3_path_l1, ncomp = 2, logratio = 'CLR')

h3_paths_pca_values <- humann3_path_l1.pca$variates$X %>%
  as.data.frame() %>%
  rownames_to_column("Library_ID") %>%
  inner_join(., full_metadata, by = "Library_ID") %>%
  drop_na(Site_code) %>%
  mutate(Sex_n = ifelse(Sex == "Female", 0,
                        ifelse(Sex == "Male", 1,2))) %>%
  mutate(Pipenotch_n = ifelse(Pipenotch == "Present",0,
                              ifelse(Pipenotch == "Absent",1,2))) %>%
  mutate(Age_n = ifelse(Age == "13-18",0,
                        ifelse(Age == "18-25",1,
                               ifelse(Age == "25-45",2,
                                      ifelse(Age == "45+",4,5))))) %>%
  full_join(., mpa3_sp_pc_corrs, by = "Library_ID")


```

```{r mid_pc_correlations}
# Compute Canonical Correlation Analysis (CCA) 
# between all pairs of selected variables
# returns absolute correlation value

# rename vriables to look nice in figures
h3_paths_pca_values_select <- h3_paths_pca_values %>%
  select(Library_ID, PC1, PC2, PC1_sp, PC2_sp, Percent_pws_pc1, Percent_neg_pc1, Percent_pws_pc2, Percent_neg_pc2, seq_len_avg, gc_avg, No_reads_total, Sanguinis, Anginosus, Age_n, Sex_n, Pipenotch_n, Min_no_Pipe_notches, `%AMTL`, `%teeth_with_caries`, calc_sup_SI_score, Peridontal_affect_score, `%_positions_with_Periodontal`, Max_Perio_Score, Calculus_extracted_mg, `extract_conc_ng/mg`, `extract_conc_ng/uL`, `extract_conc_after_storage_ng/uL`, `DNA_molecules_per_library_x10^6`,Calc_sub_SI_score, `%teeth_with_calc`, `%_positions_perioapical`) %>%
  rename(`Seq length` = seq_len_avg,
         GC = gc_avg,
         `Total reads` = No_reads_total,
         Age = Age_n,
         Sex = Sex_n,
         Pipenotch = Pipenotch_n,
         `Min no. pipenotch` = Min_no_Pipe_notches,
         `% AMTL` = `%AMTL`,
         `% Caries` = `%teeth_with_caries`,
         `Supra calc score` = calc_sup_SI_score,
         `Perio score` = Peridontal_affect_score,
         `% Perio` = `%_positions_with_Periodontal`,
         `Max perio score` = Max_Perio_Score,
         `Extracted weight (mg)` = Calculus_extracted_mg,
         `Pre-storage [extract] (ng/mg)` = `extract_conc_ng/mg`,
         `Pre-storage [extract] (ng/uL)` = `extract_conc_ng/uL`,
         `Post-storage [extract] (ng/uL)` = `extract_conc_after_storage_ng/uL`,
         `[Library] (molecules)` = `DNA_molecules_per_library_x10^6`,
         `Sub calc score` = Calc_sub_SI_score,
         `% Calculus` = `%teeth_with_calc`,
         `% Perioapical` = `%_positions_perioapical`,
         `Species PC1` = PC1_sp,
         `Species PC2` = PC2_sp,
         `Species % PC1+` = Percent_pws_pc1,
         `Species % PC1-` = Percent_neg_pc1,
         `Species % PC2+` = Percent_pws_pc2,
         `Species % PC2-` = Percent_neg_pc2) %>%
  # re-order the columns
  select(Library_ID, PC1, PC2, Age, Sex, Pipenotch, `Min no. pipenotch`, `% AMTL`, `% Caries`, `Supra calc score`, `Sub calc score`, `% Calculus`, `Perio score`, `% Perio`, `Max perio score`, `% Perioapical`, `Extracted weight (mg)`, `Pre-storage [extract] (ng/mg)`, `Pre-storage [extract] (ng/uL)`, `Post-storage [extract] (ng/uL)`, `[Library] (molecules)`, `Seq length`, GC, `Total reads`, Sanguinis, Anginosus, `Species PC1`, `Species PC2`, `Species % PC1+`, `Species % PC1-`, `Species % PC2+`, `Species % PC2-`)

# select variables
form <- ~ PC1 + PC2 + Age + Sex + Pipenotch + `Min no. pipenotch` + `% AMTL` + `% Caries` + `Supra calc score` + `Sub calc score` + `% Calculus` + `Perio score` + `% Perio` + `Max perio score` + `% Perioapical` + `Extracted weight (mg)` + `Pre-storage [extract] (ng/mg)` + `Pre-storage [extract] (ng/uL)` + `Post-storage [extract] (ng/uL)` + `[Library] (molecules)` + `Seq length` + GC + `Total reads` + Sanguinis + Anginosus + `Species PC1` + `Species PC2` +`Species % PC1+` + `Species % PC1-` + `Species % PC2+` + `Species % PC2-`

# compute CCA
C_mid = canCorPairs( form, h3_paths_pca_values_select)

```

```{r pearson_corr_mid}
# test for significant differences in diversity between the metadata categories
pearson_mid <- h3_paths_pca_values_select %>%
rstatix::cor_test(., PC1, PC2, Age, Sex, Pipenotch, `Min no. pipenotch`, `% AMTL`, `% Caries`, `Supra calc score`, `Sub calc score`, `% Calculus`, `Perio score`, `% Perio`, `Max perio score`, `% Perioapical`, `Extracted weight (mg)`, `Pre-storage [extract] (ng/mg)`, `Pre-storage [extract] (ng/uL)`, `Post-storage [extract] (ng/uL)`, `[Library] (molecules)`, `Seq length`, GC, `Total reads`, Sanguinis, Anginosus, `Species PC1`, `Species PC2`, `Species % PC1+`, `Species % PC1-`, `Species % PC2+`, `Species % PC2-`, method="pearson")

pearson_mid %>%
  mutate(same = var1 == var2) %>%
  filter(same == FALSE) %>%
  select(-same) %>%
  filter(p <= 0.001) %>%
  filter(cor > 0.40 | cor < -0.40) %>%
  distinct(statistic, .keep_all = TRUE)
  
```

```{r}

C_mid %>%
  as_data_frame(.) %>%
  mutate(names = rownames(C_mid)) %>%
  select(names, everything()) %>%
  pivot_longer(!names, names_to = "Corrs", values_to = "values") %>%
  filter(values != 1) %>%
  filter(values >= 0.4) %>%
  distinct(values, .keep_all = TRUE)

```

```{r}

colorset <- c("#ffffff","#e8e2ee","#d1c6dc","#baaacb","#a48fba","#8d74aa","#775b99","#604289","#492978","#311068")

pvals_mid <- h3_paths_pca_values_select %>%
  column_to_rownames("Library_ID") %>%
  as.matrix(.) %>%
  corrplot::cor.mtest(., method="pearson")

mid_corr_plot <- C_mid %>%
  corrplot::corrplot(., p.mat = pvals_mid$p, order = "original", type = "full", col = colorset, is.corr = FALSE, cl.cex = 0.95, diag = FALSE, #tl.srt = 45,
                     tl.col = 'black', sig.level = c(0.001), pch.cex = 1.2, insig = 'label_sig', pch.col = 'grey20')
mid_corr_plot

```


```{r, eval = F}
svg(file = "./06-publication/supplemental_figures/SXX14/mid_h3_path_corr_plot.svg") 

C_mid %>%
  corrplot::corrplot(., p.mat = pvals_mid$p, order = "original", type = "full", col = colorset, is.corr = FALSE, cl.cex = 0.95, diag = FALSE, #tl.srt = 45,
                     tl.col = 'black', sig.level = c(0.001), pch.cex = 1.2, insig = 'label_sig', pch.col = 'grey20')


dev.off()

```














