---
title: "MID/CMB, Radcliffe, Kilteasheen, Iberian medieval calculus MALT beta diversity"
author: "Irina Velsko"
date: "`r format(Sys.time(), '%b %d, %Y')`"
output:
    github_document:
        fig_width: 8
        fig_height: 6
        toc: true
---


```{r load_libraries, echo = F, message = F}
library(knitr)
library(data.table)
library(psych)
library(ape)
library(mixOmics)
library(compositions)
library(vegan)
library(janitor)
library(pander)
library(tidyverse)
library(gplots)
library(ggrepel)
library(viridis)
library(cowplot)
opts_chunk$set(echo=F, warning=F, message=F, dpi=150)
```

Set the working directory of the notebook to the parent directory of the 
notebook itself.  

```{r setup}
knitr::opts_knit$set(root.dir = normalizePath(".."))
```

# MALT read stats

```{r load_data}
# read in the species table
nt_raw <- fread("./05-results.backup/MID_CMB_malt_nt-comparison-species.txt")

# clean the file names
nt_raw <- rename(nt_raw, Species = `#Datasets`)
colnames(nt_raw) <- gsub(".unmapped","", colnames(nt_raw))
colnames(nt_raw) <- gsub(".10M","", colnames(nt_raw))

# remove human and unassigned reads, convert NA to 0, and remove the 8 MID samples with no metadata
nt_raw <- nt_raw %>% 
  filter(!str_detect(Species, "Homo sapiens|Not assigned")) %>%
  replace(is.na(.), 0) %>%
  # these samples have no metadata
  select(-c("MID025.A0101","MID033.A0101","MID052.A0101","MID056.A0101","MID065.A0101","MID068.A0101","MID076.A0101","MID078.A0101")) %>%
  arrange(Species)

```

Load in the original table to pull the LIB and EXB samples
```{r}
load("./05-results.backup/Taxonomy_tables.RData")

```

```{r}
# merge the EXB and LIB samples from here with the raw tables

nt_raw <- nt_raw %>%
  full_join(., all_species_raw %>%
              select(matches("Species|EXB|LIB")), by = "Species") %>%
  replace(is.na(.), 0) %>%
  adorn_totals(where = "col") %>%
  filter(Total > 0) %>%
  select(-Total)
   
# sub75bp_raw <- sub75bp_raw %>%
#   full_join(., all_species_raw %>%
#               select(matches("Species|EXB|LIB")), by = "Species") %>%
#   replace(is.na(.), 0) %>%
#   adorn_totals(where = "col") %>%
#   filter(Total > 0) %>%
#   select(-Total)
 
```

Now decontam the tables
```{r}
nt_contaminants <- fread("./05-results.backup/contaminant_species_nt.tsv", sep = "\t")
# sub75bp_contaminants <- fread("./05-results.backup/contaminant_species_sub75bp.tsv", sep = "\t")

# remove contaminant species from nt table
nt_species_decontam <- nt_raw %>%
  anti_join(., nt_contaminants)

# remove contaminant species from nt table
# sub75bp_species_decontam <- sub75bp_raw %>%
#   anti_join(., sub75bp_contaminants)

```


```{r load_metadata}

# load the metadata file
load("./05-results.backup/Metadata_tables.RData")

# read in the MALT mapped/unmapped gc and read length info from `MID_malt_map_unmap_rl_gc.Rmd`
malt_map_unmap_rl_gc <- fread("./05-results.backup/rl_gc_sub_data.tsv")
colnames(malt_map_unmap_rl_gc) <- gsub("_map","_malt_map", colnames(malt_map_unmap_rl_gc))
colnames(malt_map_unmap_rl_gc) <- gsub("_unmap","_malt_unmap", colnames(malt_map_unmap_rl_gc))

full_metadata <- full_metadata %>%
  mutate(`DNA_molecules_per_library_x10^6_log` = log10(`DNA_molecules_per_library_x10^6`)) %>%
  full_join(., malt_map_unmap_rl_gc)

```
 
```{r pca_name_fxn}
# this function plots the PCA with the sample names instead of points
plot_pca_names <- function(df, pc1, pc2, metadata_group) {
    metadata_group_colors <- get(paste(metadata_group, "_colors", sep = ""))

    exp_var <- paste0(round(df$explained_variance * 100, 2), "%")
    df_X <- df$variates$X %>%
              as.data.frame() %>%
              rownames_to_column("Library_ID") %>%
              inner_join(full_metadata %>%
                           select(Library_ID, Site, Region, Time_period, Sample_or_Control, Periodontal_disease_present, Caries_present, Sex, Age, Tooth_location), by = "Library_ID")
    
    metadata_group = df_X[[metadata_group]]
    
                       if (pc1 == 'PC1') {
                           pc1 <- df_X$PC1
                           exp_var_pc1 <- exp_var[1]
                           xaxis <- c("PC1")
                       }  else if (pc1 == 'PC2') {
                           pc1 <- df_X$PC2
                           exp_var_pc1 <- exp_var[2]
                           xaxis <- c("PC2")
                       } else if (pc1 == 'PC3') {
                           pc1 <- df_X$PC3
                           exp_var_pc1 <- exp_var[3]
                           xaxis <- c("PC3")
                       }

                       if (pc2 == 'PC1') {
                           pc2 <- df_X$PC1
                           exp_var_pc2 <- exp_var[1]
                           yaxis <- c("PC1")
                       }  else if (pc2 == 'PC2') {
                           pc2 <- df_X$PC2
                           exp_var_pc2 <- exp_var[2]
                           yaxis <- c("PC2")
                       } else if (pc2 == 'PC3') {
                           pc2 <- df_X$PC3
                           exp_var_pc2 <- exp_var[3]
                           yaxis <- c("PC3")
                       }

    pca_plot_names <- ggplot(df_X, aes(pc1, pc2, colour = metadata_group)) +
                        geom_text(aes(label = df_X$Library_ID), size = 4) +
                        scale_colour_manual(values = metadata_group_colors) +
                        # stat_ellipse() +
                        xlab(paste(xaxis, " - ", exp_var_pc1)) +
                        ylab(paste(yaxis, " - ", exp_var_pc2)) +
                        theme_minimal(base_size = 16) +
                        theme(text = element_text(size=16)) +
                        theme(legend.position = "top") 
    # +
    #                     theme(legend.title = element_blank())

    return(pca_plot_names)
}

```

```{r pca_plot_fxn}
# plot PCA with colored dots and the title including the # of species or genera
plot_pca <- function(df, pc1, pc2, color_group, shape_group, ncomps) {
    metadata_group_colors <- get(paste(color_group, "_colors", sep = ""))
    metadata_group_shapes <- get(paste(shape_group, "_shapes", sep = ""))

    pca.list <- mixOmics::pca(df, ncomp = ncomps, logratio = 'CLR')

    exp_var <- paste0(round(pca.list$explained_variance * 100, 2), "%")
    df_X <- pca.list$variates$X %>%
              as.data.frame() %>%
              rownames_to_column("Library_ID") %>%
              inner_join(full_metadata, by = "Library_ID")
    
    color_group = df_X[[color_group]]
    shape_group = df_X[[shape_group]]

                       if (pc1 == 'PC1') {
                           pc1 <- df_X$PC1
                           exp_var_pc1 <- exp_var[1]
                           xaxis <- c("PC1")
                       }  else if (pc1 == 'PC2') {
                           pc1 <- df_X$PC2
                           exp_var_pc1 <- exp_var[2]
                           xaxis <- c("PC2")
                       } else if (pc1 == 'PC3') {
                           pc1 <- df_X$PC3
                           exp_var_pc1 <- exp_var[3]
                           xaxis <- c("PC3")
                       }

                       if (pc2 == 'PC1') {
                           pc2 <- df_X$PC1
                           exp_var_pc2 <- exp_var[1]
                           yaxis <- c("PC1")
                       }  else if (pc2 == 'PC2') {
                           pc2 <- df_X$PC2
                           exp_var_pc2 <- exp_var[2]
                           yaxis <- c("PC2")
                       } else if (pc2 == 'PC3') {
                           pc2 <- df_X$PC3
                           exp_var_pc2 <- exp_var[3]
                           yaxis <- c("PC3")
                       }

    pca_plot <- ggplot(df_X, aes(pc1, pc2, colour = color_group, shape = shape_group)) +
                        geom_point(size = 2) +
                        scale_colour_manual(values = metadata_group_colors) +
                        scale_shape_manual(values = metadata_group_shapes) +
                        # stat_ellipse() +
                        xlab(paste(xaxis, " - ", exp_var_pc1)) +
                        ylab(paste(yaxis, " - ", exp_var_pc2)) +
                        theme_minimal(base_size = 16) +
                        theme(text = element_text(size=16)) +
                        theme(legend.title = element_blank(),
                              legend.key.size = unit(2,"mm"),
                              legend.text = element_text(size = 6)) +
                        theme(legend.position = "top")

    return(pca_plot)
}

```

```{r pca_plot_mid_fxn}
# plot PCA with colored dots 
plot_pca_mid <- function(df, pc1, pc2, color_group, shape_group, ncomps, title_text) {

    pca.list <- mixOmics::pca(df, ncomp = ncomps, logratio = 'CLR')

    exp_var <- paste0(round(pca.list$explained_variance * 100, 2), "%")
    df_X <- pca.list$variates$X %>%
              as.data.frame() %>%
              rownames_to_column("Library_ID") %>%
              inner_join(full_metadata, by = "Library_ID")
    
    color_group = df_X[[color_group]]
    shape_group = df_X[[shape_group]]

                       if (pc1 == 'PC1') {
                           pc1 <- df_X$PC1
                           exp_var_pc1 <- exp_var[1]
                           xaxis <- c("PC1")
                       }  else if (pc1 == 'PC2') {
                           pc1 <- df_X$PC2
                           exp_var_pc1 <- exp_var[2]
                           xaxis <- c("PC2")
                       } else if (pc1 == 'PC3') {
                           pc1 <- df_X$PC3
                           exp_var_pc1 <- exp_var[3]
                           xaxis <- c("PC3")
                       }

                       if (pc2 == 'PC1') {
                           pc2 <- df_X$PC1
                           exp_var_pc2 <- exp_var[1]
                           yaxis <- c("PC1")
                       }  else if (pc2 == 'PC2') {
                           pc2 <- df_X$PC2
                           exp_var_pc2 <- exp_var[2]
                           yaxis <- c("PC2")
                       } else if (pc2 == 'PC3') {
                           pc2 <- df_X$PC3
                           exp_var_pc2 <- exp_var[3]
                           yaxis <- c("PC3")
                       }

    pca_plot <- ggplot(df_X, aes(pc1, pc2, colour = color_group, shape = shape_group)) +
                        geom_point(size = 2) +
                        scale_colour_manual(values = c("#5A167E","#E95562","#FED395","#252525","#311068","#C83E73","#FEA873")) + # 
                        scale_shape_manual(values = c(16, 17, 15, 18)) +
                        # stat_ellipse() +
                        xlab(paste(xaxis, " - ", exp_var_pc1)) +
                        ylab(paste(yaxis, " - ", exp_var_pc2)) +
                        theme_minimal(base_size = 16) +
                        theme(text = element_text(size=16)) +
                        theme(legend.title = element_blank(),
                              legend.key.size = unit(2,"mm"),
                              legend.text = element_text(size = 6)) + # 
                        theme(legend.position = "top") +
                        ggtitle(title_text) + theme(plot.title = element_text(size = 10))

    return(pca_plot)
}

```

```{r pca_plot_cont_fxn}
# for continuous data
plot_pca_cont <- function(df, pc1, pc2, color_group, shape_group, ncomps, title_text) {

    pca.list <- mixOmics::pca(df, ncomp = ncomps, logratio = 'CLR')

    exp_var <- paste0(round(pca.list$explained_variance * 100, 2), "%")
    df_X <- pca.list$variates$X %>%
              as.data.frame() %>%
              rownames_to_column("Library_ID") %>%
              inner_join(full_metadata, by = "Library_ID") 
    # %>%
              # full_join(., full_source_results %>%
              #             rename(Library_ID = SampleID),  by = "Library_ID")
    
    color_group = df_X[[color_group]]
    shape_group = df_X[[shape_group]]

                       if (pc1 == 'PC1') {
                           pc1 <- df_X$PC1
                           exp_var_pc1 <- exp_var[1]
                           xaxis <- c("PC1")
                       }  else if (pc1 == 'PC2') {
                           pc1 <- df_X$PC2
                           exp_var_pc1 <- exp_var[2]
                           xaxis <- c("PC2")
                       } else if (pc1 == 'PC3') {
                           pc1 <- df_X$PC3
                           exp_var_pc1 <- exp_var[3]
                           xaxis <- c("PC3")
                       }

                       if (pc2 == 'PC1') {
                           pc2 <- df_X$PC1
                           exp_var_pc2 <- exp_var[1]
                           yaxis <- c("PC1")
                       }  else if (pc2 == 'PC2') {
                           pc2 <- df_X$PC2
                           exp_var_pc2 <- exp_var[2]
                           yaxis <- c("PC2")
                       } else if (pc2 == 'PC3') {
                           pc2 <- df_X$PC3
                           exp_var_pc2 <- exp_var[3]
                           yaxis <- c("PC3")
                       }

    pca_plot <- ggplot(df_X, aes(pc1, pc2, colour = color_group, shape = shape_group)) +
                        geom_point(size = 2) +
                        scale_colour_viridis_c(option = "C") +
                        scale_shape_manual(values = c(16, 17, 15, 18)) +
                        # stat_ellipse() +
                        xlab(paste(xaxis, " - ", exp_var_pc1)) +
                        ylab(paste(yaxis, " - ", exp_var_pc2)) +
                        theme_minimal(base_size = 16) +
                        theme(text = element_text(size=16)) +
                        theme(legend.title = element_blank(),
                              legend.key.size = unit(2,"mm"),
                              legend.text = element_text(size = 6)) + 
                        theme(legend.position = "top") +
                        ggtitle(title_text) + theme(plot.title = element_text(size = 10))

    return(pca_plot)
}


```

```{r pca_biplot_fxn}
plot_pca_bi <- function(df, pc1, pc2, metadata_group, columntitle) {
    metadata_group_colors <- get(paste(metadata_group, "_colors", sep = ""))
    metadata_group_shapes <- get(paste(metadata_group, "_shapes", sep = ""))
   
    arrow_pc <- enquo(columntitle)
    
    exp_var <- paste0(round(df$explained_variance * 100, 2), "%") # explained variance for x- and y-labels
    
    # select only the PCs from the PCA and add metadata
    df_X <- df$variates$X %>%
              as.data.frame() %>%
              rownames_to_column("Library_ID") %>%
              inner_join(full_metadata, by = "Library_ID")
    
    metadata_group = df_X[[metadata_group]]
    
    corr_lam <- df$sdev[c("PC1", "PC2", "PC3")] * sqrt(nrow(df_X))

    df_X <- df_X %>%
      mutate(PC1 = PC1 / corr_lam[1],
             PC2 = PC2 / corr_lam[2], 
             PC3 = PC3 / corr_lam[3])

    # select the correct PC column and explained variance for PC1
    if (pc1 == 'PC1') {
        Pc1 <- df_X$PC1
        exp_var_pc1 <- exp_var[1]
        xaxis <- c("PC1")
    } else if (pc1 == 'PC2') {
        Pc1 <- df_X$PC2
        exp_var_pc1 <- exp_var[2]
        xaxis <- c("PC2")
    } else if (pc1 == 'PC3') {
        Pc1 <- df_X$PC3
        exp_var_pc1 <- exp_var[3]
        xaxis <- c("PC3")
   }
    
    # select the correct PC column and explained variance for PC2
    if (pc2 == 'PC1') {
        Pc2 <- df_X$PC1
        exp_var_pc2 <- exp_var[1]
        yaxis <- c("PC1")
 } else if (pc2 == 'PC2') {
       Pc2 <- df_X$PC2
       exp_var_pc2 <- exp_var[2]
       yaxis <- c("PC2")
    } else if (pc2 == 'PC3') {
       Pc2 <- df_X$PC3
       exp_var_pc2 <- exp_var[3]
       yaxis <- c("PC3")
   }
    
    # Identify the 10 pathways that have highest positive and negative loadings in the selected PC
    pws_10 <- df$loadings$X %>%
      as.data.frame(.) %>%
      rownames_to_column(var = "Taxon") %>%
      top_n(10, !!arrow_pc)

    neg_10 <- df$loadings$X %>%
      as.data.frame(.) %>%
      rownames_to_column(var = "Taxon") %>%
      top_n(-10, !!arrow_pc)


    pca_plot_bi <- ggplot(df_X, aes(x = Pc1, y = Pc2, colour = metadata_group)) +
      geom_point(size = 2.5, aes(shape = metadata_group) )+
      geom_segment(data = pws_10,
                   aes(xend = get(paste(pc1)), yend = get(paste(pc2))),
                   x = 0, y = 0, colour = "black",
                   size = 0.5,
                   arrow = arrow(length = unit(0.03, "npc"))) +
      geom_label_repel(data = pws_10,
                   aes(x = get(paste(pc1)), y = get(paste(pc2)), label = Taxon),
                   size = 1.8, colour = "grey20", label.padding = 0.2) +
      geom_segment(data = neg_10,
                   aes(xend = get(paste(pc1)), yend = get(paste(pc2))),
                   x = 0, y = 0, colour = "grey50",
                   size = 0.5,
                   arrow = arrow(length = unit(0.03, "npc"))) +
      geom_label_repel(data = neg_10,
                   aes(x = get(paste(pc1)), y = get(paste(pc2)), label = Taxon),
                   size = 1.8, colour = "grey20", label.padding = 0.2) +
      labs(x = paste(xaxis, " - ", exp_var_pc1),
           y = paste(yaxis, " - ", exp_var_pc2)) +
      scale_colour_manual(values = metadata_group_colors) +
      scale_shape_manual(values = metadata_group_shapes) +
      theme_minimal() + theme(text = element_text(size = 16)) +
      theme(text = element_text(size=16)) +
      theme(legend.position = "top")

    return(pca_plot_bi)
}

# pca_test_biplot <- plot_pca_bi(malt_species.pca, "PC3", "PC2", "Village", PC2)
# pca_test_biplot


```


List the outliers from the plot in `MID_CMB_Rad_Kil_Iber_decontam.Rmd`. 
```{r outliers}
outliers <- c("EXB059.A2501","LIB058.A0103","LIB058.A0106","CSS","CSD", "CS14","CS27","CS48") # the lat 3 Radcliffe samples have very few reads
poor_samples_nt <- fread("./05-results.backup/cuperdec_nt_poor_samples.tsv") %>%
  pull(Sample)

poor_samples_sub75bp <- fread("./05-results.backup/cuperdec_nt_poor_samples.tsv") %>%
  pull(Sample)

outliersF <- str_c(outliers, collapse = "|")

```


#Sample Clustering with PCA

Step-by-step removal of taxa based on decontam and a filter threshold, followed
by PCA to see how contaminants/low abundance species affect the relationships of
samples to eachother between groups.
```{r pca_all_species}
# decontaminated only
nt_species_decontam_unfilt <- nt_species_decontam %>%
  select(-one_of(outliers)) %>%
  select(-one_of(poor_samples_nt)) %>%
  # adorn_totals(where = "col", name = "Sum_species.decontam") %>%
  # mutate(Percent = Sum_species.decontam/sum(Sum_species.decontam)*100) %>%
  # select(-one_of(c("Sum_species.decontam","Percent"))) %>%
  gather("Library_ID","Counts",2:ncol(.)) %>%
  mutate(Counts = Counts +1) %>%
  spread(Species,Counts) %>%
  column_to_rownames("Library_ID")

# prepare to run a PCA
# check the number of components to retain by tuning the PCA
tune.pca(nt_species_decontam_unfilt, logratio = 'CLR')

# perform and plot a PCA to see how the data cluster
plot_pca(nt_species_decontam_unfilt, "PC1", "PC2", "Site", "Time_period", 3)
plot_pca(nt_species_decontam_unfilt, "PC3", "PC2", "Site", "Time_period", 3)

# decontaminated and filtered
nt_species_decontam_filtered <- nt_species_decontam %>%
  # adorn_totals(where = "col", name = "Sum_species.decontam_filtered") %>%
  select(-one_of(outliers)) %>%
  select(-one_of(poor_samples_nt)) %>%
  # mutate(Percent = Sum_species.decontam_filtered/sum(Sum_species.decontam_filtered)*100) %>%
  # filter(Percent > 0.005) %>%
  # select(-one_of(c("Sum_species.decontam_filtered","Percent"))) %>%
  gather("Library_ID","Counts",2:ncol(.)) %>%
  mutate(Counts = Counts +1) %>%
  spread(Species,Counts) %>%
  column_to_rownames("Library_ID")

# prepare to run a PCA
# check the number of components to retain by tuning the PCA
tune.pca(nt_species_decontam_filtered, logratio = 'CLR')

# perform and plot a PCA to see how the data cluster
plot_pca(nt_species_decontam_filtered, "PC1", "PC2", "Site", "Time_period", 3)

plot_pca(nt_species_decontam_filtered, "PC3", "PC2", "Site", "Time_period", 3)


# ggsave("./06-publication/supplemental_figures/SXX28/SXX28_all_samples_sp_pca12.pdf", plot = all_samples_sp_pca12, device = "pdf",
#        scale = 1, width = 6, height = 7, units = c("in"), dpi = 300)


```


# Species-level analyses
What is the relationship of the groups in a PCA without the blanks? Let's try
the PCA using only the species that are present in at least 10% of at least one
sample group (MID, CMB, Radcliffe, Kilteasheen, Iberia)

## Middenbeemster samples only
What is the relationship of Cameroon plaque samples to eachother without any
other sample types? Do they cluster by any of the metadata categories?
Filter the species table by % abundance
```{r pca_species_mid_pc12}
nt_species_decontam_filtered_mid <- nt_species_decontam %>%
  select(-one_of(outliers)) %>%
  select(-one_of(poor_samples_nt)) %>%
  select(matches("Species|MID|CMB")) %>%
  # adorn_totals(where = "col", name = "Sum_species") %>%
  # mutate(Percent = Sum_species/sum(Sum_species)*100) %>%
  # filter(Percent > 0.001) %>%
  select(-one_of(c("Sum_species","Percent"))) %>%
  gather("Library_ID","Counts",2:ncol(.)) %>% 
  mutate(Counts = Counts +1) %>%
  spread(Species,Counts) %>%
  column_to_rownames("Library_ID")

# prepare to run a PCA
# check the number of components to retain by tuning the PCA
mixOmics::tune.pca(nt_species_decontam_filtered_mid, logratio = 'CLR')


malt_otu.pca <- mixOmics::pca(nt_species_decontam_filtered_mid, ncomp = 2, logratio = 'CLR')
malt_pca_values <- malt_otu.pca$variates$X %>%
  as.data.frame() %>%
  rownames_to_column("Library_ID") %>%
  inner_join(., full_metadata, by = "Library_ID")

ggplot(malt_pca_values, aes(PC1, PC2, color = Region, shape = Region)) +
  geom_text(aes(label = Library_ID), size = 3) +
  theme_minimal()

# perform and plot a PCA to see how the data cluster
# by pipe notch presence
mid_pipe_pc12 <- plot_pca_mid(nt_species_decontam_filtered_mid, "PC1", "PC2", "Pipenotch", "Pipenotch", 3, "Pipe notch")
mid_pipe_pc12
# ggsave("~/Desktop/MID_pca_mid_pipe_pc12.pdf", plot = mid_pipe_pc12, device = "pdf",
#        scale = 1, width = 9, height = 5, units = c("in"), dpi = 300)

# by sex
plot_pca_mid(nt_species_decontam_filtered_mid, "PC1", "PC2", "Sex", "Pipenotch", 3, "Sex")

# by age
plot_pca_mid(nt_species_decontam_filtered_mid, "PC1", "PC2", "Age", "Pipenotch", 3, "Age")

# by presence of periodontal disease
plot_pca_mid(nt_species_decontam_filtered_mid, "PC1", "PC2", "Periodontal_disease_present", "Pipenotch", 3, "Periodontal disease present")

# by presence of caries
plot_pca_mid(nt_species_decontam_filtered_mid, "PC1", "PC2", "Caries_present", "Pipenotch", 3, "Caries present")

# by ante-mortem tooth loss
plot_pca_mid(nt_species_decontam_filtered_mid, "PC1", "PC2", "AMTL", "Pipenotch", 3, "AMTL")

# by supragingival calculus score
plot_pca_cont(nt_species_decontam_filtered_mid, "PC1", "PC2", "calc_sup_SI_score", "Pipenotch", 3, "Sup calc SI score")

# by subgingival calculus score
plot_pca_cont(nt_species_decontam_filtered_mid, "PC1", "PC2", "Calc_sub_SI_score", "Pipenotch", 3, "Sub calc SI score")

# by presence of perioapical lesions
plot_pca_mid(nt_species_decontam_filtered_mid, "PC1", "PC2", "Periapical_lesions", "Pipenotch", 3, "Periapical lesions")

# by max periodontal disease score
plot_pca_cont(nt_species_decontam_filtered_mid, "PC1", "PC2", "Max_Perio_Score", "Pipenotch", 3, "Max perio score")

# by periodontal affect score
plot_pca_cont(nt_species_decontam_filtered_mid, "PC1", "PC2", "Peridontal_affect_score", "Pipenotch", 3, "Perio affect score")

# by % antemortem tooth loss
plot_pca_cont(nt_species_decontam_filtered_mid, "PC1", "PC2", "%AMTL", "Pipenotch", 3, " % AMTL")

# by % positions w/periapical lesions
plot_pca_cont(nt_species_decontam_filtered_mid, "PC1", "PC2", "%_positions_perioapical", "Pipenotch", 3, "% positions periapical lesions")

# by % teeth with caries
plot_pca_cont(nt_species_decontam_filtered_mid, "PC1", "PC2", "%teeth_with_caries", "Pipenotch", 3, "% teeth w/ caries")

# by % teeth with calculus
plot_pca_cont(nt_species_decontam_filtered_mid, "PC1", "PC2", "%teeth_with_calc", "Pipenotch", 3, "% teeth w/ calc")

# by % positions with periodontal disease
plot_pca_cont(nt_species_decontam_filtered_mid, "PC1", "PC2", "%_positions_with_Periodontal", "Pipenotch", 3, "% positions w/ PD")

# by minimum number of pipenotches 
mid_minpipe_pc12 <- plot_pca_cont(nt_species_decontam_filtered_mid, "PC1", "PC2", "Min_no_Pipe_notches", "Pipenotch", 3, "Min # pipe notches")
mid_minpipe_pc12
# ggsave("~/Desktop/MID_pca_mid_minpipe_pc12.pdf", plot = mid_minpipe_pc12, device = "pdf",
#        scale = 1, width = 9, height = 5, units = c("in"), dpi = 300)

# by sequencing depth 
mid_seqdepth_pc12 <- plot_pca_cont(nt_species_decontam_filtered_mid, "PC1", "PC2", "No_reads_total", "Pipenotch", 3, "Total reads into MALT")
mid_seqdepth_pc12
# ggsave("~/Desktop/MID_pca_mid_seqdepth_pc12.pdf", plot = mid_seqdepth_pc12, device = "pdf",
#        scale = 1, width = 9, height = 5, units = c("in"), dpi = 300)

# by sampled tooth site(s)
mid_tooth_pc12 <- plot_pca_mid(nt_species_decontam_filtered_mid, "PC1", "PC2", "Tooth_location", "Pipenotch", 3, "Tooth location")
mid_tooth_pc12
# ggsave("~/Desktop/MID_pca_mid_tooth_pc12.pdf", plot = mid_tooth_pc12, device = "pdf",
#        scale = 1, width = 9, height = 5, units = c("in"), dpi = 300)

# by sampled tooth site(s)
plot_pca_mid(nt_species_decontam_filtered_mid, "PC1", "PC2", "alt_tooth_location", "Pipenotch", 3, "Tooth location")

# by mg of extracted calculus
plot_pca_cont(nt_species_decontam_filtered_mid, "PC1", "PC2", "Calculus_extracted_mg", "Pipenotch", 3, "Weight of extracted calculus")

# by [DNA] after extraction per mg of calculus extracted
plot_pca_cont(nt_species_decontam_filtered_mid, "PC1", "PC2", "extract_conc_ng/mg", "Pipenotch", 3, "DNA extract concentration (ng/mg)")

# by [DNA] after extraction per uL of eluate
mid_extract_pc12 <- plot_pca_cont(nt_species_decontam_filtered_mid, "PC1", "PC2", "extract_conc_ng/uL", "Pipenotch", 3, "DNA extract concentration (ng/uL)")
mid_extract_pc12
# ggsave("~/Desktop/MID_pca_mid_extract_pc12.pdf", plot = mid_extract_pc12, device = "pdf",
#        scale = 1, width = 9, height = 5, units = c("in"), dpi = 300)

# by [DNA] after storage, before building libraries
plot_pca_cont(nt_species_decontam_filtered_mid, "PC1", "PC2", "extract_conc_after_storage_ng/mg", "Pipenotch", 3, "DNA extract concentration after storage (ng/uL)")

# by # of molecules in the library after indexing amplification
mid_library_pc12 <- plot_pca_cont(nt_species_decontam_filtered_mid, "PC1", "PC2", "DNA_molecules_per_library_x10^6_log", "Pipenotch", 3, "log10(DNA_molecules_per_library_x10^6)")
mid_library_pc12
# ggsave("~/Desktop/MID_pca_mid_library_pc12.pdf", plot = mid_library_pc12, device = "pdf",
#        scale = 1, width = 9, height = 5, units = c("in"), dpi = 300)

# by soil type
plot_pca_mid(nt_species_decontam_filtered_mid, "PC1", "PC2", "Soil", "Pipenotch", 3, "Soil type")

# by average GC content
plot_pca_cont(nt_species_decontam_filtered_mid, "PC1", "PC2", "gc_avg", "Time_period", 3, "Avg. GC content (%)")

# by average read length
plot_pca_cont(nt_species_decontam_filtered_mid, "PC1", "PC2", "seq_len_avg", "Time_period", 3, "Avg. read length (bp)")

# by average mapped GC content
plot_pca_cont(nt_species_decontam_filtered_mid, "PC1", "PC2", "avg_gc_malt_map_nt", "Time_period", 3, "Avg. mapped GC content (%)")

# by average mapped read length
plot_pca_cont(nt_species_decontam_filtered_mid, "PC1", "PC2", "avg_rl_malt_map_nt", "Time_period", 3, "Avg. mapped read length (bp)")

# with names
# pca.list <- mixOmics::pca(nt_species_decontam_filtered_mid, ncomp = 3, logratio = 'CLR')
# plot_pca_names(pca.list, "PC1", "PC2", "Pipenotch")

# ggsave("05-results.backup/pca_nt_species_pc1pc2.pdf", plot = pca_nt_species_pc1pc2, device = "pdf",
#        scale = 1, width = 7, height = 5, units = c("in"),
#        dpi = 300)
```

```{r pca_species_mid_pc32}
# by sex
plot_pca_mid(nt_species_decontam_filtered_mid, "PC3", "PC2", "Sex", "Pipenotch", 3, "Sex")

# by age
plot_pca_mid(nt_species_decontam_filtered_mid, "PC3", "PC2", "Age", "Pipenotch", 3, "Age")

# by pipe notch presence
plot_pca_mid(nt_species_decontam_filtered_mid, "PC3", "PC2", "Pipenotch", "Pipenotch", 3, "Pipe notch")

# by presence of periodontal disease
plot_pca_mid(nt_species_decontam_filtered_mid, "PC3", "PC2", "Periodontal_disease_present", "Pipenotch", 3, "Periodontal disease present")

# by presence of caries
plot_pca_mid(nt_species_decontam_filtered_mid, "PC3", "PC2", "Caries_present", "Pipenotch", 3, "Caries present")

# by ante-mortem tooth loss
plot_pca_mid(nt_species_decontam_filtered_mid, "PC3", "PC2", "AMTL", "Pipenotch", 3, "AMTL")

# by supragingival calculus score
plot_pca_cont(nt_species_decontam_filtered_mid, "PC3", "PC2", "calc_sup_SI_score", "Pipenotch", 3, "Sup calc SI score")

# by subgingival calculus score
plot_pca_cont(nt_species_decontam_filtered_mid, "PC3", "PC2", "Calc_sub_SI_score", "Pipenotch", 3, "Sub calc SI score")

# by presence of perioapical lesions
plot_pca_mid(nt_species_decontam_filtered_mid, "PC3", "PC2", "Periapical_lesions", "Pipenotch", 3, "Periapical lesions")

# by max periodontal disease score
plot_pca_cont(nt_species_decontam_filtered_mid, "PC3", "PC2", "Max_Perio_Score", "Pipenotch", 3, "Max perio score")

# by periodontal affect score
plot_pca_cont(nt_species_decontam_filtered_mid, "PC3", "PC2", "Peridontal_affect_score", "Pipenotch", 3, "Perio affect score")

# by % antemortem tooth loss
plot_pca_cont(nt_species_decontam_filtered_mid, "PC3", "PC2", "%AMTL", "Pipenotch", 3, "% AMTL")

# by % positions w/periapical lesions
plot_pca_cont(nt_species_decontam_filtered_mid, "PC3", "PC2", "%_positions_perioapical", "Pipenotch", 3, "% positions periapical lesions")

# by % teeth with caries
plot_pca_cont(nt_species_decontam_filtered_mid, "PC3", "PC2", "%teeth_with_caries", "Pipenotch", 3, "% teeth w/ caries")

# by % teeth with calculus
plot_pca_cont(nt_species_decontam_filtered_mid, "PC3", "PC2", "%teeth_with_calc", "Pipenotch", 3, "% teeth w/ calc")

# by % positions with periodontal disease
plot_pca_cont(nt_species_decontam_filtered_mid, "PC3", "PC2", "%_positions_with_Periodontal", "Pipenotch", 3, "% positions w/ PD")

# by minimum number of pipenotches 
plot_pca_cont(nt_species_decontam_filtered_mid, "PC3", "PC2", "Min_no_Pipe_notches", "Pipenotch", 3, "Min # pipe notches")

# by sequencing depth 
plot_pca_cont(nt_species_decontam_filtered_mid, "PC3", "PC2", "No_reads_total", "Pipenotch", 3, "Total reads into MALT")

# by sampled tooth site(s)
plot_pca_mid(nt_species_decontam_filtered_mid, "PC3", "PC2", "Tooth_location", "Pipenotch", 3, "Tooth location")

# by sampled tooth site(s)
plot_pca_mid(nt_species_decontam_filtered_mid, "PC3", "PC2", "alt_tooth_location", "Pipenotch", 3, "Tooth location")

# by mg of extracted calculus
plot_pca_cont(nt_species_decontam_filtered_mid, "PC3", "PC2", "Calculus_extracted_mg", "Pipenotch", 3, "Weight of extracted calculus")

# by [DNA] after extraction
plot_pca_cont(nt_species_decontam_filtered_mid, "PC3", "PC2", "extract_conc_ng/mg", "Pipenotch", 3, "DNA extract concentration (ng/mg)")

# by [DNA] after extraction per uL of eluate
plot_pca_cont(nt_species_decontam_filtered_mid, "PC3", "PC2", "extract_conc_ng/uL", "Pipenotch", 3, "DNA extract concentration (ng/uL)")

# by [DNA] after storage, before building libraries
plot_pca_cont(nt_species_decontam_filtered_mid, "PC3", "PC2", "extract_conc_after_storage_ng/mg", "Pipenotch", 3, "DNA extract concentration after storage (ng/uL)")

# by # of molecules in the library after indexing amplification
plot_pca_cont(nt_species_decontam_filtered_mid, "PC3", "PC2", "DNA_molecules_per_library_x10^6_log", "Pipenotch", 3, "log10(DNA_molecules_per_library_x10^6)")

# by soil type
plot_pca_mid(nt_species_decontam_filtered_mid, "PC3", "PC2", "Soil", "Pipenotch", 3, "Soil type")

# by average GC content
plot_pca_cont(nt_species_decontam_filtered_mid, "PC3", "PC2", "gc_avg", "Time_period", 33, "Avg. GC content (%)")

# by average read length
plot_pca_cont(nt_species_decontam_filtered_mid, "PC3", "PC2", "seq_len_avg", "Time_period", 33, "Avg. read length (bp)")

# by average mapped GC content
plot_pca_cont(nt_species_decontam_filtered_mid, "PC3", "PC2", "avg_gc_malt_map_nt", "Time_period", 3, "Avg. GC content (%)")

# by average mapped read length
plot_pca_cont(nt_species_decontam_filtered_mid, "PC3", "PC2", "avg_rl_malt_map_nt", "Time_period", 3, "Avg. read length (bp)")


```



```{r biplots_cmc_species}

nt_species_decontam_filtered_mid <- nt_species_decontam %>%
  select(-one_of(outliers)) %>%
  select(-one_of(poor_samples_nt)) %>%
  select(matches("Species|MID|CMB")) %>%
  # adorn_totals(where = "col", name = "Sum_species") %>%
  # mutate(Percent = Sum_species/sum(Sum_species)*100) %>%
  # filter(Percent > 0.001) %>%
  # select(-one_of(c("Sum_species","Percent"))) %>%
  gather("Library_ID","Counts",2:ncol(.)) %>% 
  mutate(Counts = Counts +1) %>%
  spread(Species,Counts) %>%
  column_to_rownames("Library_ID")

nt_species_decontam_filtered_mid.pca <- mixOmics::pca(nt_species_decontam_filtered_mid, ncomp = 3, logratio = 'CLR')
plot(nt_species_decontam_filtered_mid.pca)

# Define pipenotch colors for plotting
Pipenotch_colors = c("#311068","#C83E73")
Pipenotch_shapes = c(19,17)

nt_species_decontam_filtered_nocontrols_biplot_121 <- plot_pca_bi(nt_species_decontam_filtered_mid.pca, "PC1", "PC2", "Pipenotch", PC1)
nt_species_decontam_filtered_nocontrols_biplot_121

nt_species_decontam_filtered_nocontrols_biplot_122 <- plot_pca_bi(nt_species_decontam_filtered_mid.pca, "PC1", "PC2", "Pipenotch", PC2)
nt_species_decontam_filtered_nocontrols_biplot_122

pandoc.table(nt_species_decontam_filtered_nocontrols_biplot_121$plot_env$pws_10 %>% arrange(desc(PC1)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC1 positive loadings")

pandoc.table(nt_species_decontam_filtered_nocontrols_biplot_121$plot_env$neg_10 %>% arrange(PC1),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC1 negative loadings")


pandoc.table(nt_species_decontam_filtered_nocontrols_biplot_122$plot_env$pws_10 %>% arrange(desc(PC2)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC2 positive loadings")

pandoc.table(nt_species_decontam_filtered_nocontrols_biplot_122$plot_env$neg_10 %>% arrange(PC2),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC2 negative loadings")


nt_species_decontam_filtered_nocontrols_biplot_322 <- plot_pca_bi(nt_species_decontam_filtered_mid.pca, "PC3", "PC2", "Pipenotch", PC2)
nt_species_decontam_filtered_nocontrols_biplot_322

nt_species_decontam_filtered_nocontrols_biplot_323 <- plot_pca_bi(nt_species_decontam_filtered_mid.pca, "PC3", "PC2", "Pipenotch", PC3)
nt_species_decontam_filtered_nocontrols_biplot_323

pandoc.table(nt_species_decontam_filtered_nocontrols_biplot_323$plot_env$pws_10 %>% arrange(desc(PC3)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC3 positive loadings")

pandoc.table(nt_species_decontam_filtered_nocontrols_biplot_323$plot_env$neg_10 %>% arrange(PC3),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC3 negative loadings")

```





