---
title: "MID/CMB, Radcliffe, Kilteasheen, Iberian medieval calculus metadata and colors"
author: "Irina Velsko"
date: "`r format(Sys.time(), '%b %d, %Y')`"
output:
    github_document:
        fig_width: 8
        fig_height: 6
        toc: true
---

```{r load_libraries, echo = F, message = F}
library(knitr)
library(data.table)
library(tidyverse)

```

```{r setup}
knitr::opts_knit$set(root.dir = normalizePath(".."))
```

Pull some information from Pandora
```{r db_connection, eval = F}

if(!require('remotes')) install.packages('remotes')
remotes::install_github("sidora-tools/sidora.core")
library(sidora.core)

# establish database connection
con <- get_pandora_connection(cred_file = "~/.credentials")

```

```{r get_data, eval = F}
# use shorthand to get a list of data.frames
df_list <- get_df_list(c(
  make_complete_table_list(c("TAB_Site", "TAB_Library"))
), con = con)

jt <- join_pandora_tables(df_list)

jt <- jt %>% convert_all_ids_to_values()


library_info <- get_df("TAB_Library", con)
library_info <- library_info %>% convert_all_ids_to_values()

```

```{r, eval = F}

concentrations <- library_info %>%
  select(library.Full_Library_Id, library.Batch, `library.Quantification_post-Indexing_total`) %>%
  filter(str_detect(library.Batch, "MID")) %>%
  bind_rows(., library_info %>%
  select(library.Full_Library_Id, library.Batch, `library.Quantification_post-Indexing_total`) %>%
  filter(str_detect(library.Batch, "CMB"))) %>%
  bind_rows(., library_info %>%
  select(library.Full_Library_Id, `library.Quantification_post-Indexing_total`) %>%
  filter(str_detect(library.Full_Library_Id, "ELR|IVE")))


```


This creates an RData object that has the full metadata table, where the columns
we use for plotting are set as factors, and lists the colors, shapes, and sizes of
points for plotting PCAs and such for each of the metadata categories we're
interested in.
```{r metadata}
# load the metadata file
full_metadata <- fread("./00-documentation.backup/kilteasheen_metadata.tsv") %>%
  full_join(., fread("./00-documentation.backup/radcliffe_pfuT_metadata.tsv")) %>%
  full_join(., fread("./00-documentation.backup/iberian_medieval_metadata.tsv")) %>%
  full_join(., fread("./00-documentation.backup/MID_analysis_metadata.tsv")) %>%
  full_join(., fread("./00-documentation.backup/mod_metadata.tsv") %>%
                       mutate(Sampled_teeth_FDI = as.character(Sampled_teeth_FDI)))

full_metadata <- full_metadata %>%
  as_tibble() %>%
  # remove the MID samples that have no metadata
  # filter(!str_detect(Library_ID, "MID025.A0101|MID033.A0101|MID056.A0101|MID065.A0101|MID068.A0101|MID076.A0101|MID078.A0101")) %>%
  mutate(Site = fct_relevel(Site, "Middenbeemster", "Convento de los Mercedarios de Burtzena", "Radcliffe", "Kilteasheen",
                                   "El Raval", "Iglesia de la Virgen de la Estrella","Jaen","Valencia","Extraction_blank","Library_blank"),
         Site_code = fct_relevel(Site_code, "MID", "CMB", "RAD", "KIL",
                                   "ELR", "IVE","JAE","VLC","EXB","LIB","DEN","SOL"),
         Time_period = fct_relevel(Time_period, "Industrial","Medieval","Modern","Extraction_blank","Library_blank"),
         Region = fct_relevel(Region, "Netherlands","Spain","England", "Ireland","Iberia"),
         Periodontal_disease_present = fct_relevel(Periodontal_disease_present, "Yes","No","Unobservable","NR"),
         Caries_present = fct_relevel(Caries_present, "Yes","No","Unobservable","NR"),
         Sample_or_Control = fct_relevel(Sample_or_Control, "Sample","Control"),
         Sex = fct_relevel(Sex, "Female","Male","Indeterminate","Unknown","NR"),
         Tooth_location = fct_relevel(Tooth_location, "Anterior","Anterior/Posterior","Posterior","Unknown"),
         alt_tooth_location = fct_relevel(alt_tooth_location, "Anterior","Anterior/Posterior","Posterior","Unknown"),
         Pipenotch = fct_relevel(Pipenotch, "Present","Absent"),
         Soil = fct_relevel(Soil, "clay","clayish sand","sand","NR"))

```

Load the average GC content and read length information from multiqc from the
eager runs
```{r rl_gc_info}

mid_multiqc <- fread("~/archgen/microbiome_calculus/smoking_calculus/03-preprocessing/eager2/eager2_run/multiqc/multiqc_data/multiqc_fastqc_1.txt") %>%
  mutate(Library_ID = str_trunc(Sample, width = 12, side = "right", ellipsis = "")) %>%
  select(Library_ID, avg_sequence_length, `%GC`) %>%
  group_by(Library_ID) %>%
  summarize(seq_len_avg = mean(avg_sequence_length),
            gc_avg = mean(`%GC`))

rad_multiqc <- fread("~/archgen/microbiome_calculus/smoking_calculus/03-preprocessing/radcliffe_pfuT/eager2/multiqc/radcliffe_pfuT_tsv_multiqc_report_data/multiqc_general_stats.txt") %>%
  filter(str_detect(Sample, "unm")) %>%
  select(matches("Sample|post_adapterremoval-percent_gc|post_adapterremoval-avg_sequence_length")) %>%
  rename(seq_len_avg = `FastQC (post-AdapterRemoval)_mqc-generalstats-fastqc_post_adapterremoval-avg_sequence_length`,
         gc_avg = `FastQC (post-AdapterRemoval)_mqc-generalstats-fastqc_post_adapterremoval-percent_gc`) %>%
  mutate(Sample = str_replace_all(Sample, "_unmapped.clip","")) %>%
  rename(Library_ID = Sample)

kil_codes <- fread("./00-documentation.backup/kil_codes.tsv")
kil_multiqc <- fread("~/archgen/microbiome_calculus/smoking_calculus/03-preprocessing/kilteasheen/eager2/multiqc/multiqc_data/multiqc_general_stats.txt") %>%
  filter(str_detect(Sample, "SRR")) %>%
  select(matches("Sample|post_adapterremoval-percent_gc|post_adapterremoval-avg_sequence_length")) %>%
  full_join(., kil_codes) %>%
  rename(avg_sequence_length = `FastQC (post-AdapterRemoval)_mqc-generalstats-fastqc_post_adapterremoval-avg_sequence_length`,
         `%GC` = `FastQC (post-AdapterRemoval)_mqc-generalstats-fastqc_post_adapterremoval-percent_gc`) %>%
  group_by(Library_ID) %>%
  summarize(seq_len_avg = mean(avg_sequence_length),
            gc_avg = mean(`%GC`))

ibe_multiqc <- fread("~/archgen/projects1/microbiome_calculus/iberian/03-preprocessing/screening/eager2/multiqc/multiqc_data/multiqc_fastqc_1.txt") %>%
  filter(str_detect(Sample, "ELR|IVE|EXB|LIB|JAE|VLC")) %>%
  mutate(Library_ID = str_trunc(Sample, width = 12, side = "right", ellipsis = "")) %>%
  select(Library_ID, avg_sequence_length, `%GC`) %>%
  group_by(Library_ID) %>%
  summarize(seq_len_avg = mean(avg_sequence_length),
            gc_avg = mean(`%GC`))

full_rl_gc <- mid_multiqc %>%
  bind_rows(., rad_multiqc) %>%
  bind_rows(., kil_multiqc) %>%
  bind_rows(., ibe_multiqc)

```

Now join the GC and read length data to the full metadata table
```{r}
full_metadata <- full_metadata %>%
  inner_join(., full_rl_gc)

```

Make an additional table for the sub10M and sub75bp multiqc avg read length and avg GC content
```{r}
# sub10M
sub10M_multiqc <- fread("~/archgen/microbiome_calculus/smoking_calculus/03-preprocessing/depth_check/sub10M/FastQC/multiqc_data/multiqc_general_stats.txt") %>%
  mutate(Library_ID = str_trunc(Sample, width = 12, side = "right", ellipsis = "")) %>%
  select(Library_ID, `FastQC_mqc-generalstats-fastqc-avg_sequence_length`, `FastQC_mqc-generalstats-fastqc-percent_gc`) %>%
  group_by(Library_ID) %>%
  rename(seq_len_avg = `FastQC_mqc-generalstats-fastqc-avg_sequence_length`,
            gc_avg = `FastQC_mqc-generalstats-fastqc-percent_gc`) %>%
  mutate(Library_ID = str_replace_all(Library_ID, ".unmappe",""),
         Library_ID = str_replace_all(Library_ID, ".unm",""))

# sub75bp
sub75bp_multiqc <- fread("~/archgen/microbiome_calculus/smoking_calculus/03-preprocessing/read_len_check/sub75bp/FastQC/multiqc_data/multiqc_general_stats.txt") %>%
  mutate(Library_ID = str_trunc(Sample, width = 12, side = "right", ellipsis = "")) %>%
  select(Library_ID, `FastQC_mqc-generalstats-fastqc-avg_sequence_length`, `FastQC_mqc-generalstats-fastqc-percent_gc`) %>%
  group_by(Library_ID) %>%
  rename(seq_len_avg_75bp = `FastQC_mqc-generalstats-fastqc-avg_sequence_length`,
            gc_avg_75bp = `FastQC_mqc-generalstats-fastqc-percent_gc`) %>%
  mutate(Library_ID = str_replace_all(Library_ID, ".unmappe",""),
         Library_ID = str_replace_all(Library_ID, ".unm",""))


# now join them together, adding the full data for all samples that were alredy below 10M reads
# first get the information from the below-10M reads samples

all_sub_rl_gc <- full_rl_gc %>%
  anti_join(., sub10M_multiqc %>%
              select(Library_ID)) %>%
  filter(!str_detect(Library_ID, "EXB|LIB|CSS|CSD|CSL|CSN")) %>%
  bind_rows(., sub10M_multiqc) %>%
  arrange(Library_ID) %>%
  rename(seq_len_avg_10M = seq_len_avg,
         gc_avg_10M = gc_avg) %>%
  full_join(., sub75bp_multiqc)
           
          
```

Now join the subsampled GC and read length data to the full metadata table
```{r}
full_metadata <- full_metadata %>%
  full_join(., all_sub_rl_gc) 
# note that MID052.A0101 has no other metadata, I don't know why

```


```{r colors}
# lacroix_palette("PeachPear", n = 8, type = "continuous")
# "#FF3200" "#EF8158" "#E9BD8E" "#CBDDA7" "#38BCAD" "#0B91B5" "#065FA3" "#172869"

Site_code_colors <- c(MID = "#311068", CMB = "#7B2382FF", RAD = "#C83E73", KIL = "#E95562", ELR = "#FEA873",
                      IVE = "#FED395", JAE = "#41b6c4", VLC = "#a0dae1", EXB = "#252525", LIB = "#737373", DEN = "#252525", SOL = "#252525")
Site_code_shapes <- c(MID = 16, CMB = 15, RAD = 17, KIL = 18, ELR = 25, IVE = 14, JAE = 13, VLC = 13, EXB = 4, LIB = 3, Dentin = 3, Soil = 3)
Site_code_size <- c(MID = 4.5, CMB = 4, RAD = 4, KIL = 6, ELR = 4, IVE = 4, JAE = 4, VLC = 4, EXB = 4, LIB = 4, Dentin = 3, Soil = 3)

Time_period_colors <- c(Industrial = "#311068", Medieval = "#C83E73", Modern = "#41b6c4", Extraction_blank = "#252525", Library_blank = "#737373", Soil = "black")
Time_period_shapes <- c(Industrial = 19, Medieval = 17, Modern = 15, Extraction_blank = 4, Library_blank = 3)
Time_period_size <- c(Industrial = 4, Medieval = 4, Modern = 4, Extraction_blank = 4, Library_blank = 4)

Region_colors <- c(Netherlands = "#311068", Spain = "#5A167E", England = "#C83E73", Iberia = "#FEA873", Ireland = "#E95562",
                Extraction_blank = "#252525", Library_blank = "#737373")
Region_shapes <- c(Netherlands = 9, Spain = 12, England = 10, Iberia = 7, Ireland = 13, Extraction_blank = 4, Library_blank = 3)
Region_size <- c(Netherlands = 4, Spain = 4, England = 4, Iberia = 4, Ireland = 4, Extraction_blank = 4, Library_blank = 4)

Periodontal_disease_present_colors <- c(Yes = "#311068", No = "#C83E73", Unobservable = "#737373", NR = "gray80")
Periodontal_disease_present_shapes <- c(Yes = 19, No = 14, Unobservable = 10, NR = 2)
Periodontal_disease_present_size <- c(Yes = 4, No = 4, Unobservable = 4, NR = 4)

Sample_or_Control_colors <- c(Sample = "#311068", Control = "#737373")
Sample_or_Control_shapes <- c(Sample = 19, Control = 8)
Sample_or_Control_size <- c(Sample = 4, Control = 4)

Caries_present_colors <- c(Yes = "#311068", No = "#C83E73", Unobservable = "#737373", NR = "gray80")
Caries_present_shapes <- c(Yes = 19, No = 14, Unobservable = 10, NR = 2)
Caries_present_size <- c(Yes = 4, No = 4, Unobservable = 4, NR = 4)

Sex_colors <- c(Female = "#311068", Male = "#C83E73", Indeterminate = "#737373",Unknown = "grey50", NR = "gray80")
Sex_shapes <- c(Female = 19, Male = 17, Indeterminate = 8, NR = 2)
Sex_size <- c(Female = 4, Male = 4, Indeterminate = 4, NR = 4)

Tooth_location_colors <- c(Anterior = "#311068", `Anterior/Posterior` = "#C83E73", Posterior = "#FEA873", Unknown = "#737373")
Tooth_location_shapes <- c(Anterior = 19, `Anterior/Posterior` = 18, Posterior = 17, Unknown = 8)
Tooth_location_size <- c(Anterior = 4, `Anterior/Posterior` = 4.5, Posterior = 4, Unknown = 4)

alt_tooth_location_colors <- c(Anterior = "#311068", `Anterior/Posterior` = "#C83E73", Posterior = "#FEA873", Unknown = "#737373")
alt_tooth_location_shapes <- c(Anterior = 19, `Anterior/Posterior` = 18, Posterior = 17, Unknown = 8)
alt_tooth_location_size <- c(Anterior = 4, `Anterior/Posterior` = 4.5, Posterior = 4, Unknown = 4)

```


```{r save_data, eval = F}
save(full_metadata, 
     Site_code_colors, Site_code_shapes, Site_code_size,
     Time_period_colors, Time_period_shapes, Time_period_size,
     Region_colors, Region_shapes, Region_size,
     Periodontal_disease_present_colors, Periodontal_disease_present_shapes, Periodontal_disease_present_size,
     Caries_present_colors, Caries_present_shapes, Caries_present_size, 
     Sample_or_Control_colors, Sample_or_Control_shapes, Sample_or_Control_size, 
     Sex_colors, Sex_shapes, Sex_size, 
     Tooth_location_colors,Tooth_location_shapes,Tooth_location_size,
     alt_tooth_location_colors,alt_tooth_location_shapes,alt_tooth_location_size,
     file = "./05-results.backup/Metadata_tables.RData")

# fwrite(full_metadata, "00-documentation.backup/full_combined_metadata.tsv", sep = "\t", quote = F, na = "NA")

```


