---
title: "MID/CMB, Radcliffe, Kilteasheen, Iberian medieval calculus MALT beta diversity"
author: "Irina Velsko"
date: "`r format(Sys.time(), '%b %d, %Y')`"
output:
    github_document:
        fig_width: 8
        fig_height: 6
        toc: true
---


```{r load_libraries, echo = F, message = F}
library(knitr)
library(data.table)
library(psych)
library(ape)
library(mixOmics)
library(compositions)
library(vegan)
library(janitor)
library(pander)
library(tidyverse)
library(gplots)
library(ggrepel)
library(viridis)
library(cowplot)
opts_chunk$set(echo=F, warning=F, message=F, dpi=150)
```

Set the working directory of the notebook to the parent directory of the 
notebook itself.  

```{r setup}
knitr::opts_knit$set(root.dir = normalizePath(".."))
```

# MALT read stats

```{r load_data}

# load the taxonomy tables that have been decontaminated and the column headers are cleaned up
load("./05-results.backup/Taxonomy_tables.RData")

# load the metadata file
load("./05-results.backup/Metadata_tables.RData")

```


```{r pca_name_fxn}
# this function plots the PCA with the sample names instead of points
plot_pca_names <- function(df, pc1, pc2, metadata_group) {
    metadata_group_colors <- get(paste(metadata_group, "_colors", sep = ""))

    exp_var <- paste0(round(df$explained_variance * 100, 2), "%")
    df_X <- df$variates$X %>%
              as.data.frame() %>%
              rownames_to_column("Library_ID") %>%
              inner_join(full_metadata %>%
                           select(Library_ID, Site, Region, Time_period, Sample_or_Control, Periodontal_disease_present, Caries_present, Sex, Age, Tooth_location), by = "Library_ID")
    
    metadata_group = df_X[[metadata_group]]
    
                       if (pc1 == 'PC1') {
                           pc1 <- df_X$PC1
                           exp_var_pc1 <- exp_var[1]
                           xaxis <- c("PC1")
                       }  else if (pc1 == 'PC2') {
                           pc1 <- df_X$PC2
                           exp_var_pc1 <- exp_var[2]
                           xaxis <- c("PC2")
                       } else if (pc1 == 'PC3') {
                           pc1 <- df_X$PC3
                           exp_var_pc1 <- exp_var[3]
                           xaxis <- c("PC3")
                       }

                       if (pc2 == 'PC1') {
                           pc2 <- df_X$PC1
                           exp_var_pc2 <- exp_var[1]
                           yaxis <- c("PC1")
                       }  else if (pc2 == 'PC2') {
                           pc2 <- df_X$PC2
                           exp_var_pc2 <- exp_var[2]
                           yaxis <- c("PC2")
                       } else if (pc2 == 'PC3') {
                           pc2 <- df_X$PC3
                           exp_var_pc2 <- exp_var[3]
                           yaxis <- c("PC3")
                       }

    pca_plot_names <- ggplot(df_X, aes(pc1, pc2, colour = metadata_group)) +
                        geom_text(aes(label = df_X$Library_ID), size = 4) +
                        scale_colour_manual(values = metadata_group_colors) +
                        # stat_ellipse() +
                        xlab(paste(xaxis, " - ", exp_var_pc1)) +
                        ylab(paste(yaxis, " - ", exp_var_pc2)) +
                        theme_minimal(base_size = 16) +
                        theme(text = element_text(size=16)) +
                        theme(legend.position = "top") 
    # +
    #                     theme(legend.title = element_blank())

    return(pca_plot_names)
}

```

```{r pca_plot_fxn}
# plot PCA with colored dots and the title including the # of species or genera
plot_pca <- function(df, pc1, pc2, color_group, shape_group, ncomps) {
    metadata_group_colors <- get(paste(color_group, "_colors", sep = ""))
    metadata_group_shapes <- get(paste(shape_group, "_shapes", sep = ""))

    pca.list <- mixOmics::pca(df, ncomp = ncomps, logratio = 'CLR')

    exp_var <- paste0(round(pca.list$explained_variance * 100, 2), "%")
    df_X <- pca.list$variates$X %>%
              as.data.frame() %>%
              rownames_to_column("Library_ID") %>%
              inner_join(full_metadata, by = "Library_ID")
    
    color_group = df_X[[color_group]]
    shape_group = df_X[[shape_group]]

                       if (pc1 == 'PC1') {
                           pc1 <- df_X$PC1
                           exp_var_pc1 <- exp_var[1]
                           xaxis <- c("PC1")
                       }  else if (pc1 == 'PC2') {
                           pc1 <- df_X$PC2
                           exp_var_pc1 <- exp_var[2]
                           xaxis <- c("PC2")
                       } else if (pc1 == 'PC3') {
                           pc1 <- df_X$PC3
                           exp_var_pc1 <- exp_var[3]
                           xaxis <- c("PC3")
                       }

                       if (pc2 == 'PC1') {
                           pc2 <- df_X$PC1
                           exp_var_pc2 <- exp_var[1]
                           yaxis <- c("PC1")
                       }  else if (pc2 == 'PC2') {
                           pc2 <- df_X$PC2
                           exp_var_pc2 <- exp_var[2]
                           yaxis <- c("PC2")
                       } else if (pc2 == 'PC3') {
                           pc2 <- df_X$PC3
                           exp_var_pc2 <- exp_var[3]
                           yaxis <- c("PC3")
                       }

    pca_plot <- ggplot(df_X, aes(pc1, pc2, colour = color_group, shape = shape_group)) +
                        geom_point(size = 2) +
                        scale_colour_manual(values = metadata_group_colors) +
                        scale_shape_manual(values = metadata_group_shapes) +
                        # stat_ellipse() +
                        xlab(paste(xaxis, " - ", exp_var_pc1)) +
                        ylab(paste(yaxis, " - ", exp_var_pc2)) +
                        theme_minimal(base_size = 16) +
                        theme(text = element_text(size=16)) +
                        theme(legend.title = element_blank(),
                              legend.key.size = unit(2,"mm"),
                              legend.text = element_text(size = 6)) +
                        theme(legend.position = "top")

    return(pca_plot)
}

```

```{r pca_plot_mid_fxn}
# plot PCA with colored dots 
plot_pca_mid <- function(df, pc1, pc2, color_group, shape_group, ncomps, title_text) {

    pca.list <- mixOmics::pca(df, ncomp = ncomps, logratio = 'CLR')

    exp_var <- paste0(round(pca.list$explained_variance * 100, 2), "%")
    df_X <- pca.list$variates$X %>%
              as.data.frame() %>%
              rownames_to_column("Library_ID") %>%
              inner_join(full_metadata, by = "Library_ID")
    
    color_group = df_X[[color_group]]
    shape_group = df_X[[shape_group]]

                       if (pc1 == 'PC1') {
                           pc1 <- df_X$PC1
                           exp_var_pc1 <- exp_var[1]
                           xaxis <- c("PC1")
                       }  else if (pc1 == 'PC2') {
                           pc1 <- df_X$PC2
                           exp_var_pc1 <- exp_var[2]
                           xaxis <- c("PC2")
                       } else if (pc1 == 'PC3') {
                           pc1 <- df_X$PC3
                           exp_var_pc1 <- exp_var[3]
                           xaxis <- c("PC3")
                       }

                       if (pc2 == 'PC1') {
                           pc2 <- df_X$PC1
                           exp_var_pc2 <- exp_var[1]
                           yaxis <- c("PC1")
                       }  else if (pc2 == 'PC2') {
                           pc2 <- df_X$PC2
                           exp_var_pc2 <- exp_var[2]
                           yaxis <- c("PC2")
                       } else if (pc2 == 'PC3') {
                           pc2 <- df_X$PC3
                           exp_var_pc2 <- exp_var[3]
                           yaxis <- c("PC3")
                       }

    pca_plot <- ggplot(df_X, aes(pc1, pc2, colour = color_group, shape = shape_group)) +
                        geom_point(size = 2) +
                        scale_colour_manual(values = c("#5A167E","#E95562","#FED395","#252525","#311068","#C83E73","#FEA873")) + # 
                        scale_shape_manual(values = c(16, 17, 15, 18)) +
                        # stat_ellipse() +
                        xlab(paste(xaxis, " - ", exp_var_pc1)) +
                        ylab(paste(yaxis, " - ", exp_var_pc2)) +
                        theme_minimal(base_size = 16) +
                        theme(text = element_text(size=16)) +
                        theme(legend.title = element_blank(),
                              legend.key.size = unit(2,"mm"),
                              legend.text = element_text(size = 6)) + # 
                        theme(legend.position = "top") +
                        ggtitle(title_text) + theme(plot.title = element_text(size = 10))

    return(pca_plot)
}

```

```{r pca_plot_cont_fxn}
# for continuous data
plot_pca_cont <- function(df, pc1, pc2, color_group, shape_group, ncomps, title_text) {

    pca.list <- mixOmics::pca(df, ncomp = ncomps, logratio = 'CLR')

    exp_var <- paste0(round(pca.list$explained_variance * 100, 2), "%")
    df_X <- pca.list$variates$X %>%
              as.data.frame() %>%
              rownames_to_column("Library_ID") %>%
              inner_join(full_metadata, by = "Library_ID")
    
    color_group = df_X[[color_group]]
    shape_group = df_X[[shape_group]]

                       if (pc1 == 'PC1') {
                           pc1 <- df_X$PC1
                           exp_var_pc1 <- exp_var[1]
                           xaxis <- c("PC1")
                       }  else if (pc1 == 'PC2') {
                           pc1 <- df_X$PC2
                           exp_var_pc1 <- exp_var[2]
                           xaxis <- c("PC2")
                       } else if (pc1 == 'PC3') {
                           pc1 <- df_X$PC3
                           exp_var_pc1 <- exp_var[3]
                           xaxis <- c("PC3")
                       }

                       if (pc2 == 'PC1') {
                           pc2 <- df_X$PC1
                           exp_var_pc2 <- exp_var[1]
                           yaxis <- c("PC1")
                       }  else if (pc2 == 'PC2') {
                           pc2 <- df_X$PC2
                           exp_var_pc2 <- exp_var[2]
                           yaxis <- c("PC2")
                       } else if (pc2 == 'PC3') {
                           pc2 <- df_X$PC3
                           exp_var_pc2 <- exp_var[3]
                           yaxis <- c("PC3")
                       }

    pca_plot <- ggplot(df_X, aes(pc1, pc2, colour = color_group, shape = shape_group)) +
                        geom_point(size = 2) +
                        scale_colour_viridis_c(option = "C") +
                        scale_shape_manual(values = c(16, 17, 15, 18)) +
                        # stat_ellipse() +
                        xlab(paste(xaxis, " - ", exp_var_pc1)) +
                        ylab(paste(yaxis, " - ", exp_var_pc2)) +
                        theme_minimal(base_size = 16) +
                        theme(text = element_text(size=16)) +
                        theme(legend.title = element_blank(),
                              legend.key.size = unit(2,"mm"),
                              legend.text = element_text(size = 6)) + 
                        theme(legend.position = "top") +
                        ggtitle(title_text) + theme(plot.title = element_text(size = 10))

    return(pca_plot)
}


```

```{r pca_biplot_fxn}
plot_pca_bi <- function(df, pc1, pc2, metadata_group, columntitle) {
    metadata_group_colors <- get(paste(metadata_group, "_colors", sep = ""))
    metadata_group_shapes <- get(paste(metadata_group, "_shapes", sep = ""))
   
    arrow_pc <- enquo(columntitle)
    
    exp_var <- paste0(round(df$explained_variance * 100, 2), "%") # explained variance for x- and y-labels
    
    # select only the PCs from the PCA and add metadata
    df_X <- df$variates$X %>%
              as.data.frame() %>%
              rownames_to_column("Library_ID") %>%
              inner_join(full_metadata, by = "Library_ID")
    
    metadata_group = df_X[[metadata_group]]
    
    corr_lam <- df$sdev[c("PC1", "PC2", "PC3")] * sqrt(nrow(df_X))

    df_X <- df_X %>%
      mutate(PC1 = PC1 / corr_lam[1],
             PC2 = PC2 / corr_lam[2], 
             PC3 = PC3 / corr_lam[3])

    # select the correct PC column and explained variance for PC1
    if (pc1 == 'PC1') {
        Pc1 <- df_X$PC1
        exp_var_pc1 <- exp_var[1]
        xaxis <- c("PC1")
    } else if (pc1 == 'PC2') {
        Pc1 <- df_X$PC2
        exp_var_pc1 <- exp_var[2]
        xaxis <- c("PC2")
    } else if (pc1 == 'PC3') {
        Pc1 <- df_X$PC3
        exp_var_pc1 <- exp_var[3]
        xaxis <- c("PC3")
   }
    
    # select the correct PC column and explained variance for PC2
    if (pc2 == 'PC1') {
        Pc2 <- df_X$PC1
        exp_var_pc2 <- exp_var[1]
        yaxis <- c("PC1")
 } else if (pc2 == 'PC2') {
       Pc2 <- df_X$PC2
       exp_var_pc2 <- exp_var[2]
       yaxis <- c("PC2")
    } else if (pc2 == 'PC3') {
       Pc2 <- df_X$PC3
       exp_var_pc2 <- exp_var[3]
       yaxis <- c("PC3")
   }
    
    # Identify the 10 pathways that have highest positive and negative loadings in the selected PC
    pws_10 <- df$loadings$X %>%
      as.data.frame(.) %>%
      rownames_to_column(var = "Taxon") %>%
      top_n(10, !!arrow_pc)

    neg_10 <- df$loadings$X %>%
      as.data.frame(.) %>%
      rownames_to_column(var = "Taxon") %>%
      top_n(-10, !!arrow_pc)


    pca_plot_bi <- ggplot(df_X, aes(x = Pc1, y = Pc2, colour = metadata_group)) +
      geom_point(size = 2.5, aes(shape = metadata_group) )+
      geom_segment(data = pws_10,
                   aes(xend = get(paste(pc1)), yend = get(paste(pc2))),
                   x = 0, y = 0, colour = "black",
                   size = 0.5,
                   arrow = arrow(length = unit(0.03, "npc"))) +
      geom_label_repel(data = pws_10,
                   aes(x = get(paste(pc1)), y = get(paste(pc2)), label = Taxon),
                   size = 1.8, colour = "grey20", label.padding = 0.2) +
      geom_segment(data = neg_10,
                   aes(xend = get(paste(pc1)), yend = get(paste(pc2))),
                   x = 0, y = 0, colour = "grey50",
                   size = 0.5,
                   arrow = arrow(length = unit(0.03, "npc"))) +
      geom_label_repel(data = neg_10,
                   aes(x = get(paste(pc1)), y = get(paste(pc2)), label = Taxon),
                   size = 1.8, colour = "grey20", label.padding = 0.2) +
      labs(x = paste(xaxis, " - ", exp_var_pc1),
           y = paste(yaxis, " - ", exp_var_pc2)) +
      scale_colour_manual(values = metadata_group_colors) +
      scale_shape_manual(values = metadata_group_shapes) +
      theme_minimal() + theme(text = element_text(size = 16)) +
      theme(text = element_text(size=16)) +
      theme(legend.position = "top")

    return(pca_plot_bi)
}

# pca_test_biplot <- plot_pca_bi(malt_species.pca, "PC3", "PC2", "Village", PC2)
# pca_test_biplot


```


List the outliers from the plot in `MID_CMB_Rad_Kil_Iber_decontam.Rmd`. 
```{r outliers}
outliers <- c("EXB059.A2501","LIB058.A0103","LIB058.A0106","CSS","CSD")
poor_samples <- fread("./05-results.backup/cuperdec_poor_samples.tsv") %>%
  pull(Sample)

outliersF <- str_c(outliers, collapse = "|")

```


#Sample Clustering with PCA

Step-by-step removal of taxa based on decontam and a filter threshold, followed
by PCA to see how contaminants/low abundance species affect the relationships of
samples to eachother between groups.
```{r pca_all_species}
# decontaminated only
all_species_decontam_unfilt <- all_species_decontam %>%
  select(-one_of(outliers)) %>%
  select(-one_of(poor_samples)) %>%
  adorn_totals(where = "col", name = "Sum_species.decontam") %>%
  mutate(Percent = Sum_species.decontam/sum(Sum_species.decontam)*100) %>%
  select(-one_of(c("Sum_species.decontam","Percent"))) %>%
  gather("Library_ID","Counts",2:ncol(.)) %>%
  mutate(Counts = Counts +1) %>%
  spread(Species,Counts) %>%
  column_to_rownames("Library_ID")

# prepare to run a PCA
# check the number of components to retain by tuning the PCA
tune.pca(all_species_decontam_unfilt, logratio = 'CLR')

# perform and plot a PCA to see how the data cluster
plot_pca(all_species_decontam_unfilt, "PC1", "PC2", "Site", "Time_period", 3)
plot_pca(all_species_decontam_unfilt, "PC3", "PC2", "Site", "Time_period", 3)

# decontaminated and filtered
all_species_decontam_filtered <- all_species_decontam %>%
  adorn_totals(where = "col", name = "Sum_species.decontam_filtered") %>%
  select(-one_of(outliers)) %>%
  select(-one_of(poor_samples)) %>%
  mutate(Percent = Sum_species.decontam_filtered/sum(Sum_species.decontam_filtered)*100) %>%
  filter(Percent > 0.005) %>%
  select(-one_of(c("Sum_species.decontam_filtered","Percent"))) %>%
  gather("Library_ID","Counts",2:ncol(.)) %>%
  mutate(Counts = Counts +1) %>%
  spread(Species,Counts) %>%
  column_to_rownames("Library_ID")

# prepare to run a PCA
# check the number of components to retain by tuning the PCA
tune.pca(all_species_decontam_filtered, logratio = 'CLR')

# perform and plot a PCA to see how the data cluster
all_samples_sp_pca12 <- plot_pca(all_species_decontam_filtered, "PC1", "PC2", "Site", "Time_period", 3)
all_samples_sp_pca12
all_samples_sp_pca32 <- plot_pca(all_species_decontam_filtered, "PC3", "PC2", "Site", "Time_period", 3)
all_samples_sp_pca32

# ggsave("./06-publication/supplemental_figures/SXX28/SXX28_all_samples_sp_pca12.pdf", plot = all_samples_sp_pca12, device = "pdf",
#        scale = 1, width = 6, height = 7, units = c("in"), dpi = 300)


```
PC1 and PC2 don't distinguish the samples by time period, but PC3 does. 

# Species-level analyses
What is the relationship of the groups in a PCA without the blanks? Let's try
the PCA using only the species that are present in at least 10% of at least one
sample group (MID, CMB, Radcliffe, Kilteasheen, Iberia)

## Industrial and Medieval samples
What is the relationship of the groups in a PCA without the blanks or the
Yanomami samples? Here we'll use a cut-off of 10% prevalence in any group to
keep the species. If a species is not present in at
least 10% of at least 1 group, it will be removed from analysis.
```{r pca_pct10_noblanks}

# Determine the prevalence of species in each group
all_species_decontam_noblanks_presence_more_10 <- all_species_decontam %>%
  select(-matches("EXB|LIB|CSS|CSD|CSN|CSL")) %>%
  select(-one_of(outliers)) %>%
  select(-one_of(poor_samples)) %>%
  mutate_if(is.numeric, ~1 * (. > 0)) %>%
  gather("Library_ID","Counts",2:ncol(.)) %>%
  inner_join(., full_metadata %>%
               select(Library_ID, Region), by = "Library_ID") %>%
  group_by(Region, Species) %>%
  summarize(Percent_presence = sum(Counts)/length(Library_ID)*100) %>%
  # select the species that are present in at least 10% of at least one group
  ungroup() %>%
  filter(Percent_presence >= 10) %>%
  distinct(Species) %>%
  select(Species)


# Species, no blanks 
all_species_decontam_noblanks <- all_species_decontam %>%
  select(-matches("EXB|LIB|CSS|CSD|CSN|CSL")) %>%
  select(-one_of(outliers)) %>%
  select(-one_of(poor_samples)) %>%
  inner_join(., all_species_decontam_noblanks_presence_more_10, by = "Species") %>%
  gather("Library_ID","Counts",2:ncol(.)) %>%
  mutate(Counts = Counts +1) %>%
  spread(Species,Counts) %>%
  column_to_rownames("Library_ID")

# perform PCA
plot_pca(all_species_decontam_noblanks, "PC1", "PC2", "Region", "Time_period", 3)
plot_pca(all_species_decontam_noblanks, "PC3", "PC2", "Region", "Time_period", 3)

plot_pca(all_species_decontam_noblanks, "PC1", "PC2", "Site", "Time_period", 3)
plot_pca(all_species_decontam_noblanks, "PC1", "PC2", "Time_period", "Time_period", 3)
plot_pca(all_species_decontam_noblanks, "PC1", "PC2", "Sex", "Time_period", 3)
plot_pca(all_species_decontam_noblanks, "PC1", "PC2", "Periodontal_disease_present", "Time_period", 3)
plot_pca(all_species_decontam_noblanks, "PC1", "PC2", "Caries_present", "Time_period", 3)
plot_pca(all_species_decontam_noblanks, "PC1", "PC2", "Tooth_location", "Time_period", 3)
plot_pca(all_species_decontam_noblanks, "PC1", "PC2", "alt_tooth_location", "Time_period", 3)

```

Here we use a cut-off of 0.001% abundance across all samples. This leaves 424
species, only slightly fewer than the 497 kept by filtering based on prevalence.
The PCA plots are indistinguishable to me. Is one necessarily better than the
other to use?
```{r pca_abd_noblanks_species}
all_species_decontam_filtered_noblanks <- all_species_decontam %>%
  select(-matches("EXB|LIB|CSS|CSD|CSN|CSL")) %>%
  select(-one_of(outliers)) %>%
  select(-one_of(poor_samples)) %>%
  adorn_totals(where = "col", name = "Sum_species") %>%
  mutate(Percent = Sum_species/sum(Sum_species)*100) %>%
  filter(Percent > 0.001) %>%
  select(-one_of(c("Sum_species","Percent"))) %>%
  gather("Library_ID","Counts",2:ncol(.)) %>% 
  mutate(Counts = Counts +1) %>%
  spread(Species,Counts) %>% 
  column_to_rownames("Library_ID")

# prepare to run a PCA
# check the number of components to retain by tuning the PCA
tune.pca(all_species_decontam_filtered_noblanks, logratio = 'CLR')

# perform and plot a PCA to see how the data cluster
# by Site
plot_pca(all_species_decontam_filtered_noblanks, "PC1", "PC2", "Site", "Time_period", 3)
plot_pca(all_species_decontam_filtered_noblanks, "PC1", "PC3", "Site", "Time_period", 3)

# ggsave("05-results.backup/pca_village.pdf", plot = pca_village, device = "pdf",
#        scale = 1, width = 7, height = 5, units = c("in"), dpi = 300)

# by Region
plot_pca(all_species_decontam_filtered_noblanks, "PC1", "PC2", "Region", "Time_period", 3)

# by Time period
plot_pca(all_species_decontam_filtered_noblanks, "PC1", "PC2", "Time_period", "Time_period", 3)

# by Sex
plot_pca(all_species_decontam_filtered_noblanks, "PC1", "PC2", "Sex", "Time_period", 3)

# by periodontal diseae (anywhere in the mouth)
plot_pca(all_species_decontam_filtered_noblanks, "PC1", "PC2", "Periodontal_disease_present", "Time_period", 3)

# by caries (anywhere in the mouth)
plot_pca(all_species_decontam_filtered_noblanks, "PC1", "PC2", "Caries_present", "Time_period", 3)

# by sampled tooth/teeth location
plot_pca(all_species_decontam_filtered_noblanks, "PC1", "PC2", "Tooth_location", "Time_period", 3)

# by sampled tooth/teeth location
plot_pca(all_species_decontam_filtered_noblanks, "PC1", "PC2", "alt_tooth_location", "Time_period", 3)

# by Individual sequencing depth
# perform a PCA to see how the data cluster

malt_species_decontam_filtered_noblanks.pca <- mixOmics::pca(all_species_decontam_filtered_noblanks, ncomp = 3, logratio = 'CLR')
plot(malt_species_decontam_filtered_noblanks.pca)

# Select out the PC variates into a new table and add the metadata for plotting
malt_species_decontam_filtered_noblanks.pca.variates <- as.data.frame(malt_species_decontam_filtered_noblanks.pca$variates$X)
malt_species_decontam_filtered_noblanks.pca.variates <- malt_species_decontam_filtered_noblanks.pca.variates %>%
  rownames_to_column("Library_ID")

malt_species_decontam_filtered_noblanks.pca.varmet <- inner_join(malt_species_decontam_filtered_noblanks.pca.variates, full_metadata, by = "Library_ID")

malt_species_decontam_filtered_noblanks.pca.varmet %>%
  ggplot(., aes(PC1, PC2, colour = No_reads_total, shape = Time_period)) +
    geom_point(size = 2) +
    scale_shape_manual(values = Time_period_shapes) +
    scale_color_viridis_c(option = "A") +
    xlab(paste("PC1 - ", 100*(round(malt_species_decontam_filtered_noblanks.pca$explained_variance[1], 2)), "%", sep = "")) +
    ylab(paste("PC2 - ", 100*(round(malt_species_decontam_filtered_noblanks.pca$explained_variance[2], 2)), "%", sep = "")) +
    theme_minimal(base_size = 14) +
    labs(colour = "Seq. depth\n(reads)", shape = "Time period") +
    theme(legend.title = element_text(size = 10))

malt_species_decontam_filtered_noblanks.pca.varmet %>%
  mutate(No_reads_total_log = log10(No_reads_total)) %>%
  ggplot(., aes(PC1, PC2, colour = No_reads_total_log, shape = Time_period)) +
    geom_point(size = 2) +
    scale_shape_manual(values = Time_period_shapes) +
    scale_color_viridis_c(option = "A") +
    xlab(paste("PC1 - ", 100*(round(malt_species_decontam_filtered_noblanks.pca$explained_variance[1], 2)), "%", sep = "")) +
    ylab(paste("PC2 - ", 100*(round(malt_species_decontam_filtered_noblanks.pca$explained_variance[2], 2)), "%", sep = "")) +
    theme_minimal(base_size = 14) +
    labs(colour = "Log10\n(Seq. depth)", shape = "Time period") +
    theme(legend.title = element_text(size = 10))


# ggsave("05-results.backup/librarybatch.pdf", plot = extractionbatch, device = "pdf",
#        scale = 1, width = 7, height = 5, units = c("in"), dpi = 300)

```

```{r}

# by Site
plot_pca(all_species_decontam_filtered_noblanks, "PC3", "PC2", "Site", "Time_period", 3)

# by Region
plot_pca(all_species_decontam_filtered_noblanks, "PC3", "PC2", "Region", "Time_period", 3)

# by Time period
plot_pca(all_species_decontam_filtered_noblanks, "PC3", "PC2", "Time_period", "Time_period", 3)

# by Sex
plot_pca(all_species_decontam_filtered_noblanks, "PC3", "PC2", "Sex", "Time_period", 3)

# by periodontal diseae (anywhere in the mouth)
plot_pca(all_species_decontam_filtered_noblanks, "PC3", "PC2", "Periodontal_disease_present", "Time_period", 3)

# by caries (anywhere in the mouth)
plot_pca(all_species_decontam_filtered_noblanks, "PC3", "PC2", "Caries_present", "Time_period", 3)

# by sampled tooth/teeth location
plot_pca(all_species_decontam_filtered_noblanks, "PC3", "PC2", "Tooth_location", "Time_period", 3)

# by sampled tooth/teeth location
plot_pca(all_species_decontam_filtered_noblanks, "PC3", "PC2", "alt_tooth_location", "Time_period", 3)

# by sequencing depth

malt_species_decontam_filtered_noblanks.pca.varmet %>%
  ggplot(., aes(PC3, PC2, colour = No_reads_total, shape = Time_period)) +
    geom_point(size = 2) +
    scale_shape_manual(values = Time_period_shapes) +
    scale_color_viridis_c(option = "A") +
    xlab(paste("PC3 - ", 100*(round(malt_species_decontam_filtered_noblanks.pca$explained_variance[3], 2)), "%", sep = "")) +
    ylab(paste("PC2 - ", 100*(round(malt_species_decontam_filtered_noblanks.pca$explained_variance[2], 2)), "%", sep = "")) +
    theme_minimal(base_size = 14) +
    labs(colour = "Seq. depth\n(reads)", shape = "Time period") +
    theme(legend.title = element_text(size = 10))

malt_species_decontam_filtered_noblanks.pca.varmet %>%
  mutate(No_reads_total_log = log10(No_reads_total)) %>%
  ggplot(., aes(PC3, PC2, colour = No_reads_total_log, shape = Time_period)) +
    geom_point(size = 2) +
    scale_shape_manual(values = Time_period_shapes) +
    scale_color_viridis_c(option = "A") +
    xlab(paste("PC3 - ", 100*(round(malt_species_decontam_filtered_noblanks.pca$explained_variance[3], 2)), "%", sep = "")) +
    ylab(paste("PC2 - ", 100*(round(malt_species_decontam_filtered_noblanks.pca$explained_variance[2], 2)), "%", sep = "")) +
    theme_minimal(base_size = 14) +
    labs(colour = "Log10\n(Seq. depth)", shape = "Time period") +
    theme(legend.title = element_text(size = 10))


```


Make biplots of the top 10 loading species in positive and negative directions
of PC1 and PC2, using the abundance filter.
```{r biplot_noblanks}
# remove the Yanomami samples from the input table
all_species_decontam_filtered_noblanks <- all_species_decontam %>%
  select(-matches("EXB|LIB|CSS|CSD|CSN|CSL")) %>%
  select(-one_of(outliers)) %>%
  select(-one_of(poor_samples)) %>%
  adorn_totals(where = "col", name = "Sum_species") %>%
  mutate(Percent = Sum_species/sum(Sum_species)*100) %>%
  filter(Percent > 0.001) %>%
  select(-one_of(c("Sum_species","Percent"))) %>%
  gather("Library_ID","Counts",2:ncol(.)) %>% 
  mutate(Counts = Counts +1) %>%
  spread(Species,Counts) %>%
  arrange(Library_ID) %>%
  column_to_rownames("Library_ID")

# perform PCA
all_species_decontam_filtered_noblanks.pca <- mixOmics::pca(all_species_decontam_filtered_noblanks, ncomp = 3, logratio = 'CLR')

# First plot by Description (do HMP supragingival and HMP subgingival plaque samples separate?) to see what it looks like before adding the biplot over the points
plot_pca(all_species_decontam_filtered_noblanks, "PC1", "PC2", "Region", "Time_period", 3)
plot_pca(all_species_decontam_filtered_noblanks, "PC3", "PC2", "Region", "Time_period", 3)

# Now plot biplots, and make tables of the top 10 species in each loading
malt_species_decontam_filtered_biplot_121 <- plot_pca_bi(all_species_decontam_filtered_noblanks.pca, "PC1", "PC2", "Region", PC1)
malt_species_decontam_filtered_biplot_121

pandoc.table(malt_species_decontam_filtered_biplot_121$plot_env$pws_10 %>% arrange(desc(PC1)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC1 positive loadings")

pandoc.table(malt_species_decontam_filtered_biplot_121$plot_env$neg_10 %>% arrange(PC1),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC1 negative loadings")


malt_species_decontam_filtered_biplot_122 <- plot_pca_bi(all_species_decontam_filtered_noblanks.pca, "PC1", "PC2", "Region", PC2)
malt_species_decontam_filtered_biplot_122

pandoc.table(malt_species_decontam_filtered_biplot_122$plot_env$pws_10 %>% arrange(desc(PC2)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC2 positive loadings")

pandoc.table(malt_species_decontam_filtered_biplot_122$plot_env$neg_10 %>% arrange(PC2),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC2 negative loadings")


malt_species_decontam_filtered_biplot_322 <- plot_pca_bi(all_species_decontam_filtered_noblanks.pca, "PC3", "PC2", "Region", PC2)
malt_species_decontam_filtered_biplot_322

malt_species_decontam_filtered_biplot_323 <- plot_pca_bi(all_species_decontam_filtered_noblanks.pca, "PC3", "PC2", "Region", PC3)
malt_species_decontam_filtered_biplot_323

pandoc.table(malt_species_decontam_filtered_biplot_323$plot_env$pws_10 %>% arrange(desc(PC3)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC3 positive loadings")

pandoc.table(malt_species_decontam_filtered_biplot_323$plot_env$neg_10 %>% arrange(PC3),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC3 negative loadings")

```


Run a permanova with function adonis to test if the clusters are significantly
different. First load the metadata again so the blanks and Yanomami samples can
be removed and then make them factors.
```{r metadata_betadisper_noblanks}

metadata_noblanks <- fread("./00-documentation.backup/radcliffe_pfuT_metadata.tsv") %>%
  bind_rows(., fread("./00-documentation.backup/iberian_medieval_metadata.tsv")) %>%
  bind_rows(., fread("./00-documentation.backup/kilteasheen_metadata.tsv")) %>%
  full_join(., fread("./00-documentation.backup/MID_analysis_metadata.tsv"))


metadata_noblanks <- metadata_noblanks %>%
  as_tibble() %>%
  filter(!Library_ID %in% outliers) %>%
  filter(!Library_ID %in% poor_samples) %>%
  filter(!str_detect(Library_ID, "CSN|CSL|EXB|LIB")) %>%
  # remove the MID samples that have no metadata
  filter(!str_detect(Library_ID, "MID025.A0101|MID033.A0101|MID056.A0101|MID065.A0101|MID068.A0101|MID076.A0101|MID078.A0101")) %>%
  mutate(Site = fct_relevel(Site, "Middenbeemster", "CMB", "Radcliffe", "Kilteasheen",
                                   "ELR", "IVE"),
         Time_period = fct_relevel(Time_period, "Industrial","Medieval"),
         Region = fct_relevel(Region, "Netherlands","Spain","England", "Ireland","Iberia"),
         Periodontal_disease_present = fct_relevel(Periodontal_disease_present, "Yes","No","Unobservable"),
         Caries_present = fct_relevel(Caries_present, "Yes","No","Unobservable"),
         Sex = fct_relevel(Sex, "Female","Male","Indeterminate")) %>%
  select(Library_ID, Site, Time_period, Region, Periodontal_disease_present, Caries_present, Sex) %>%
  arrange(Library_ID) %>%
  unique()

```

Run a PERMANOVA with adonis2 on the abundance-filtered clr-transformed species matrix.
```{r adonis_noblanks_species}
# Euclidean distance, uses the clr-transformed matrix from the mixOmics PCA above
all_species_decontam_filtered_noblanks.euc <- vegdist(all_species_decontam_filtered_noblanks.pca$X, method = "euclidean", na.rm = T)
all_species_decontam_filtered_noblanks.euc.pcoa <- ape::pcoa(all_species_decontam_filtered_noblanks.euc)

all_species_decontam_filtered_noblanks.euc.pcoavectors <- as.data.frame(all_species_decontam_filtered_noblanks.euc.pcoa$vectors)
all_species_decontam_filtered_noblanks.euc.pcoavectors <- all_species_decontam_filtered_noblanks.euc.pcoavectors %>%
  rownames_to_column("Library_ID")

all_species_decontam_filtered_noblanks.euc.vecmet <- inner_join(all_species_decontam_filtered_noblanks.euc.pcoavectors, full_metadata, by = "Library_ID")

all_species_decontam_filtered_noblanks.euc.vecmet %>%
  ggplot(., aes(Axis.1, Axis.2, colour = Time_period, shape = Time_period)) +
  geom_point(size = 3) +
  scale_colour_manual(values = Time_period_colors) +
  scale_shape_manual(values = Time_period_shapes) +
  stat_ellipse(aes(colour = Time_period, group = Time_period)) +
  theme_minimal(base_size = 7) +
  ggtitle("Time_period")

# test for difference between different metadata categories 
adonis2(all_species_decontam_filtered_noblanks.euc ~ Region, metadata_noblanks, permutations = 999, by = "margin")
adonis2(all_species_decontam_filtered_noblanks.euc ~ Site, metadata_noblanks, permutations = 999, by = "margin")
adonis2(all_species_decontam_filtered_noblanks.euc ~ Time_period, metadata_noblanks, permutations = 999, by = "margin")


# controlled for tooth site
adonis2(all_species_decontam_filtered_noblanks.euc ~ Time_period, metadata_noblanks, permutations = 999, by = "margin", strata = metadata_noblanks$Region)

```

Then run betadisper to see if the groups with significant differences in
clustering have different dispersion ("tight" vs. "loose" clustering of groups)
- is one group more variable than the other(s)?
```{r betadisper_noblanks_species}
# Test the 3 metadata categories that are significant (p < 0.05) based on adonis for significant differences in dispersion
# Site, Region, Time period

# Site
groups_site <- metadata_noblanks %>%
  select(Site) %>% 
  pull()

## Calculate multivariate dispersions
site_betadisper_noblanks <- betadisper(all_species_decontam_filtered_noblanks.euc, groups_site, bias.adjust =TRUE)
site_betadisper_noblanks

## Perform test
anova(site_betadisper_noblanks)

## Permutation test for F
permutest(site_betadisper_noblanks, pairwise = TRUE, permutations = 99)

## Tukey's Honest Significant Differences
site_betadisper_noblanks.HSD <- TukeyHSD(site_betadisper_noblanks)
plot(site_betadisper_noblanks.HSD)

## Plot the groups and distances to centroids on the
## first two PCoA axes
## with data ellipses instead of hulls
plot(site_betadisper_noblanks, ellipse = TRUE, hull = FALSE) # 1 sd data ellipse

## can also specify which axes to plot, ordering respected
# plot(site_betadisper_noblanks, axes = c(3,1), seg.col = "forestgreen", seg.lty = "dashed")

## Draw a boxplot of the distances to centroid for each group
boxplot(site_betadisper_noblanks)


# Region
groups_region <- metadata_noblanks %>%
  select(Region) %>% 
  pull()

## Calculate multivariate dispersions
region_betadisper_noblanks <- betadisper(all_species_decontam_filtered_noblanks.euc, groups_region, bias.adjust =TRUE)
region_betadisper_noblanks

## Perform test
anova(region_betadisper_noblanks)

## Permutation test for F
permutest(region_betadisper_noblanks, pairwise = TRUE, permutations = 99)

## Tukey's Honest Significant Differences
region_betadisper_noblanks.HSD <- TukeyHSD(region_betadisper_noblanks)
plot(region_betadisper_noblanks.HSD)

## Plot with data ellipses instead of hulls
plot(region_betadisper_noblanks, ellipse = TRUE, hull = FALSE) # 1 sd data ellipse

## Draw a boxplot of the distances to centroid for each group
boxplot(region_betadisper_noblanks)


# Time period
groups_tp <- metadata_noblanks %>%
  select(Time_period) %>% 
  pull()

## Calculate multivariate dispersions
tp_betadisper_noblanks <- betadisper(all_species_decontam_filtered_noblanks.euc, groups_tp, bias.adjust =TRUE)
tp_betadisper_noblanks

## Perform test
anova(tp_betadisper_noblanks)

## Permutation test for F
permutest(tp_betadisper_noblanks, pairwise = TRUE, permutations = 99)

## Tukey's Honest Significant Differences
tp_betadisper_noblanks.HSD <- TukeyHSD(tp_betadisper_noblanks)
plot(tp_betadisper_noblanks.HSD)

## Plot with data ellipses instead of hulls
plot(tp_betadisper_noblanks, ellipse = TRUE, hull = FALSE) # 1 sd data ellipse

## Draw a boxplot of the distances to centroid for each group
boxplot(tp_betadisper_noblanks)

```


## Middenbeemster samples only
What is the relationship of Cameroon plaque samples to eachother without any
other sample types? Do they cluster by any of the metadata categories?
Filter the species table by % abundance
```{r pca_species_mid_pc12}
all_species_decontam_filtered_mid <- all_species_decontam %>%
  select(-one_of(outliers)) %>%
  select(-one_of(poor_samples)) %>%
  select(matches("Species|MID")) %>%
  # adorn_totals(where = "col", name = "Sum_species") %>%
  # mutate(Percent = Sum_species/sum(Sum_species)*100) %>%
  # filter(Percent > 0.001) %>%
  select(-one_of(c("Sum_species","Percent"))) %>%
  gather("Library_ID","Counts",2:ncol(.)) %>% 
  mutate(Counts = Counts +1) %>%
  spread(Species,Counts) %>%
  column_to_rownames("Library_ID")

# prepare to run a PCA
# check the number of components to retain by tuning the PCA
tune.pca(all_species_decontam_filtered_mid, logratio = 'CLR')

# perform and plot a PCA to see how the data cluster
# by pipe notch presence
plot_pca_mid(all_species_decontam_filtered_mid, "PC1", "PC2", "Pipenotch", "Pipenotch", 3, "Pipe notch")

# ggsave("05-results.backup/pca_mid_village.pdf", plot = pca_mid_village, device = "pdf",
#        scale = 1, width = 7, height = 5, units = c("in"), dpi = 300)

# by sex
plot_pca_mid(all_species_decontam_filtered_mid, "PC1", "PC2", "Sex", "Pipenotch", 3, "Sex")

# by age
plot_pca_mid(all_species_decontam_filtered_mid, "PC1", "PC2", "Age", "Pipenotch", 3, "Age")

# by presence of periodontal disease
plot_pca_mid(all_species_decontam_filtered_mid, "PC1", "PC2", "Periodontal_disease_present", "Pipenotch", 3, "Periodontal disease present")

# by presence of caries
plot_pca_mid(all_species_decontam_filtered_mid, "PC1", "PC2", "Caries_present", "Pipenotch", 3, "Caries present")

# by ante-mortem tooth loss
plot_pca_mid(all_species_decontam_filtered_mid, "PC1", "PC2", "AMTL", "Pipenotch", 3, "AMTL")

# by supragingival calculus score
plot_pca_cont(all_species_decontam_filtered_mid, "PC1", "PC2", "calc_sup_SI_score", "Pipenotch", 3, "Sup calc SI score")

# by subgingival calculus score
plot_pca_cont(all_species_decontam_filtered_mid, "PC1", "PC2", "Calc_sub_SI_score", "Pipenotch", 3, "Sub calc SI score")

# by presence of perioapical lesions
plot_pca_mid(all_species_decontam_filtered_mid, "PC1", "PC2", "Periapical_lesions", "Pipenotch", 3, "Periapical lesions")

# by max periodontal disease score
plot_pca_cont(all_species_decontam_filtered_mid, "PC1", "PC2", "Max_Perio_Score", "Pipenotch", 3, "Max perio score")

# by periodontal affect score
plot_pca_cont(all_species_decontam_filtered_mid, "PC1", "PC2", "Peridontal_affect_score", "Pipenotch", 3, "Perio affect score")

# by % antemortem tooth loss
plot_pca_cont(all_species_decontam_filtered_mid, "PC1", "PC2", "%AMTL", "Pipenotch", 3, " % AMTL")

# by % positions w/periapical lesions
plot_pca_cont(all_species_decontam_filtered_mid, "PC1", "PC2", "%_positions_perioapical", "Pipenotch", 3, "% positions periapical lesions")

# by % teeth with caries
plot_pca_cont(all_species_decontam_filtered_mid, "PC1", "PC2", "%teeth_with_caries", "Pipenotch", 3, "% teeth w/ caries")

# by % teeth with calculus
plot_pca_cont(all_species_decontam_filtered_mid, "PC1", "PC2", "%teeth_with_calc", "Pipenotch", 3, "% teeth w/ calc")

# by % positions with periodontal disease
plot_pca_cont(all_species_decontam_filtered_mid, "PC1", "PC2", "%_positions_with_Periodontal", "Pipenotch", 3, "% positions w/ PD")

# by minimum number of pipenotches 
plot_pca_cont(all_species_decontam_filtered_mid, "PC1", "PC2", "Min_no_Pipe_notches", "Pipenotch", 3, "Min # pipe notches")

# by sequencing depth 
plot_pca_cont(all_species_decontam_filtered_mid, "PC1", "PC2", "No_reads_total", "Pipenotch", 3, "Total reads into MALT")

# by sampled tooth site(s)
plot_pca_mid(all_species_decontam_filtered_mid, "PC1", "PC2", "Tooth_location", "Pipenotch", 3, "Tooth location")

# by sampled tooth site(s)
plot_pca_mid(all_species_decontam_filtered_mid, "PC1", "PC2", "alt_tooth_location", "Pipenotch", 3, "Tooth location")

# by mg of extracted calculus
plot_pca_cont(all_species_decontam_filtered_mid, "PC1", "PC2", "Calculus_extracted_mg", "Pipenotch", 3, "Weight of extracted calculus")

# with names
# pca.list <- mixOmics::pca(all_species_decontam_filtered_mid, ncomp = 3, logratio = 'CLR')
# plot_pca_names(pca.list, "PC1", "PC2", "Pipenotch")

# ggsave("05-results.backup/pca_nt_species_pc1pc2.pdf", plot = pca_nt_species_pc1pc2, device = "pdf",
#        scale = 1, width = 7, height = 5, units = c("in"),
#        dpi = 300)
```

```{r pca_species_mid_pc32}
# by sex
plot_pca_mid(all_species_decontam_filtered_mid, "PC3", "PC2", "Sex", "Pipenotch", 3, "Sex")

# by age
plot_pca_mid(all_species_decontam_filtered_mid, "PC3", "PC2", "Age", "Pipenotch", 3, "Age")

# by pipe notch presence
plot_pca_mid(all_species_decontam_filtered_mid, "PC3", "PC2", "Pipenotch", "Pipenotch", 3, "Pipe notch")

# by presence of periodontal disease
plot_pca_mid(all_species_decontam_filtered_mid, "PC3", "PC2", "Periodontal_disease_present", "Pipenotch", 3, "Periodontal disease present")

# by presence of caries
plot_pca_mid(all_species_decontam_filtered_mid, "PC3", "PC2", "Caries_present", "Pipenotch", 3, "Caries present")

# by ante-mortem tooth loss
plot_pca_mid(all_species_decontam_filtered_mid, "PC3", "PC2", "AMTL", "Pipenotch", 3, "AMTL")

# by supragingival calculus score
plot_pca_cont(all_species_decontam_filtered_mid, "PC3", "PC2", "calc_sup_SI_score", "Pipenotch", 3, "Sup calc SI score")

# by subgingival calculus score
plot_pca_cont(all_species_decontam_filtered_mid, "PC3", "PC2", "Calc_sub_SI_score", "Pipenotch", 3, "Sub calc SI score")

# by presence of perioapical lesions
plot_pca_mid(all_species_decontam_filtered_mid, "PC3", "PC2", "Periapical_lesions", "Pipenotch", 3, "Periapical lesions")

# by max periodontal disease score
plot_pca_cont(all_species_decontam_filtered_mid, "PC3", "PC2", "Max_Perio_Score", "Pipenotch", 3, "Max perio score")

# by periodontal affect score
plot_pca_cont(all_species_decontam_filtered_mid, "PC3", "PC2", "Peridontal_affect_score", "Pipenotch", 3, "Perio affect score")

# by % antemortem tooth loss
plot_pca_cont(all_species_decontam_filtered_mid, "PC3", "PC2", "%AMTL", "Pipenotch", 3, "% AMTL")

# by % positions w/periapical lesions
plot_pca_cont(all_species_decontam_filtered_mid, "PC3", "PC2", "%_positions_perioapical", "Pipenotch", 3, "% positions periapical lesions")

# by % teeth with caries
plot_pca_cont(all_species_decontam_filtered_mid, "PC3", "PC2", "%teeth_with_caries", "Pipenotch", 3, "% teeth w/ caries")

# by % teeth with calculus
plot_pca_cont(all_species_decontam_filtered_mid, "PC3", "PC2", "%teeth_with_calc", "Pipenotch", 3, "% teeth w/ calc")

# by % positions with periodontal disease
plot_pca_cont(all_species_decontam_filtered_mid, "PC3", "PC2", "%_positions_with_Periodontal", "Pipenotch", 3, "% positions w/ PD")

# by minimum number of pipenotches 
plot_pca_cont(all_species_decontam_filtered_mid, "PC3", "PC2", "Min_no_Pipe_notches", "Pipenotch", 3, "Min # pipe notches")

# by sequencing depth 
plot_pca_cont(all_species_decontam_filtered_mid, "PC3", "PC2", "No_reads_total", "Pipenotch", 3, "Total reads into MALT")

# by sampled tooth site(s)
plot_pca_mid(all_species_decontam_filtered_mid, "PC3", "PC2", "Tooth_location", "Pipenotch", 3, "Tooth location")

# by sampled tooth site(s)
plot_pca_mid(all_species_decontam_filtered_mid, "PC3", "PC2", "alt_tooth_location", "Pipenotch", 3, "Tooth location")

# by mg of extracted calculus
plot_pca_cont(all_species_decontam_filtered_mid, "PC3", "PC2", "Calculus_extracted_mg", "Pipenotch", 3, "Weight of extracted calculus")


```



```{r biplots_cmc_species}

all_species_decontam_filtered_mid <- all_species_decontam %>%
  select(-one_of(outliers)) %>%
  select(-one_of(poor_samples)) %>%
  select(matches("Species|MID")) %>%
  adorn_totals(where = "col", name = "Sum_species") %>%
  mutate(Percent = Sum_species/sum(Sum_species)*100) %>%
  filter(Percent > 0.001) %>%
  select(-one_of(c("Sum_species","Percent"))) %>%
  gather("Library_ID","Counts",2:ncol(.)) %>% 
  mutate(Counts = Counts +1) %>%
  spread(Species,Counts) %>%
  column_to_rownames("Library_ID")

all_species_decontam_filtered_mid.pca <- mixOmics::pca(all_species_decontam_filtered_mid, ncomp = 3, logratio = 'CLR')
plot(all_species_decontam_filtered_mid.pca)

# Define pipenotch colors for plotting
Pipenotch_colors = c("#311068","#C83E73")
Pipenotch_shapes = c(19,17)

all_species_decontam_filtered_nocontrols_biplot_121 <- plot_pca_bi(all_species_decontam_filtered_mid.pca, "PC1", "PC2", "Pipenotch", PC1)
all_species_decontam_filtered_nocontrols_biplot_121

all_species_decontam_filtered_nocontrols_biplot_122 <- plot_pca_bi(all_species_decontam_filtered_mid.pca, "PC1", "PC2", "Pipenotch", PC2)
all_species_decontam_filtered_nocontrols_biplot_122

pandoc.table(all_species_decontam_filtered_nocontrols_biplot_121$plot_env$pws_10 %>% arrange(desc(PC1)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC1 positive loadings")

pandoc.table(all_species_decontam_filtered_nocontrols_biplot_121$plot_env$neg_10 %>% arrange(PC1),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC1 negative loadings")


pandoc.table(all_species_decontam_filtered_nocontrols_biplot_122$plot_env$pws_10 %>% arrange(desc(PC2)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC2 positive loadings")

pandoc.table(all_species_decontam_filtered_nocontrols_biplot_122$plot_env$neg_10 %>% arrange(PC2),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC2 negative loadings")


all_species_decontam_filtered_nocontrols_biplot_322 <- plot_pca_bi(all_species_decontam_filtered_mid.pca, "PC3", "PC2", "Pipenotch", PC2)
all_species_decontam_filtered_nocontrols_biplot_322

all_species_decontam_filtered_nocontrols_biplot_323 <- plot_pca_bi(all_species_decontam_filtered_mid.pca, "PC3", "PC2", "Pipenotch", PC3)
all_species_decontam_filtered_nocontrols_biplot_323

pandoc.table(all_species_decontam_filtered_nocontrols_biplot_323$plot_env$pws_10 %>% arrange(desc(PC3)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC3 positive loadings")

pandoc.table(all_species_decontam_filtered_nocontrols_biplot_323$plot_env$neg_10 %>% arrange(PC3),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC3 negative loadings")

```





