---
title: "MID/CMB, Radcliffe, Kilteasheen, Iberian Medieval, calculus MetaPhlAn3 beta diversity"
author: "Irina Velsko"
date: "`r format(Sys.time(), '%b %d, %Y')`"
output:
    github_document:
        fig_width: 8
        fig_height: 6
        toc: true
---


```{r load_libraries, echo = F, message = F}
library(knitr)
library(data.table)
library(psych)
library(ape)
library(mixOmics)
library(compositions)
library(vegan)
library(janitor)
library(pander)
library(tidyverse)
library(gplots)
library(ggrepel)
library(viridis)
library(cowplot)
opts_chunk$set(echo=F, warning=F, message=F, dpi=150)
```

Set the working directory of the notebook to the parent directory of the 
notebook itself.  

```{r setup}
knitr::opts_knit$set(root.dir = normalizePath(".."))
```

# MetaPhlAn3

```{r load_data}
# read in the species table
mpa3_raw <- fread("./05-results.backup/mpa3_abundance_table.tsv") %>%
  select(-NCBI_tax_id)

# clean the file names
colnames(mpa3_raw) <- gsub(".metaphlan3","", colnames(mpa3_raw))
colnames(mpa3_raw) <- gsub(".SG1","", colnames(mpa3_raw))

# filter to have only species
mpa3_sp <- mpa3_raw %>% 
  filter(str_detect(clade_name, "s__")) %>%
  separate(clade_name, into = c("K","P","C","O","F","G","Species"), sep = "\\|") %>%
  select(-c("K","P","C","O","F","G")) %>%
  mutate(Species = str_replace_all(Species, "s__",""),
         Species = str_replace_all(Species, "_"," ")) %>%
  # these samples have no metadata
  select(-c("MID025.A0101","MID033.A0101","MID052.A0101","MID056.A0101","MID065.A0101","MID068.A0101","MID076.A0101","MID078.A0101")) %>%
  arrange(Species)

```


Now decontam the tables
```{r}
mpa3_contaminants <- fread("./05-results.backup/contaminant_species_mid_mpa3.tsv", sep = "\t") %>%
  bind_rows(., fread("./05-results.backup/contaminant_species_kilteasheen_mpa3.tsv", sep = "\t")) %>%
  bind_rows(., fread("./05-results.backup/contaminant_species_iberia_mpa3.tsv", sep = "\t")) %>%
  unique()

# remove contaminant species from table
mpa3_species_decontam <- mpa3_sp %>%
  anti_join(., mpa3_contaminants)

```

```{r load_strep_groups}
strep_groups <- fread("./00-documentation.backup/25-streptococcus_cladegroup_amylasegroup_database.tsv") %>%
  select(Taxon, Clade_Consensus) %>%
  rename(Species = Taxon) %>%
  mutate(Species = str_replace_all(Species, "_", " ")) %>%
  rename(Strep_group = Clade_Consensus) %>%
  mutate(Strep_group = fct_relevel(Strep_group, "sanguinis","mitis","salivarius","anginosus","bovis","pyogenic","mutans","unknown","NA"))

# get the proportion of sanguinis and anginosis for each sample
strep_group_info <- mpa3_species_decontam %>%
  filter(str_detect(Species, "Streptococcus")) %>%
  # join with the Strep grup table for plotting
  inner_join(., strep_groups) %>%
  select(-Species) %>%
  select(Strep_group, everything()) %>%
  # transpose
  gather("Library_ID","Counts", 2:ncol(.)) %>%
  group_by(Library_ID, Strep_group) %>%
  summarize(Reads = sum(Counts)) %>%
  spread(Strep_group, Reads) %>%
  # convert table to proportions
  adorn_percentages(denominator = "row") %>%
  # arrange by sanguinis and for the ones w/o sanguinis, by anginosis
  arrange(desc(sanguinis))

```

```{r load_metadata}

# load the metadata file
load("./05-results.backup/Metadata_tables.RData")

# read in the MALT mapped/unmapped gc and read length info from `MID_malt_map_unmap_rl_gc.Rmd`
malt_map_unmap_rl_gc <- fread("./05-results.backup/rl_gc_sub_data.tsv")
colnames(malt_map_unmap_rl_gc) <- gsub("_map","_malt_map", colnames(malt_map_unmap_rl_gc))
colnames(malt_map_unmap_rl_gc) <- gsub("_unmap","_malt_unmap", colnames(malt_map_unmap_rl_gc))

# full_metadata <- full_metadata %>%
#   mutate(`DNA_molecules_per_library_x10^6_log` = log10(`DNA_molecules_per_library_x10^6`)) %>%
#   full_join(., malt_map_unmap_rl_gc) %>%
#   full_join(., strep_groups, by = "Library_ID")

```



```{r pca_name_fxn}
# this function plots the PCA with the sample names instead of points
plot_pca_names <- function(df, pc1, pc2, metadata_group) {
    metadata_group_colors <- get(paste(metadata_group, "_colors", sep = ""))

    exp_var <- paste0(round(df$prop_expl_var$X * 100, 2), "%")
    df_X <- df$variates$X %>%
              as.data.frame() %>%
              rownames_to_column("Library_ID") %>%
              inner_join(full_metadata %>%
                         select(Library_ID, Site_code, Region, Time_period, Sample_or_Control, Periodontal_disease_present, Caries_present, Sex, Age, Tooth_location), by = "Library_ID") %>%
              inner_join(., strep_group_info)
    
    metadata_group = df_X[[metadata_group]]
    
    if (pc1 == 'PC1') {
        pc1 <- df_X$PC1
        exp_var_pc1 <- exp_var[1]
        xaxis <- c("PC1")
    }  else if (pc1 == 'PC2') {
        pc1 <- df_X$PC2
        exp_var_pc1 <- exp_var[2]
        xaxis <- c("PC2")
    } else if (pc1 == 'PC3') {
        pc1 <- df_X$PC3
        exp_var_pc1 <- exp_var[3]
        xaxis <- c("PC3")
    }

    if (pc2 == 'PC1') {
        pc2 <- df_X$PC1
        exp_var_pc2 <- exp_var[1]
        yaxis <- c("PC1")
    }  else if (pc2 == 'PC2') {
        pc2 <- df_X$PC2
        exp_var_pc2 <- exp_var[2]
        yaxis <- c("PC2")
    } else if (pc2 == 'PC3') {
        pc2 <- df_X$PC3
        exp_var_pc2 <- exp_var[3]
        yaxis <- c("PC3")
    }

    pca_plot_names <- ggplot(df_X, aes(pc1, pc2, colour = metadata_group)) +
     geom_text(aes(label = df_X$Library_ID), size = 4) +
     scale_colour_manual(values = metadata_group_colors) +
     # stat_ellipse() +
     xlab(paste(xaxis, " - ", exp_var_pc1)) +
     ylab(paste(yaxis, " - ", exp_var_pc2)) +
     theme_minimal(base_size = 16) +
     theme(text = element_text(size=16)) +
     theme(legend.position = "top") 
    # +
    #                     theme(legend.title = element_blank())

    return(pca_plot_names)
}

```

```{r pca_plot_fxn}
# plot PCA with colored dots and the title including the # of species or genera
plot_pca <- function(df, pc1, pc2, color_group, shape_group, ncomps) {
    metadata_group_colors <- get(paste(color_group, "_colors", sep = ""))
    metadata_group_shapes <- get(paste(shape_group, "_shapes", sep = ""))

    pca.list <- mixOmics::pca(df, ncomp = ncomps, logratio = 'CLR')

    exp_var <- paste0(round(pca.list$prop_expl_var$X * 100, 2), "%")
    df_X <- pca.list$variates$X %>%
              as.data.frame() %>%
              rownames_to_column("Library_ID") %>%
              inner_join(full_metadata, by = "Library_ID") %>%
              inner_join(., strep_group_info)
              
    color_group = df_X[[color_group]]
    shape_group = df_X[[shape_group]]

    if (pc1 == 'PC1') {
        pc1 <- df_X$PC1
        exp_var_pc1 <- exp_var[1]
        xaxis <- c("PC1")
    }  else if (pc1 == 'PC2') {
        pc1 <- df_X$PC2
        exp_var_pc1 <- exp_var[2]
        xaxis <- c("PC2")
    } else if (pc1 == 'PC3') {
        pc1 <- df_X$PC3
        exp_var_pc1 <- exp_var[3]
        xaxis <- c("PC3")
    }

    if (pc2 == 'PC1') {
        pc2 <- df_X$PC1
        exp_var_pc2 <- exp_var[1]
        yaxis <- c("PC1")
    }  else if (pc2 == 'PC2') {
        pc2 <- df_X$PC2
        exp_var_pc2 <- exp_var[2]
        yaxis <- c("PC2")
    } else if (pc2 == 'PC3') {
        pc2 <- df_X$PC3
        exp_var_pc2 <- exp_var[3]
        yaxis <- c("PC3")
    }

    pca_plot <- ggplot(df_X, aes(pc1, pc2, colour = color_group, shape = shape_group)) +
     geom_point(size = 2) +
     scale_colour_manual(values = metadata_group_colors) +
     scale_shape_manual(values = metadata_group_shapes) +
     # stat_ellipse() +
     xlab(paste(xaxis, " - ", exp_var_pc1)) +
     ylab(paste(yaxis, " - ", exp_var_pc2)) +
     theme_minimal(base_size = 16) +
     theme(text = element_text(size=16)) +
     theme(legend.title = element_blank(),
           legend.key.size = unit(2,"mm"),
           legend.text = element_text(size = 6)) +
     theme(legend.position = "top")

    return(pca_plot)
}

```

```{r pca_plot_mid_fxn}
# plot PCA with colored dots 
plot_pca_mid <- function(df, pc1, pc2, color_group, shape_group, ncomps, title_text) {

    pca.list <- mixOmics::pca(df, ncomp = ncomps, logratio = 'CLR')

    exp_var <- paste0(round(pca.list$prop_expl_var$X * 100, 2), "%")
    df_X <- pca.list$variates$X %>%
              as.data.frame() %>%
              rownames_to_column("Library_ID") %>%
              inner_join(full_metadata, by = "Library_ID")
    
    color_group = df_X[[color_group]]
    shape_group = df_X[[shape_group]]

    if (pc1 == 'PC1') {
        pc1 <- df_X$PC1
        exp_var_pc1 <- exp_var[1]
        xaxis <- c("PC1")
    }  else if (pc1 == 'PC2') {
        pc1 <- df_X$PC2
        exp_var_pc1 <- exp_var[2]
        xaxis <- c("PC2")
    } else if (pc1 == 'PC3') {
        pc1 <- df_X$PC3
        exp_var_pc1 <- exp_var[3]
        xaxis <- c("PC3")
    }

    if (pc2 == 'PC1') {
        pc2 <- df_X$PC1
        exp_var_pc2 <- exp_var[1]
        yaxis <- c("PC1")
    }  else if (pc2 == 'PC2') {
        pc2 <- df_X$PC2
        exp_var_pc2 <- exp_var[2]
        yaxis <- c("PC2")
    } else if (pc2 == 'PC3') {
        pc2 <- df_X$PC3
        exp_var_pc2 <- exp_var[3]
        yaxis <- c("PC3")
    }

    pca_plot <- ggplot(df_X, aes(pc1, pc2, colour = color_group, shape = shape_group)) +
     geom_point(size = 3) +
     scale_colour_manual(values = c("#5A167E","#E95562","#FED395","#252525","#C83E73","#311068","#FEA873")) + # 
     scale_shape_manual(values = c(17, 19, 15, 18)) +
     # stat_ellipse() +
     xlab(paste(xaxis, " - ", exp_var_pc1)) +
     ylab(paste(yaxis, " - ", exp_var_pc2)) +
     theme_minimal(base_size = 16) +
     theme(text = element_text(size=16)) +
     theme(legend.title = element_blank(),
           legend.key.size = unit(2,"mm"),
           legend.text = element_text(size = 6)) + # 
     theme(legend.position = "top") +
     ggtitle(title_text) + theme(plot.title = element_text(size = 10))

    return(pca_plot)
}

```

```{r pca_plot_cont_fxn}
# for continuous data
plot_pca_cont <- function(df, pc1, pc2, color_group, shape_group, ncomps, title_text) {

    pca.list <- mixOmics::pca(df, ncomp = ncomps, logratio = 'CLR')

    exp_var <- paste0(round(pca.list$prop_expl_var$X * 100, 2), "%")
    df_X <- pca.list$variates$X %>%
              as.data.frame() %>%
              rownames_to_column("Library_ID") %>%
              inner_join(full_metadata, by = "Library_ID") %>%
              inner_join(., strep_group_info, by = "Library_ID")
    # %>%
              # full_join(., full_source_results %>%
              #             rename(Library_ID = SampleID),  by = "Library_ID")
    
    color_group = df_X[[color_group]]
    shape_group = df_X[[shape_group]]

    if (pc1 == 'PC1') {
        pc1 <- df_X$PC1
        exp_var_pc1 <- exp_var[1]
        xaxis <- c("PC1")
    }  else if (pc1 == 'PC2') {
        pc1 <- df_X$PC2
        exp_var_pc1 <- exp_var[2]
        xaxis <- c("PC2")
    } else if (pc1 == 'PC3') {
        pc1 <- df_X$PC3
        exp_var_pc1 <- exp_var[3]
        xaxis <- c("PC3")
    }

    if (pc2 == 'PC1') {
        pc2 <- df_X$PC1
        exp_var_pc2 <- exp_var[1]
        yaxis <- c("PC1")
    }  else if (pc2 == 'PC2') {
        pc2 <- df_X$PC2
        exp_var_pc2 <- exp_var[2]
        yaxis <- c("PC2")
    } else if (pc2 == 'PC3') {
        pc2 <- df_X$PC3
        exp_var_pc2 <- exp_var[3]
        yaxis <- c("PC3")
    }

    pca_plot <- ggplot(df_X, aes(pc1, pc2, colour = color_group, shape = shape_group)) +
     geom_point(size = 2) +
     scale_colour_viridis_c(option = "C") +
     scale_shape_manual(values = c(16, 17, 15, 18, 14,9,13,24)) +
     # stat_ellipse() +
     xlab(paste(xaxis, " - ", exp_var_pc1)) +
     ylab(paste(yaxis, " - ", exp_var_pc2)) +
     theme_minimal(base_size = 16) +
     theme(text = element_text(size=16)) +
     theme(legend.title = element_blank(),
           legend.key.size = unit(2,"mm"),
           legend.text = element_text(size = 6)) + 
     theme(legend.position = "top") +
     ggtitle(title_text) + theme(plot.title = element_text(size = 10))

    return(pca_plot)
}


```

```{r pca_biplot_fxn}
plot_pca_bi <- function(df, pc1, pc2, metadata_group, columntitle) {
    metadata_group_colors <- get(paste(metadata_group, "_colors", sep = ""))
    metadata_group_shapes <- get(paste(metadata_group, "_shapes", sep = ""))
   
    arrow_pc <- enquo(columntitle)
    
    exp_var <- paste0(round(df$prop_expl_var$X * 100, 2), "%") # explained variance for x- and y-labels
    
    # select only the PCs from the PCA and add metadata
    df_X <- df$variates$X %>%
              as.data.frame() %>%
              rownames_to_column("Library_ID") %>%
              inner_join(full_metadata, by = "Library_ID")
    
    metadata_group = df_X[[metadata_group]]
    
    corr_lam <- df$sdev[c("PC1", "PC2", "PC3")] * sqrt(nrow(df_X))

    df_X <- df_X %>%
      mutate(PC1 = PC1 / corr_lam[1],
             PC2 = PC2 / corr_lam[2], 
             PC3 = PC3 / corr_lam[3])

    # select the correct PC column and explained variance for PC1
    if (pc1 == 'PC1') {
        Pc1 <- df_X$PC1
        exp_var_pc1 <- exp_var[1]
        xaxis <- c("PC1")
    } else if (pc1 == 'PC2') {
        Pc1 <- df_X$PC2
        exp_var_pc1 <- exp_var[2]
        xaxis <- c("PC2")
    } else if (pc1 == 'PC3') {
        Pc1 <- df_X$PC3
        exp_var_pc1 <- exp_var[3]
        xaxis <- c("PC3")
   }
    
    # select the correct PC column and explained variance for PC2
    if (pc2 == 'PC1') {
        Pc2 <- df_X$PC1
        exp_var_pc2 <- exp_var[1]
        yaxis <- c("PC1")
 } else if (pc2 == 'PC2') {
       Pc2 <- df_X$PC2
       exp_var_pc2 <- exp_var[2]
       yaxis <- c("PC2")
    } else if (pc2 == 'PC3') {
       Pc2 <- df_X$PC3
       exp_var_pc2 <- exp_var[3]
       yaxis <- c("PC3")
   }
    
    # Identify the 10 pathways that have highest positive and negative loadings in the selected PC
    pws_10 <- df$loadings$X %>%
      as.data.frame(.) %>%
      rownames_to_column(var = "Taxon") %>%
      top_n(10, !!arrow_pc)

    neg_10 <- df$loadings$X %>%
      as.data.frame(.) %>%
      rownames_to_column(var = "Taxon") %>%
      top_n(-10, !!arrow_pc)


    pca_plot_bi <- ggplot(df_X, aes(x = Pc1, y = Pc2, color = metadata_group)) +
      geom_point(size = 2.5, aes(shape = metadata_group) )+
      geom_segment(data = pws_10,
                   aes(xend = get(paste(pc1)), yend = get(paste(pc2))),
                   x = 0, y = 0, colour = "black",
                   size = 0.5,
                   arrow = arrow(length = unit(0.03, "npc"))) +
      geom_label_repel(data = pws_10,
                   aes(x = get(paste(pc1)), y = get(paste(pc2)), label = Taxon),
                   size = 2.5, colour = "grey20", label.padding = 0.2, force = 5) +
      geom_segment(data = neg_10,
                   aes(xend = get(paste(pc1)), yend = get(paste(pc2))),
                   x = 0, y = 0, colour = "grey50",
                   size = 0.5,
                   arrow = arrow(length = unit(0.03, "npc"))) +
      geom_label_repel(data = neg_10,
                   aes(x = get(paste(pc1)), y = get(paste(pc2)), label = Taxon),
                   size = 2.5, colour = "grey20", label.padding = 0.2) +
      labs(x = paste(xaxis, " - ", exp_var_pc1),
           y = paste(yaxis, " - ", exp_var_pc2)) +
      scale_color_manual(values = metadata_group_colors) +
      scale_shape_manual(values = metadata_group_shapes) +
      theme_minimal() + theme(text = element_text(size = 16)) +
      theme(text = element_text(size=16)) +
      theme(legend.position = "top")

    return(pca_plot_bi)
}

# pca_test_biplot <- plot_pca_bi(malt_species.pca, "PC3", "PC2", "Village", PC2)
# pca_test_biplot


```

```{r pca_lm_biplot_fxn}
plot_pca_bi_lm <- function(df, metadata_table, pc1, pc2, metadata_group, columntitle) {
    metadata_group_colors <- get(paste(metadata_group, "_colors", sep = ""))
    metadata_group_shapes <- get(paste(metadata_group, "_shapes", sep = ""))
   
    arrow_pc <- enquo(columntitle)
    
    exp_var <- paste0(round(df$prop_expl_var$X * 100, 2), "%") # explained variance for x- and y-labels
    
    # select only the PCs from the PCA and add metadata
    df_X <- df$variates$X %>%
              as.data.frame() %>%
              rownames_to_column("Library_ID") %>%
              inner_join(metadata_table, by = "Library_ID")
    
    metadata_group = df_X[[metadata_group]]
    
    corr_lam <- df$sdev[c("PC1", "PC2", "PC3")] * sqrt(nrow(df_X))

    df_X <- df_X %>%
      mutate(PC1 = PC1 / corr_lam[1],
             PC2 = PC2 / corr_lam[2], 
             PC3 = PC3 / corr_lam[3])

    # select the correct PC column and explained variance for PC1
    if (pc1 == 'PC1') {
        Pc1 <- df_X$PC1
        exp_var_pc1 <- exp_var[1]
        xaxis <- c("PC1")
    } else if (pc1 == 'PC2') {
        Pc1 <- df_X$PC2
        exp_var_pc1 <- exp_var[2]
        xaxis <- c("PC2")
    } else if (pc1 == 'PC3') {
        Pc1 <- df_X$PC3
        exp_var_pc1 <- exp_var[3]
        xaxis <- c("PC3")
   }
    
    # select the correct PC column and explained variance for PC2
    if (pc2 == 'PC1') {
        Pc2 <- df_X$PC1
        exp_var_pc2 <- exp_var[1]
        yaxis <- c("PC1")
 } else if (pc2 == 'PC2') {
       Pc2 <- df_X$PC2
       exp_var_pc2 <- exp_var[2]
       yaxis <- c("PC2")
    } else if (pc2 == 'PC3') {
       Pc2 <- df_X$PC3
       exp_var_pc2 <- exp_var[3]
       yaxis <- c("PC3")
   }
    
    # Identify the 10 pathways that have highest positive and negative loadings in the selected PC
    pws_10 <- df$loadings$X %>%
      as.data.frame(.) %>%
      rownames_to_column(var = "Taxon") %>%
      top_n(10, !!arrow_pc)

    neg_10 <- df$loadings$X %>%
      as.data.frame(.) %>%
      rownames_to_column(var = "Taxon") %>%
      top_n(-10, !!arrow_pc)


    pca_plot_bi <- ggplot(df_X, aes(x = Pc1, y = Pc2, colour = metadata_group)) +
      geom_point(size = 2.5, aes(shape = metadata_group) )+
      geom_segment(data = pws_10,
                   aes(xend = get(paste(pc1)), yend = get(paste(pc2))),
                   x = 0, y = 0, colour = "black",
                   size = 0.5,
                   arrow = arrow(length = unit(0.03, "npc"))) +
      geom_label_repel(data = pws_10,
                   aes(x = get(paste(pc1)), y = get(paste(pc2)), label = Taxon),
                   size = 2.5, colour = "grey20", label.padding = 0.2) +
      geom_segment(data = neg_10,
                   aes(xend = get(paste(pc1)), yend = get(paste(pc2))),
                   x = 0, y = 0, colour = "grey50",
                   size = 0.5,
                   arrow = arrow(length = unit(0.03, "npc"))) +
      geom_label_repel(data = neg_10,
                   aes(x = get(paste(pc1)), y = get(paste(pc2)), label = Taxon),
                   size = 2.5, colour = "grey20", label.padding = 0.2) +
      labs(x = paste(xaxis, " - ", exp_var_pc1),
           y = paste(yaxis, " - ", exp_var_pc2)) +
      scale_colour_manual(values = metadata_group_colors) +
      scale_shape_manual(values = metadata_group_shapes) +
      theme_minimal() + theme(text = element_text(size = 16)) +
      theme(text = element_text(size=16)) +
      theme(legend.position = "top")

    return(pca_plot_bi)
}

# pca_test_biplot <- plot_pca_bi(malt_species.pca, "PC3", "PC2", "Village", PC2)
# pca_test_biplot


```


List the outliers from the plot in `MID_CMB_Rad_Kil_Iber_decontam.Rmd`. 
```{r outliers}
outliers_mpa3 <- c("EXB059.A2101","EXB059.A2501","EXB015.A3301","EXB034.A2701","EXB059.A2201","EXB059.A2301","EXB059.A2401","LIB058.A0103","LIB058.A0106","LIB058.A0104")
poor_samples_mpa3 <- fread("./05-results.backup/cuperdec_mpa3_poor_samples.tsv") %>%
  pull(Sample)

outliersF <- str_c(outliers_mpa3, collapse = "|")

```


#Sample Clustering with PCA

Step-by-step removal of taxa based on decontam and a filter threshold, followed
by PCA to see how contaminants/low abundance species affect the relationships of
samples to eachother between groups.
```{r pca_all_species}
# decontaminated only
mpa3_species_decontam_unfilt <- mpa3_species_decontam %>%
  select(-one_of(outliers_mpa3)) %>%
  select(-one_of(poor_samples_mpa3)) %>%
  select(-matches("ARS|CSS|CSL")) %>%
  gather("Library_ID","Counts",2:ncol(.)) %>%
  mutate(Counts = Counts +1) %>%
  spread(Species,Counts) %>%
  column_to_rownames("Library_ID")

# prepare to run a PCA
# check the number of components to retain by tuning the PCA
tune.pca(mpa3_species_decontam_unfilt, logratio = 'CLR')

# perform and plot a PCA to see how the data cluster
pc1pc2 <- plot_pca(mpa3_species_decontam_unfilt, "PC1", "PC2", "Site_code", "Time_period", 3)
pc1pc2

plot_pca(mpa3_species_decontam_unfilt, "PC3", "PC2", "Site_code", "Time_period", 3)


```


# Species-level analyses
What is the relationship of the groups in a PCA without the blanks? Let's try
the PCA using only the species that are present in at least 10% of at least one
sample group (MID, CMB, Radcliffe, Kilteasheen, Iberia)

```{r}

# decontaminated only
mpa3_species_decontam_unfilt <- mpa3_species_decontam %>%
  select(-one_of(outliers_mpa3)) %>%
  select(-one_of(poor_samples_mpa3)) %>%
  select(-matches("EXB|LIB|CSN|CSD|ARS|CSS|CSL")) %>%
  gather("Library_ID","Counts",2:ncol(.)) %>%
  mutate(Counts = Counts +1) %>%
  spread(Species,Counts) %>%
  column_to_rownames("Library_ID")

# prepare to run a PCA
# check the number of components to retain by tuning the PCA
tune.pca(mpa3_species_decontam_unfilt, logratio = 'CLR')

# perform and plot a PCA to see how the data cluster
plot_pca(mpa3_species_decontam_unfilt, "PC1", "PC2", "Site_code", "Time_period", 3)
plot_pca(mpa3_species_decontam_unfilt, "PC3", "PC2", "Site_code", "Time_period", 3)

```

```{r no_methanobrevibacter}
# remove Methanobrevibacter oralis from the species table
mpa3_species_decontam_unfilt_no_Moralis <- mpa3_species_decontam %>%
  filter(Species != "Methanobrevibacter oralis") %>%
  select(-one_of(outliers_mpa3)) %>%
  select(-one_of(poor_samples_mpa3)) %>%
  select(-matches("EXB|LIB|CSN|CSD|ARS|CSS|CSL")) %>%
  gather("Library_ID","Counts",2:ncol(.)) %>%
  mutate(Counts = Counts +1) %>%
  spread(Species,Counts) %>%
  column_to_rownames("Library_ID")


# perform and plot a PCA to see how the data cluster
nomethano_beta <- plot_pca(mpa3_species_decontam_unfilt_no_Moralis, "PC1", "PC2", "Site_code", "Time_period", 3)
nomethano_beta

# by average GC content
nomethano_gc <- plot_pca_cont(mpa3_species_decontam_unfilt_no_Moralis, "PC1", "PC2", "gc_avg", "Time_period", 3, "Avg. GC content (%)")
nomethano_gc

# by average read length
plot_pca_cont(mpa3_species_decontam_unfilt_no_Moralis, "PC1", "PC2", "seq_len_avg", "Time_period", 3, "Avg. read length (bp)")

# biplot to see strongest loading species
mpa3_species_decontam_unfilt_no_Moralis.pca <- mixOmics::pca(mpa3_species_decontam_unfilt_no_Moralis, ncomp = 3, logratio = 'CLR')
plot(mpa3_species_decontam_unfilt_no_Moralis.pca)

# define shapes
Site_code_shapes <- c(MID = 21, CMB = 21, RAD = 21, KIL = 24, ELR = 24, IVE = 24, JAE = 22, VLC = 22, EXB = 25, LIB = 25)

nomethano_bi <- plot_pca_bi(mpa3_species_decontam_unfilt_no_Moralis.pca, "PC1", "PC2", "Site_code", PC1)
nomethano_bi

no_methano <- plot_grid(nomethano_beta, nomethano_gc, nomethano_bi,
          ncol = 1, labels = c("A", "B","C"), rel_widths = c(1), scale = 0.95)


# ggsave("./06-publication/supplemental_figures/SXX5/no_methano.pdf", plot = no_methano,
#        device = "pdf", scale = 1, width = 6, height = 12, units = c("in"), dpi = 300)


```

```{r color_by_Moralis}

mpa3_species_decontam_unfilt <- mpa3_species_decontam %>%
  select(-one_of(outliers_mpa3)) %>%
  select(-one_of(poor_samples_mpa3)) %>%
  select(-matches("EXB|LIB|CSN|CSD|ARS|CSS|CSL")) %>%
  gather("Library_ID","Counts",2:ncol(.)) %>%
  mutate(Counts = Counts +1) %>%
  spread(Species,Counts) %>%
  column_to_rownames("Library_ID")

# prepare to run a PCA

mpa3_all_otu.pca <- mixOmics::pca(mpa3_species_decontam_unfilt, ncomp = 2, logratio = 'CLR')
mpa3_all_pca_values_wMo <- mpa3_all_otu.pca$variates$X %>%
  as.data.frame() %>%
  rownames_to_column("Library_ID") %>%
  inner_join(., full_metadata, by = "Library_ID") %>%
  inner_join(., mpa3_species_decontam_unfilt %>%
               rownames_to_column("Library_ID") %>%
               select(Library_ID, `Methanobrevibacter oralis`) %>%
               mutate(`Methanobrevibacter oralis` = `Methanobrevibacter oralis` - 1))

ggplot(mpa3_all_pca_values_wMo, aes(PC1, PC2, color = `Methanobrevibacter oralis`, shape = Time_period)) +
  geom_point(size = 2) +
  # geom_text(aes(label = Library_ID), size = 3) +
  theme_minimal() +
  theme(legend.position = "top")


 mpa3_species_decontam_unfilt %>%
   rownames_to_column("Library_ID") %>%
   pivot_longer(!Library_ID, names_to = "Species", values_to = "Counts") %>%
   mutate(Counts = Counts - 1) %>%
   pivot_wider(names_from = "Library_ID", values_from = "Counts") %>%
   select(Species, CS03) %>%
   arrange(desc(CS03))
 
 
```


```{r biplots_all_species}

mpa3_species_decontam_unfilt <- mpa3_species_decontam %>%
  select(-one_of(outliers_mpa3)) %>%
  select(-one_of(poor_samples_mpa3)) %>%
  select(-matches("EXB|LIB|CSN|CSD|ARS|CSS|CSL")) %>%
  gather("Library_ID","Counts",2:ncol(.)) %>%
  mutate(Counts = Counts +1) %>%
  spread(Species,Counts) %>%
  column_to_rownames("Library_ID")

mpa3_species_decontam_unfilt.pca <- mixOmics::pca(mpa3_species_decontam_unfilt, ncomp = 3, logratio = 'CLR')
plot(mpa3_species_decontam_unfilt.pca)

# Define pipenotch colors for plotting
Pipenotch_colors = c("#311068","#C83E73")
Pipenotch_shapes = c(19,17)

mpa3_species_decontam_filtered_nocontrols_biplot_121 <- plot_pca_bi(mpa3_species_decontam_unfilt.pca, "PC1", "PC2", "Site_code", PC1)
mpa3_species_decontam_filtered_nocontrols_biplot_121

mpa3_species_decontam_filtered_nocontrols_biplot_122 <- plot_pca_bi(mpa3_species_decontam_unfilt.pca, "PC1", "PC2", "Site_code", PC2)
mpa3_species_decontam_filtered_nocontrols_biplot_122

pandoc.table(mpa3_species_decontam_filtered_nocontrols_biplot_121$plot_env$pws_10 %>% arrange(desc(PC1)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC1 positive loadings")

pandoc.table(mpa3_species_decontam_filtered_nocontrols_biplot_121$plot_env$neg_10 %>% arrange(PC1),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC1 negative loadings")


pandoc.table(mpa3_species_decontam_filtered_nocontrols_biplot_122$plot_env$pws_10 %>% arrange(desc(PC2)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC2 positive loadings")

pandoc.table(mpa3_species_decontam_filtered_nocontrols_biplot_122$plot_env$neg_10 %>% arrange(PC2),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC2 negative loadings")


mpa3_species_decontam_filtered_nocontrols_biplot_322 <- plot_pca_bi(mpa3_species_decontam_unfilt.pca, "PC3", "PC2", "Site_code", PC2)
mpa3_species_decontam_filtered_nocontrols_biplot_322

mpa3_species_decontam_filtered_nocontrols_biplot_323 <- plot_pca_bi(mpa3_species_decontam_unfilt.pca, "PC3", "PC2", "Site_code", PC3)
mpa3_species_decontam_filtered_nocontrols_biplot_323

pandoc.table(mpa3_species_decontam_filtered_nocontrols_biplot_323$plot_env$pws_10 %>% arrange(desc(PC3)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC3 positive loadings")

pandoc.table(mpa3_species_decontam_filtered_nocontrols_biplot_323$plot_env$neg_10 %>% arrange(PC3),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC3 negative loadings")

```

```{r pca_species_mid_pc12}
mpa3_species_decontam_unfilt <- mpa3_species_decontam %>%
  select(-one_of(outliers_mpa3)) %>%
  select(-one_of(poor_samples_mpa3)) %>%
  select(-matches("EXB|LIB|CSN|CSD|ARS|CSS|CSL")) %>%
  adorn_totals(where = "col", name = "Sum_species.decontam") %>%
  mutate(Percent = Sum_species.decontam/sum(Sum_species.decontam)*100) %>%
  filter(Percent > 0.001) %>%
  select(-one_of(c("Sum_species.decontam","Percent"))) %>%
  gather("Library_ID","Counts",2:ncol(.)) %>%
  mutate(Counts = Counts +1) %>%
  spread(Species,Counts) %>%
  column_to_rownames("Library_ID")

# prepare to run a PCA
# check the number of components to retain by tuning the PCA
mixOmics::tune.pca(mpa3_species_decontam_unfilt, logratio = 'CLR')


mpa3_all_otu.pca <- mixOmics::pca(mpa3_species_decontam_unfilt, ncomp = 2, logratio = 'CLR')
mpa3_all_pca_values <- mpa3_all_otu.pca$variates$X %>%
  as.data.frame() %>%
  rownames_to_column("Library_ID") %>%
  inner_join(., full_metadata, by = "Library_ID") %>%
  left_join(., strep_group_info, by = "Library_ID")

# ggplot(mpa3_all_pca_values, aes(PC1, PC2, color = Site_code, shape = Time_period)) +
#   geom_point(size = 2) +
#   # geom_text(aes(label = Library_ID), size = 3) +
#   theme_minimal() +
#   scale_color_manual(values = Site_code_colors) +
#      theme(legend.position = "top")

plot_pca(mpa3_species_decontam_unfilt, "PC1", "PC2","Time_period","Time_period", 3) 
site_code_beta <- plot_pca(mpa3_species_decontam_unfilt, "PC1", "PC2","Site_code","Time_period", 3) 
site_code_beta

# ggsave("./06-publication/supplemental_figures/SXX8/site_code_beta.pdf", plot = site_code_beta, device = "pdf",
#        scale = 1, width = 6, height = 5, units = c("in"), dpi = 300)


# check if the Radliffe samples separate by characteristics of the calculus deposits, Deposit_density, Deposit_location, Deposit_spread
ggplot(mpa3_all_pca_values, aes(PC1, PC2, color = Deposit_spread, shape = Region)) +
  geom_point(size = 2) +
  # geom_text(aes(label = Library_ID), size = 3) +
  theme_minimal()  +
  theme(legend.position = "top")

# perform and plot a PCA to see how the data cluster
# by sequencing depth 
mid_seqdepth_pc12 <- plot_pca_cont(mpa3_species_decontam_unfilt, "PC1", "PC2", "No_reads_total", "Site_code", 3, "Total reads into MetaPhlAn3")
mid_seqdepth_pc12

# by average GC content
plot_pca_cont(mpa3_species_decontam_unfilt, "PC1", "PC2", "gc_avg", "Time_period", 3, "Avg. GC content (%)")

# by average read length
plot_pca_cont(mpa3_species_decontam_unfilt, "PC1", "PC2", "seq_len_avg", "Time_period", 3, "Avg. read length (bp)")

# by Strep group proportion - sanguinis
plot_pca_cont(mpa3_species_decontam_unfilt, "PC1", "PC2", "sanguinis", "Time_period", 3, "Proportion of Sanguinis group")

# by Strep group proportion - mitis
plot_pca_cont(mpa3_species_decontam_unfilt, "PC1", "PC2", "mitis", "Time_period", 3, "Proportion of Sanguinis group")

# by average read length
rl <- plot_pca_cont(mpa3_species_decontam_unfilt, "PC1", "PC2", "seq_len_avg", "Time_period", 3, "Avg. read length (bp)")
rl

```

```{r}
# load the MALT RefSeq taxonomy tables that have been decontaminated 
# and the column headers are cleaned up
# this is to get the Strep groups for coloring the PCA plot
load("./05-results.backup/Taxonomy_tables.RData")

# list poor samples from cuperdec
poor_samples_rs <- fread("./05-results.backup/cuperdec_poor_samples.tsv") %>%
  rename(Library_ID = Sample) %>%
  filter(str_detect(Library_ID, "MID")) %>%
  pull(Library_ID)

# we'll use the prevalence-filtered dataset
malt_rs <- all_species_decontam_prev %>%
  # select(matches("Species|MID|CMB")) %>%
  adorn_totals(where = "col")  %>%
  filter(Total >  0) %>%
  select(-Total) %>%
  select(-poor_samples_rs)

strep_groups <- fread("./00-documentation.backup/dRep_strep_groups.tsv") %>%
  # remove the extrac column with the old dRep
  select(-Clade_Consensus) %>%
  # make a factor for plotting
  mutate(dRep = str_replace_all(dRep, "downei", "Downei"),
         dRep = fct_relevel(dRep, "Sanguinis","Mitis","Salivarius","Anginosus","Bovis","Pyogenic","Mutans","Other_oral","Other","Unknown"))

strep_rs <- malt_rs %>%
  filter(str_detect(Species, "Streptococcus")) %>%
  # join with the Strep grup table for plotting
  inner_join(., strep_groups) %>%
  select(Species, dRep, everything()) %>%
  # convert to long-format
  gather("Library_ID","Counts", 3:ncol(.)) %>%
  # mutate(Library_ID = fct_relevel(Library_ID, sample_order_rs)) %>%
  dplyr::group_by(Library_ID, dRep) %>%
  summarize(Count_sum = sum(Counts)) %>%
  ungroup() %>%
  group_by(Library_ID) %>%
  mutate(Percent = Count_sum / sum(Count_sum)) %>%
  select(Library_ID, dRep, Percent) %>%
  pivot_wider(names_from = dRep, values_from = Percent)

mpa3_all_otu.pca <- mixOmics::pca(mpa3_species_decontam_unfilt, ncomp = 2, logratio = 'CLR')
mpa3_all_pca_values_rs_strep <- mpa3_all_otu.pca$variates$X %>%
  as.data.frame() %>%
  rownames_to_column("Library_ID") %>%
  inner_join(., full_metadata, by = "Library_ID") %>%
  left_join(., strep_rs, by = "Library_ID")

ggplot(mpa3_all_pca_values_rs_strep, aes(PC1, PC2, fill = Sanguinis, shape = Time_period)) +
  geom_point(size = 2, color = "black") +
     scale_fill_viridis_c(option = "C") +
     scale_shape_manual(values = c(21, 24, 22, 23, 23)) +
     # stat_ellipse() +
     # xlab(paste(xaxis, " - ", exp_var_pc1)) +
     # ylab(paste(yaxis, " - ", exp_var_pc2)) +
     theme_minimal(base_size = 16) +
     theme(text = element_text(size=16)) +
     theme(legend.title = element_blank(),
           legend.key.size = unit(2,"mm"),
           legend.text = element_text(size = 6)) + 
     theme(legend.position = "top") +
     ggtitle("") + theme(plot.title = element_text(size = 10))

ggplot(mpa3_all_pca_values_rs_strep, aes(PC1, PC2, fill = Anginosus, shape = Time_period)) +
  geom_point(size = 2, color = "black") +
     scale_fill_viridis_c(option = "C") +
     scale_shape_manual(values = c(21, 24, 22, 23, 23)) +
     # stat_ellipse() +
     # xlab(paste(xaxis, " - ", exp_var_pc1)) +
     # ylab(paste(yaxis, " - ", exp_var_pc2)) +
     theme_minimal(base_size = 16) +
     theme(text = element_text(size=16)) +
     theme(legend.title = element_blank(),
           legend.key.size = unit(2,"mm"),
           legend.text = element_text(size = 6)) + 
     theme(legend.position = "top") +
     ggtitle("") + theme(plot.title = element_text(size = 10))

ggplot(mpa3_all_pca_values_rs_strep, aes(PC1, PC2, fill = Anginosus, shape = Site_code)) +
  geom_point(size = 2, color = "black") +
     scale_fill_viridis_c(option = "C") +
     scale_shape_manual(values = c(21, 24, 22, 23, 25,21, 24, 22, 23, 25)) +
     # stat_ellipse() +
     # xlab(paste(xaxis, " - ", exp_var_pc1)) +
     # ylab(paste(yaxis, " - ", exp_var_pc2)) +
     theme_minimal(base_size = 16) +
     theme(text = element_text(size=16)) +
     theme(legend.title = element_blank(),
           legend.key.size = unit(2,"mm"),
           legend.text = element_text(size = 6)) + 
     theme(legend.position = "top") +
     ggtitle("") + theme(plot.title = element_text(size = 10))



```

```{r}
# select the species with strongest loadings in PC1 + and - directions
pc1_mpa3 <- fread("./05-results.backup/mpa3_pc1_species_list.tsv") 

pc1pws <- pc1_mpa3 %>%
  filter(PC1 > 0) %>%
  pull(Taxon)

pc1neg <- pc1_mpa3 %>%
  filter(PC1 < 0) %>%
  pull(Taxon)

pc_species_pws <- mpa3_species_decontam %>%
  select(-one_of(outliers_mpa3)) %>%
  select(-one_of(poor_samples_mpa3)) %>%
  select(-matches("EXB|LIB|CSN|CSD|ARS|CSS|CSL")) %>%
  filter(Species %in% pc1pws) %>%
  # convert to long-format
  pivot_longer(!Species, names_to = "Library_ID", values_to = "Counts") %>%
  dplyr::group_by(Library_ID) %>%
  summarize(Count_sum = sum(Counts)) %>%
  ungroup() %>%
  full_join(., mpa3_species_decontam %>%
  select(-one_of(outliers_mpa3)) %>%
  select(-one_of(poor_samples_mpa3)) %>%
  select(-matches("EXB|LIB|CSN|CSD|ARS|CSS|CSL")) %>%
              pivot_longer(!Species, names_to = "Library_ID", values_to = "Counts") %>%
              dplyr::group_by(Library_ID) %>%
              summarize(Total = sum(Counts)) %>%
              select(Library_ID, Total)) %>%
  mutate(Percent_pws = Count_sum / Total * 100) %>%
  select(Library_ID, Percent_pws)

pc_species_neg <- mpa3_species_decontam %>%
  select(-one_of(outliers_mpa3)) %>%
  select(-one_of(poor_samples_mpa3)) %>%
  select(-matches("EXB|LIB|CSN|CSD|ARS|CSS|CSL")) %>%
  filter(Species %in% pc1neg) %>%
  # convert to long-format
  pivot_longer(!Species, names_to = "Library_ID", values_to = "Counts") %>%
  dplyr::group_by(Library_ID) %>%
  summarize(Count_sum = sum(Counts)) %>%
  ungroup() %>%
  full_join(., mpa3_species_decontam %>%
  select(-one_of(outliers_mpa3)) %>%
  select(-one_of(poor_samples_mpa3)) %>%
  select(-matches("EXB|LIB|CSN|CSD|ARS|CSS|CSL")) %>%
              pivot_longer(!Species, names_to = "Library_ID", values_to = "Counts") %>%
              dplyr::group_by(Library_ID) %>%
              summarize(Total = sum(Counts)) %>%
              select(Library_ID, Total)) %>%
  mutate(Percent_neg = Count_sum / Total * 100) %>%
  select(Library_ID, Percent_neg)

mpa3_all_pca_values_rs_pc <- mpa3_all_otu.pca$variates$X %>%
  as.data.frame() %>%
  rownames_to_column("Library_ID") %>%
  inner_join(., full_metadata, by = "Library_ID") %>%
  left_join(., pc_species_pws, by = "Library_ID") %>%
  left_join(., pc_species_neg, by = "Library_ID")

pc1pws_plot <- ggplot(mpa3_all_pca_values_rs_pc, aes(PC1, PC2, fill = Percent_pws, shape = Time_period)) +
  geom_point(size = 2, color = "black") +
     scale_fill_viridis_c(option = "C") +
     scale_shape_manual(values = c(21, 24, 22, 23, 23)) +
     # stat_ellipse() +
     # xlab(paste(xaxis, " - ", exp_var_pc1)) +
     # ylab(paste(yaxis, " - ", exp_var_pc2)) +
     theme_minimal(base_size = 16) +
     theme(text = element_text(size=16)) +
     theme(legend.title = element_blank(),
           legend.key.size = unit(2,"mm"),
           legend.text = element_text(size = 6)) + 
     theme(legend.position = "top") +
     ggtitle("") + theme(plot.title = element_text(size = 10))
pc1pws_plot

# ggsave("./06-publication/supplemental_figures/SXX8/pc1pws_plot.pdf", plot = pc1pws_plot,
#        device = "pdf", scale = 1, width = 6, height = 5, units = c("in"), dpi = 300)


ggplot(mpa3_all_pca_values_rs_pc, aes(PC1, PC2, fill = Percent_neg, shape = Time_period)) +
  geom_point(size = 2, color = "black") +
     scale_fill_viridis_c(option = "C") +
     scale_shape_manual(values = c(21, 24, 22, 23, 23)) +
     # stat_ellipse() +
     # xlab(paste(xaxis, " - ", exp_var_pc1)) +
     # ylab(paste(yaxis, " - ", exp_var_pc2)) +
     theme_minimal(base_size = 16) +
     theme(text = element_text(size=16)) +
     theme(legend.title = element_blank(),
           legend.key.size = unit(2,"mm"),
           legend.text = element_text(size = 6)) + 
     theme(legend.position = "top") +
     ggtitle("") + theme(plot.title = element_text(size = 10))



```


```{r, eval = F}
# linear modeling to control for GC content along PC2
# by Ainash

head(mpa3_all_pca_values)

length(mpa3_all_pca_values$Time_period)

# summary(lm(PC2 ~ factor(Time_period) + gc_avg , data = mpa3_all_pca_values))
summary(lm(PC2 ~ gc_avg , data = mpa3_all_pca_values))

model <- lm(PC2 ~ gc_avg , data = mpa3_all_pca_values)

resid(model)

pc2_resid <- as.data.frame(model$residuals) %>%
  rename(pc2_residuals = `model$residuals`)

mpa3_all_pca_values %>%
  bind_cols(., pc2_resid) %>%
ggplot(., aes(PC1, pc2_resid, colour = gc_avg, shape = Site_code, size = Site_code)) +
  geom_point() +
  scale_shape_manual(values = Site_code_shapes) +
  scale_colour_viridis_c(option = "C") +
  # scale_color_manual(values = Site_code_colors) +
  scale_size_manual(values = c(2,2,2,3,2,2)) +
   # stat_ellipse() +
  # xlab(paste(exp_var[1], " - ", exp_var_pc1)) +
  # ylab(paste(exp_var[2], " - ", exp_var_pc2)) +
  theme_minimal(base_size = 16) +
  theme(text = element_text(size=16)) +
  theme(legend.title = element_blank(),
        legend.key.size = unit(2,"mm"),
        legend.text = element_text(size = 6)) +
  theme(legend.position = "top")


```

```{r, eval = F}

# Extract coordinates from mixOmics object and convert into long table dataframe
mpa3_all_pca_values_short <- mpa3_all_pca_values %>%
  select(Library_ID,gc_avg,PC1,PC2,pc2_residuals, Site_code)
# Extract explained variance
pca_expl_vars <- paste0(round(mpa3_all_otu.pca$explained_variance * 100, 2), "%")
# Extract loadings (eigenvectors) from PC analysis
pca_loadings_df <- as.data.frame(mpa3_all_otu.pca$loadings$X) %>%
  rownames_to_column(var = "Species")
# Identify the 10 species that have highest positive and negative loadings in PC1 and PC2
pc1_pws <- pca_loadings_df %>%
  top_n(10, PC1)
pc1_neg <- pca_loadings_df %>%
  top_n(-10, PC1) 
pc2_pws <- pca_loadings_df %>%
  top_n(10, PC2)
pc2_neg <- pca_loadings_df %>%
  top_n(-10, PC2) 
# Calculate correction factor for coordinate systems of principal coordinates
# so that they have the same units as the rotations of PC1 and PC2; from
# ggfortify::autoplot
corr_lam <- mpa3_all_otu.pca$sdev[c("PC1", "PC2")] * sqrt(nrow(mpa3_all_pca_values_short))
# Plot the top 10 negative and positive paths that are strongest loading in PC1 
residuals_pc2_plot_bi <- mpa3_all_pca_values_short %>%
  mutate(PC1 = PC1 / corr_lam[1],
         pc2_residuals = pc2_residuals / corr_lam[2]) %>%
  ggplot(.,
         aes(x = PC1, y = pc2_residuals, colour = gc_avg)) +
  geom_point(size = 2, aes(shape = Site_code)) +
  geom_segment(data = pc1_pws,
               aes(xend = PC1, yend = PC2),
               x = 0, y = 0, colour = "black", size = 0.5,
               arrow = arrow(length = unit(0.03, "npc"))) +
  geom_label_repel(data = pc1_pws,
                   aes(x = PC1, y = PC2, label = Species),
                   size = 2.5, colour = "black", label.padding = 0.2) +
  geom_segment(data = pc1_neg,
               aes(xend = PC1, yend = PC2),
               x = 0, y = 0, colour = "grey50", size = 0.5,
               arrow = arrow(length = unit(0.03, "npc"))) +
  geom_label_repel(data = pc1_neg,
                   aes(x = PC1, y = PC2, label = Species),
                   size = 2.5, colour = "black", label.padding = 0.2) +
  labs(x = paste("PC1 -", pca_expl_vars[1]),
       y = paste("PC2 -", pca_expl_vars[2])) +
  scale_colour_viridis(option = "C") +
  # scale_shape_manual(values = Site_code_colors) +
  theme_minimal() + theme(text = element_text(size = 14))
residuals_pc2_plot_bi

```

Run a PERMANOVA with adonis2 on the abundance-filtered clr-transformed species matrix.
```{r adonis_noblanks_species}
metadata_all <- full_metadata %>%
  filter(!str_detect(Library_ID, outliers_mpa3 %>% str_c(collapse = "|"))) %>%
  filter(!str_detect(Library_ID, poor_samples_mpa3 %>% str_c(collapse = "|"))) %>%
  filter(!str_detect(Library_ID, "EXB|LIB|CSN|CSD|ARS|CSS|CSL")) %>%
  filter(!str_detect(Library_ID,"MID025|MID033|MID052|MID056|MID065|MID068.A|MID076|MID078")) %>%
  left_join(., strep_rs, by = "Library_ID")

mpa3_species_decontam_unfilt <- mpa3_species_decontam %>%
  select(-one_of(outliers_mpa3)) %>%
  select(-one_of(poor_samples_mpa3)) %>%
  select(-matches("EXB|LIB|CSN|CSD|ARS|CSS|CSL")) %>%
  adorn_totals(where = "col", name = "Sum_species.decontam") %>%
  mutate(Percent = Sum_species.decontam/sum(Sum_species.decontam)*100) %>%
  filter(Percent > 0.001) %>%
  select(-one_of(c("Sum_species.decontam","Percent"))) %>%
  gather("Library_ID","Counts",2:ncol(.)) %>%
  mutate(Counts = Counts +1) %>%
  spread(Species,Counts) %>%
  column_to_rownames("Library_ID")

mpa3_species_decontam_unfilt.pca <- mixOmics::pca(mpa3_species_decontam_unfilt, ncomp = 3, logratio = 'CLR')


# Euclidean distance, uses the clr-transformed matrix from the mixOmics PCA above
mpa3_species_decontam_unfilt.euc <- vegdist(mpa3_species_decontam_unfilt.pca$X, method = "euclidean", na.rm = F)
mpa3_species_decontam_unfilt.euc.pcoa <- ape::pcoa(mpa3_species_decontam_unfilt.euc)

mpa3_species_decontam_unfilt.euc.pcoavectors <- as.data.frame(mpa3_species_decontam_unfilt.euc.pcoa$vectors)
mpa3_species_decontam_unfilt.euc.pcoavectors <- mpa3_species_decontam_unfilt.euc.pcoavectors %>%
  rownames_to_column("Library_ID")

mpa3_species_decontam_unfilt.euc.vecmet <- left_join(mpa3_species_decontam_unfilt.euc.pcoavectors, full_metadata, by = "Library_ID")

mpa3_species_decontam_unfilt.euc.vecmet %>%
  ggplot(., aes(Axis.1, Axis.2, colour = Site_code, shape = Site_code)) +
  geom_point(size = 3) +
  scale_colour_manual(values = Site_code_colors) +
  scale_shape_manual(values = Site_code_shapes) +
  # stat_ellipse(aes(colour = Site_code, group = Site_code)) +
  theme_minimal(base_size = 7) +
  ggtitle("Site_code")

# mpa3_species_decontam_unfilt %>%
#   rownames_to_column("Library_ID") %>%
#   select(Library_ID) %>%
#   anti_join(., metadata %>% select(Library_ID, Anginosus), by  = "Library_ID")


# special tables for Strep group analysis
metadata_all_strep <- metadata_all %>%
  filter(!str_detect(Library_ID, "MID046|MID075"))

mpa3_species_decontam_unfilt_strep <- mpa3_species_decontam %>%
  select(-one_of(outliers_mpa3)) %>%
  select(-one_of(poor_samples_mpa3)) %>%
  select(-matches("EXB|LIB|CSN|CSD|ARS|CSS|CSL|MID046|MID075")) %>%
  adorn_totals(where = "col", name = "Sum_species.decontam") %>%
  mutate(Percent = Sum_species.decontam/sum(Sum_species.decontam)*100) %>%
  filter(Percent > 0.001) %>%
  select(-one_of(c("Sum_species.decontam","Percent"))) %>%
  gather("Library_ID","Counts",2:ncol(.)) %>%
  mutate(Counts = Counts +1) %>%
  spread(Species,Counts) %>%
  column_to_rownames("Library_ID")

mpa3_species_decontam_unfilt_strep.pca <- mixOmics::pca(mpa3_species_decontam_unfilt_strep, ncomp = 3, logratio = 'CLR')

# Euclidean distance, uses the clr-transformed matrix from the mixOmics PCA above
mpa3_species_decontam_unfilt_strep.euc <- vegdist(mpa3_species_decontam_unfilt_strep.pca$X, method = "euclidean", na.rm = F)

# test for difference between different metadata categories 
adonis2(mpa3_species_decontam_unfilt.euc ~ Site_code, metadata_all, permutations = 999, by = "margin")
adonis2(mpa3_species_decontam_unfilt.euc ~ Time_period, metadata_all, permutations = 999, by = "margin")
adonis2(mpa3_species_decontam_unfilt.euc ~ No_reads_total, metadata_all, permutations = 999, by = "margin")
adonis2(mpa3_species_decontam_unfilt.euc ~ gc_avg, metadata_all, permutations = 999, by = "margin")
adonis2(mpa3_species_decontam_unfilt.euc ~ seq_len_avg, metadata_all, permutations = 999, by = "margin")
adonis2(mpa3_species_decontam_unfilt_strep.euc ~ Sanguinis, metadata_all_strep, permutations = 999, by = "margin")
adonis2(mpa3_species_decontam_unfilt_strep.euc ~ Anginosus, metadata_all_strep, permutations = 999, by = "margin")




all_permanova <- adonis2(mpa3_species_decontam_unfilt.euc ~ Site_code, metadata_all, permutations = 999, by = "margin") %>% rownames_to_column("Metadata")
all_permanova  <- all_permanova %>% bind_rows(., adonis2(mpa3_species_decontam_unfilt.euc ~ Time_period, metadata_all, permutations = 999, by = "margin") %>% rownames_to_column("Metadata")) %>%
  bind_rows(adonis2(mpa3_species_decontam_unfilt.euc ~ No_reads_total, metadata_all, permutations = 999, by = "margin") %>% rownames_to_column("Metadata")) %>%
  bind_rows(adonis2(mpa3_species_decontam_unfilt.euc ~ gc_avg, metadata_all, permutations = 999, by = "margin") %>% rownames_to_column("Metadata")) %>%
  bind_rows(adonis2(mpa3_species_decontam_unfilt.euc ~ seq_len_avg, metadata_all, permutations = 999, by = "margin") %>% rownames_to_column("Metadata")) %>%
  bind_rows(adonis2(mpa3_species_decontam_unfilt_strep.euc ~ Sanguinis, metadata_all_strep, permutations = 999, by = "margin") %>% rownames_to_column("Metadata")) %>%
  bind_rows(adonis2(mpa3_species_decontam_unfilt_strep.euc ~ Anginosus, metadata_all_strep, permutations = 999, by = "margin") %>% rownames_to_column("Metadata"))


all_permanova %>%
  as_tibble() %>%
  drop_na(`F`) %>%
  arrange(desc(`F`)) %>%
  bind_cols(., as_tibble(c("**","***","**","*","ns","ns","ns"))) %>%
  arrange(desc(R2))
  # fwrite(., "./05-results.backup/all_permanova.tsv", sep = "\t", quote =  F)

# controlled for tooth site
# adonis2(mpa3_species_decontam_unfilt.euc ~ Pipenotch, metadata, permutations = 999, by = "margin", strata = metadata$Tooth_site)

```



## Middenbeemster samples only
```{r pca_species_mid_pc12}
mpa3_species_decontam_unfilt_mid <- mpa3_species_decontam %>%
  select(-one_of(outliers_mpa3)) %>%
  select(-one_of(poor_samples_mpa3)) %>%
  select(matches("Species|MID|CMB")) %>%
  # adorn_totals(where = "col", name = "Sum_species.decontam") %>%
  # mutate(Percent = Sum_species.decontam/sum(Sum_species.decontam)*100) %>%
  # filter(Percent > 0.001) %>%
  # select(-one_of(c("Sum_species.decontam","Percent"))) %>%
  gather("Library_ID","Counts",2:ncol(.)) %>% 
  mutate(Counts = Counts +1) %>%
  spread(Species,Counts) %>%
  column_to_rownames("Library_ID")

# prepare to run a PCA
# check the number of components to retain by tuning the PCA
mixOmics::tune.pca(mpa3_species_decontam_unfilt_mid, logratio = 'CLR')


malt_otu.pca <- mixOmics::pca(mpa3_species_decontam_unfilt_mid, ncomp = 2, logratio = 'CLR')
malt_pca_values <- malt_otu.pca$variates$X %>%
  as.data.frame() %>%
  rownames_to_column("Library_ID") %>%
  inner_join(., full_metadata, by = "Library_ID")

ggplot(malt_pca_values, aes(PC1, PC2, color = Region, shape = Region)) +
  geom_text(aes(label = Library_ID), size = 3) +
  theme_minimal()

# perform and plot a PCA to see how the data cluster
# by pipe notch presence
mid_pipe_pc12 <- plot_pca_mid(mpa3_species_decontam_unfilt_mid, "PC1", "PC2", "Pipenotch", "Pipenotch", 3, "Pipe notch")
mid_pipe_pc12
# ggsave("~/Desktop/MID_pca_mid_pipe_pc12.pdf", plot = mid_pipe_pc12, device = "pdf",
#        scale = 1, width = 9, height = 5, units = c("in"), dpi = 300)

# by sex
plot_pca_mid(mpa3_species_decontam_unfilt_mid, "PC1", "PC2", "Sex", "Pipenotch", 3, "Sex")

# by age
plot_pca_mid(mpa3_species_decontam_unfilt_mid, "PC1", "PC2", "Age", "Pipenotch", 3, "Age")

# by presence of periodontal disease
plot_pca_mid(mpa3_species_decontam_unfilt_mid, "PC1", "PC2", "Periodontal_disease_present", "Pipenotch", 3, "Periodontal disease present")

# by presence of caries
plot_pca_mid(mpa3_species_decontam_unfilt_mid, "PC1", "PC2", "Caries_present", "Pipenotch", 3, "Caries present")

# by ante-mortem tooth loss
plot_pca_mid(mpa3_species_decontam_unfilt_mid, "PC1", "PC2", "AMTL", "Pipenotch", 3, "AMTL")

# by supragingival calculus score
plot_pca_cont(mpa3_species_decontam_unfilt_mid, "PC1", "PC2", "calc_sup_SI_score", "Pipenotch", 3, "Sup calc SI score")

# by subgingival calculus score
plot_pca_cont(mpa3_species_decontam_unfilt_mid, "PC1", "PC2", "Calc_sub_SI_score", "Pipenotch", 3, "Sub calc SI score")

# by presence of perioapical lesions
plot_pca_mid(mpa3_species_decontam_unfilt_mid, "PC1", "PC2", "Periapical_lesions", "Pipenotch", 3, "Periapical lesions")

# by max periodontal disease score
plot_pca_cont(mpa3_species_decontam_unfilt_mid, "PC1", "PC2", "Max_Perio_Score", "Pipenotch", 3, "Max perio score")

# by periodontal affect score
plot_pca_cont(mpa3_species_decontam_unfilt_mid, "PC1", "PC2", "Peridontal_affect_score", "Pipenotch", 3, "Perio affect score")

# by % antemortem tooth loss
plot_pca_cont(mpa3_species_decontam_unfilt_mid, "PC1", "PC2", "%AMTL", "Pipenotch", 3, " % AMTL")

# by % positions w/periapical lesions
plot_pca_cont(mpa3_species_decontam_unfilt_mid, "PC1", "PC2", "%_positions_perioapical", "Pipenotch", 3, "% positions periapical lesions")

# by % teeth with caries
plot_pca_cont(mpa3_species_decontam_unfilt_mid, "PC1", "PC2", "%teeth_with_caries", "Pipenotch", 3, "% teeth w/ caries")

# by % teeth with calculus
plot_pca_cont(mpa3_species_decontam_unfilt_mid, "PC1", "PC2", "%teeth_with_calc", "Pipenotch", 3, "% teeth w/ calc")

# by % positions with periodontal disease
plot_pca_cont(mpa3_species_decontam_unfilt_mid, "PC1", "PC2", "%_positions_with_Periodontal", "Pipenotch", 3, "% positions w/ PD")

# by minimum number of pipenotches 
mid_minpipe_pc12 <- plot_pca_cont(mpa3_species_decontam_unfilt_mid, "PC1", "PC2", "Min_no_Pipe_notches", "Pipenotch", 3, "Min # pipe notches")
mid_minpipe_pc12
# ggsave("~/Desktop/MID_pca_mid_minpipe_pc12.pdf", plot = mid_minpipe_pc12, device = "pdf",
#        scale = 1, width = 9, height = 5, units = c("in"), dpi = 300)

# by sequencing depth 
mid_seqdepth_pc12 <- plot_pca_cont(mpa3_species_decontam_unfilt_mid, "PC1", "PC2", "No_reads_total", "Pipenotch", 3, "Total reads into MetaPhlAn3")
mid_seqdepth_pc12
# ggsave("~/Desktop/MID_pca_mid_seqdepth_pc12.pdf", plot = mid_seqdepth_pc12, device = "pdf",
#        scale = 1, width = 9, height = 5, units = c("in"), dpi = 300)

# by sampled tooth site(s)
mid_tooth_pc12 <- plot_pca_mid(mpa3_species_decontam_unfilt_mid, "PC1", "PC2", "Tooth_location", "Pipenotch", 3, "Tooth location")
mid_tooth_pc12
# ggsave("06-publication/supplemental_figures/SXX14/MID_pca_mid_tooth_pc12.pdf", plot = mid_tooth_pc12, device = "pdf",
#        scale = 1, width = 5, height = 5, units = c("in"), dpi = 300)

# by sampled tooth site(s)
plot_pca_mid(mpa3_species_decontam_unfilt_mid, "PC1", "PC2", "alt_tooth_location", "Pipenotch", 3, "Tooth location")

# by mg of extracted calculus
plot_pca_cont(mpa3_species_decontam_unfilt_mid, "PC1", "PC2", "Calculus_extracted_mg", "Pipenotch", 3, "Weight of extracted calculus")

# by [DNA] after extraction per mg of calculus extracted
plot_pca_cont(mpa3_species_decontam_unfilt_mid, "PC1", "PC2", "extract_conc_ng/mg", "Pipenotch", 3, "DNA extract concentration (ng/mg)")

# by [DNA] after extraction per uL of eluate
mid_extract_pc12 <- plot_pca_cont(mpa3_species_decontam_unfilt_mid, "PC1", "PC2", "extract_conc_ng/uL", "Pipenotch", 3, "DNA extract concentration (ng/uL)")
mid_extract_pc12
# ggsave("~/Desktop/MID_pca_mid_extract_pc12.pdf", plot = mid_extract_pc12, device = "pdf",
#        scale = 1, width = 9, height = 5, units = c("in"), dpi = 300)

# by soil type
plot_pca_mid(mpa3_species_decontam_unfilt_mid, "PC1", "PC2", "Soil", "Pipenotch", 3, "Soil type")

# by average GC content
plot_pca_cont(mpa3_species_decontam_unfilt_mid, "PC1", "PC2", "gc_avg", "Time_period", 3, "Avg. GC content (%)")

# by average read length
plot_pca_cont(mpa3_species_decontam_unfilt_mid, "PC1", "PC2", "seq_len_avg", "Time_period", 3, "Avg. read length (bp)")

# by Strep group proportion - sanguinis
plot_pca_cont(mpa3_species_decontam_unfilt_mid, "PC1", "PC2", "sanguinis", "Time_period", 3, "Proportion of Sanguinis group")

# by Strep group proportion - mitis
plot_pca_cont(mpa3_species_decontam_unfilt_mid, "PC1", "PC2", "mitis", "Time_period", 3, "Proportion of Mitis group")

# with names
# pca.list <- mixOmics::pca(mpa3_species_decontam_unfilt_mid, ncomp = 3, logratio = 'CLR')
# plot_pca_names(pca.list, "PC1", "PC2", "Pipenotch")

# ggsave("05-results.backup/pca_mpa3_species_pc1pc2.pdf", plot = pca_mpa3_species_pc1pc2, device = "pdf",
#        scale = 1, width = 7, height = 5, units = c("in"),
#        dpi = 300)
```

```{r pca_species_mid_pc32}
# by sex
plot_pca_mid(mpa3_species_decontam_unfilt_mid, "PC3", "PC2", "Sex", "Pipenotch", 3, "Sex")

# by age
plot_pca_mid(mpa3_species_decontam_unfilt_mid, "PC3", "PC2", "Age", "Pipenotch", 3, "Age")

# by pipe notch presence
plot_pca_mid(mpa3_species_decontam_unfilt_mid, "PC3", "PC2", "Pipenotch", "Pipenotch", 3, "Pipe notch")

# by presence of periodontal disease
plot_pca_mid(mpa3_species_decontam_unfilt_mid, "PC3", "PC2", "Periodontal_disease_present", "Pipenotch", 3, "Periodontal disease present")

# by presence of caries
plot_pca_mid(mpa3_species_decontam_unfilt_mid, "PC3", "PC2", "Caries_present", "Pipenotch", 3, "Caries present")

# by ante-mortem tooth loss
plot_pca_mid(mpa3_species_decontam_unfilt_mid, "PC3", "PC2", "AMTL", "Pipenotch", 3, "AMTL")

# by supragingival calculus score
plot_pca_cont(mpa3_species_decontam_unfilt_mid, "PC3", "PC2", "calc_sup_SI_score", "Pipenotch", 3, "Sup calc SI score")

# by subgingival calculus score
plot_pca_cont(mpa3_species_decontam_unfilt_mid, "PC3", "PC2", "Calc_sub_SI_score", "Pipenotch", 3, "Sub calc SI score")

# by presence of perioapical lesions
plot_pca_mid(mpa3_species_decontam_unfilt_mid, "PC3", "PC2", "Periapical_lesions", "Pipenotch", 3, "Periapical lesions")

# by max periodontal disease score
plot_pca_cont(mpa3_species_decontam_unfilt_mid, "PC3", "PC2", "Max_Perio_Score", "Pipenotch", 3, "Max perio score")

# by periodontal affect score
plot_pca_cont(mpa3_species_decontam_unfilt_mid, "PC3", "PC2", "Peridontal_affect_score", "Pipenotch", 3, "Perio affect score")

# by % antemortem tooth loss
plot_pca_cont(mpa3_species_decontam_unfilt_mid, "PC3", "PC2", "%AMTL", "Pipenotch", 3, "% AMTL")

# by % positions w/periapical lesions
plot_pca_cont(mpa3_species_decontam_unfilt_mid, "PC3", "PC2", "%_positions_perioapical", "Pipenotch", 3, "% positions periapical lesions")

# by % teeth with caries
plot_pca_cont(mpa3_species_decontam_unfilt_mid, "PC3", "PC2", "%teeth_with_caries", "Pipenotch", 3, "% teeth w/ caries")

# by % teeth with calculus
plot_pca_cont(mpa3_species_decontam_unfilt_mid, "PC3", "PC2", "%teeth_with_calc", "Pipenotch", 3, "% teeth w/ calc")

# by % positions with periodontal disease
plot_pca_cont(mpa3_species_decontam_unfilt_mid, "PC3", "PC2", "%_positions_with_Periodontal", "Pipenotch", 3, "% positions w/ PD")

# by minimum number of pipenotches 
plot_pca_cont(mpa3_species_decontam_unfilt_mid, "PC3", "PC2", "Min_no_Pipe_notches", "Pipenotch", 3, "Min # pipe notches")

# by sequencing depth 
plot_pca_cont(mpa3_species_decontam_unfilt_mid, "PC3", "PC2", "No_reads_total", "Pipenotch", 3, "Total reads into MALT")

# by sampled tooth site(s)
plot_pca_mid(mpa3_species_decontam_unfilt_mid, "PC3", "PC2", "Tooth_location", "Pipenotch", 3, "Tooth location")

# by sampled tooth site(s)
mid_tooth_pc23 <- plot_pca_mid(mpa3_species_decontam_unfilt_mid, "PC3", "PC2", "Tooth_location", "Pipenotch", 3, "Tooth location")
mid_tooth_pc23
# ggsave("06-publication/supplemental_figures/SXX14/MID_pca_mid_tooth_pc23.pdf", plot = mid_tooth_pc23, device = "pdf",
#        scale = 1, width = 5, height = 5, units = c("in"), dpi = 300)
# 
# by mg of extracted calculus
plot_pca_cont(mpa3_species_decontam_unfilt_mid, "PC3", "PC2", "Calculus_extracted_mg", "Pipenotch", 3, "Weight of extracted calculus")

# by [DNA] after extraction
plot_pca_cont(mpa3_species_decontam_unfilt_mid, "PC3", "PC2", "extract_conc_ng/mg", "Pipenotch", 3, "DNA extract concentration (ng/mg)")

# by [DNA] after extraction per uL of eluate
plot_pca_cont(mpa3_species_decontam_unfilt_mid, "PC3", "PC2", "extract_conc_ng/uL", "Pipenotch", 3, "DNA extract concentration (ng/uL)")

# by soil type
plot_pca_mid(mpa3_species_decontam_unfilt_mid, "PC3", "PC2", "Soil", "Pipenotch", 3, "Soil type")

# by average GC content
plot_pca_cont(mpa3_species_decontam_unfilt_mid, "PC3", "PC2", "gc_avg", "Time_period", 33, "Avg. GC content (%)")

# by average read length
plot_pca_cont(mpa3_species_decontam_unfilt_mid, "PC3", "PC2", "seq_len_avg", "Time_period", 33, "Avg. read length (bp)")

```



```{r biplots_mid_species}

mpa3_species_decontam_unfilt_mid <- mpa3_species_decontam %>%
  select(-one_of(outliers_mpa3)) %>%
  select(-one_of(poor_samples_mpa3)) %>%
  select(matches("Species|MID|CMB")) %>%
  gather("Library_ID","Counts",2:ncol(.)) %>% 
  mutate(Counts = Counts +1) %>%
  spread(Species,Counts) %>%
  column_to_rownames("Library_ID")

mpa3_species_decontam_unfilt_mid.pca <- mixOmics::pca(mpa3_species_decontam_unfilt_mid, ncomp = 3, logratio = 'CLR')
plot(mpa3_species_decontam_unfilt_mid.pca)

# Define pipenotch colors for plotting
Pipenotch_colors = c("#311068","#C83E73")
Pipenotch_shapes = c(17,19)

mpa3_species_decontam_filtered_nocontrols_biplot_121 <- plot_pca_bi(mpa3_species_decontam_unfilt_mid.pca, "PC1", "PC2", "Pipenotch", PC1)
mpa3_species_decontam_filtered_nocontrols_biplot_121

mpa3_species_decontam_filtered_nocontrols_biplot_122 <- plot_pca_bi(mpa3_species_decontam_unfilt_mid.pca, "PC1", "PC2", "Pipenotch", PC2)
mpa3_species_decontam_filtered_nocontrols_biplot_122

pandoc.table(mpa3_species_decontam_filtered_nocontrols_biplot_121$plot_env$pws_10 %>% arrange(desc(PC1)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC1 positive loadings")

pandoc.table(mpa3_species_decontam_filtered_nocontrols_biplot_121$plot_env$neg_10 %>% arrange(PC1),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC1 negative loadings")


pandoc.table(mpa3_species_decontam_filtered_nocontrols_biplot_122$plot_env$pws_10 %>% arrange(desc(PC2)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC2 positive loadings")

pandoc.table(mpa3_species_decontam_filtered_nocontrols_biplot_122$plot_env$neg_10 %>% arrange(PC2),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC2 negative loadings")


mpa3_species_decontam_filtered_nocontrols_biplot_322 <- plot_pca_bi(mpa3_species_decontam_unfilt_mid.pca, "PC3", "PC2", "Pipenotch", PC2)
mpa3_species_decontam_filtered_nocontrols_biplot_322

mpa3_species_decontam_filtered_nocontrols_biplot_323 <- plot_pca_bi(mpa3_species_decontam_unfilt_mid.pca, "PC3", "PC2", "Pipenotch", PC3)
mpa3_species_decontam_filtered_nocontrols_biplot_323

pandoc.table(mpa3_species_decontam_filtered_nocontrols_biplot_323$plot_env$pws_10 %>% arrange(desc(PC3)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC3 positive loadings")

pandoc.table(mpa3_species_decontam_filtered_nocontrols_biplot_323$plot_env$neg_10 %>% arrange(PC3),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC3 negative loadings")

```

```{r biplot_species_lists, eval = F}

pc1 <- mpa3_species_decontam_filtered_nocontrols_biplot_121$plot_env$pws_10 %>% 
  arrange(desc(PC1)) %>%
  bind_rows(., mpa3_species_decontam_filtered_nocontrols_biplot_121$plot_env$neg_10 %>% 
  arrange(PC1)) %>%
  select(Taxon, PC1)

fwrite(pc1, "./05-results.backup/mpa3_pc1_species_list.tsv", quote = F, sep = "\t")

pc2 <- mpa3_species_decontam_filtered_nocontrols_biplot_122$plot_env$pws_10 %>% 
  arrange(desc(PC2)) %>%
  bind_rows(., mpa3_species_decontam_filtered_nocontrols_biplot_122$plot_env$neg_10 %>% 
  arrange(PC2)) %>%
  select(Taxon, PC2)

fwrite(pc2, "./05-results.backup/mpa3_pc2_species_list.tsv", quote = F, sep = "\t")

```

Run a permanova with function adonis to test if the clusters are significantly
different. 
```{r metadata_prep_permanova}

metadata_mid <- full_metadata %>%
  as_tibble() %>%
  filter(!Library_ID %in% outliers_mpa3) %>%
  filter(!Library_ID %in% poor_samples_mpa3) %>%
  filter(str_detect(Library_ID, "MID")) %>%
  # remove the MID samples that have no metadata
  filter(!str_detect(Library_ID, "MID025.A0101|MID033.A0101|MID056.A0101|MID065.A0101|MID068.A0101|MID076.A0101|MID078.A0101|MID052.A0101")) %>%
  mutate(Periodontal_disease_present = fct_relevel(Periodontal_disease_present, "Yes","No","Unobservable"),
         Caries_present = fct_relevel(Caries_present, "Yes","No","Unobservable"),
         Sex = fct_relevel(Sex, "Female","Male","Indeterminate")) %>%
  # select(Library_ID, Periodontal_disease_present, Caries_present, Sex, Tooth_location, Pipenotch) %>%
  arrange(Library_ID) %>%
  unique() %>%
  left_join(., strep_rs)

mpa3_species_decontam_unfilt_mid <- mpa3_species_decontam %>%
  select(-one_of(outliers_mpa3)) %>%
  select(-one_of(poor_samples_mpa3)) %>%
  select(matches("Species|MID")) %>%
  adorn_totals(where = "col", name = "Sum_species.decontam") %>%
  mutate(Percent = Sum_species.decontam/sum(Sum_species.decontam)*100) %>%
  filter(Percent > 0.001) %>%
  select(-one_of(c("Sum_species.decontam","Percent"))) %>%
  gather("Library_ID","Counts",2:ncol(.)) %>% 
  mutate(Counts = Counts +1) %>%
  spread(Species,Counts) %>%
  column_to_rownames("Library_ID")

mpa3_species_decontam_unfilt_mid.pca <- mixOmics::pca(mpa3_species_decontam_unfilt_mid, ncomp = 3, logratio = 'CLR')

```

```{r special_permanova_tables}
# some special input/metadata tables to remove NAs from specific columns

# extraction weight (mg)
metadata_mid_exmg <- metadata_mid %>% drop_na(Calculus_extracted_mg)

mpa3_species_decontam_unfilt_mid_exmg <- mpa3_species_decontam %>%
  select(-one_of(outliers_mpa3)) %>%
  select(-one_of(poor_samples_mpa3)) %>%
  select(matches("Species|MID")) %>%
  # no data for weight of extracted calculus
  select(-matches("MID029|MID030|MID031")) %>%
  adorn_totals(where = "col", name = "Sum_species.decontam") %>%
  mutate(Percent = Sum_species.decontam/sum(Sum_species.decontam)*100) %>%
  filter(Percent > 0.001) %>%
  select(-one_of(c("Sum_species.decontam","Percent"))) %>%
  gather("Library_ID","Counts",2:ncol(.)) %>% 
  mutate(Counts = Counts +1) %>%
  spread(Species,Counts) %>%
  column_to_rownames("Library_ID")

mpa3_species_decontam_unfilt_mid_exmg.pca <- mixOmics::pca(mpa3_species_decontam_unfilt_mid_exmg, ncomp = 3, logratio = 'CLR')
mpa3_species_decontam_unfilt_mid_exmg.euc <- vegdist(mpa3_species_decontam_unfilt_mid_exmg.pca$X, method = "euclidean", na.rm = T)

# % teeth with caries (also kworks for % positions perioapical)
metadata_mid_pctc <- metadata_mid %>% drop_na(`%teeth_with_caries`)

mpa3_species_decontam_unfilt_mid_pctc <- mpa3_species_decontam %>%
  select(-one_of(outliers_mpa3)) %>%
  select(-one_of(poor_samples_mpa3)) %>%
  select(matches("Species|MID")) %>%
  # no data for % teeth with caries
  select(-matches("MID028|MID033|MID052|MID056|MID065|MID068.A|MID076|MID078")) %>%
  adorn_totals(where = "col", name = "Sum_species.decontam") %>%
  mutate(Percent = Sum_species.decontam/sum(Sum_species.decontam)*100) %>%
  filter(Percent > 0.001) %>%
  select(-one_of(c("Sum_species.decontam","Percent"))) %>%
  gather("Library_ID","Counts",2:ncol(.)) %>% 
  mutate(Counts = Counts +1) %>%
  spread(Species,Counts) %>%
  column_to_rownames("Library_ID")

mpa3_species_decontam_unfilt_mid_pctc.pca <- mixOmics::pca(mpa3_species_decontam_unfilt_mid_pctc, ncomp = 3, logratio = 'CLR')
mpa3_species_decontam_unfilt_mid_exmg_pctc.euc <- vegdist(mpa3_species_decontam_unfilt_mid_pctc.pca$X, method = "euclidean", na.rm = T)

# % teeth with calculus
metadata_mid_pctcal <- metadata_mid %>% drop_na(`%teeth_with_calc`)

mpa3_species_decontam_unfilt_mid_pctcal <- mpa3_species_decontam %>%
  select(-one_of(outliers_mpa3)) %>%
  select(-one_of(poor_samples_mpa3)) %>%
  select(matches("Species|MID")) %>%
  # no data for % teeth with calculus
  select(-matches("MID033|MID052|MID056|MID065|MID068.A|MID076|MID078|MID079|MID092")) %>%
  adorn_totals(where = "col", name = "Sum_species.decontam") %>%
  mutate(Percent = Sum_species.decontam/sum(Sum_species.decontam)*100) %>%
  filter(Percent > 0.001) %>%
  select(-one_of(c("Sum_species.decontam","Percent"))) %>%
  gather("Library_ID","Counts",2:ncol(.)) %>% 
  mutate(Counts = Counts +1) %>%
  spread(Species,Counts) %>%
  column_to_rownames("Library_ID")

mpa3_species_decontam_unfilt_mid_pctcal.pca <- mixOmics::pca(mpa3_species_decontam_unfilt_mid_pctcal, ncomp = 3, logratio = 'CLR')
mpa3_species_decontam_unfilt_mid_pctcal.euc <- vegdist(mpa3_species_decontam_unfilt_mid_pctcal.pca$X, method = "euclidean", na.rm = T)

# periodontal affect score
metadata_mid_pas <- metadata_mid %>% drop_na(Peridontal_affect_score)

mpa3_species_decontam_unfilt_mid_pas <- mpa3_species_decontam %>%
  select(-one_of(outliers_mpa3)) %>%
  select(-one_of(poor_samples_mpa3)) %>%
  select(matches("Species|MID")) %>%
  # no data for periodontal affect score
  select(-matches("MID025|MID033|MID052|MID056|MID065|MID068.A|MID076|MID078|MID081|MID094|MID097")) %>%
  adorn_totals(where = "col", name = "Sum_species.decontam") %>%
  mutate(Percent = Sum_species.decontam/sum(Sum_species.decontam)*100) %>%
  filter(Percent > 0.001) %>%
  select(-one_of(c("Sum_species.decontam","Percent"))) %>%
  gather("Library_ID","Counts",2:ncol(.)) %>% 
  mutate(Counts = Counts +1) %>%
  spread(Species,Counts) %>%
  column_to_rownames("Library_ID")

mpa3_species_decontam_unfilt_mid_pas.pca <- mixOmics::pca(mpa3_species_decontam_unfilt_mid_pas, ncomp = 3, logratio = 'CLR')
mpa3_species_decontam_unfilt_mid_pas.euc <- vegdist(mpa3_species_decontam_unfilt_mid_pas.pca$X, method = "euclidean", na.rm = T)

# % positions with periodontal
metadata_mid_perio <- metadata_mid %>% drop_na(`%_positions_with_Periodontal`)

mpa3_species_decontam_unfilt_mid_perio <- mpa3_species_decontam %>%
  select(-one_of(outliers_mpa3)) %>%
  select(-one_of(poor_samples_mpa3)) %>%
  select(matches("Species|MID")) %>%
  # no data for periodontal affect score
  select(-matches("MID033|MID052|MID056|MID065|MID068.A|MID076|MID078|MID081|MID094|MID097")) %>%
  adorn_totals(where = "col", name = "Sum_species.decontam") %>%
  mutate(Percent = Sum_species.decontam/sum(Sum_species.decontam)*100) %>%
  filter(Percent > 0.001) %>%
  select(-one_of(c("Sum_species.decontam","Percent"))) %>%
  gather("Library_ID","Counts",2:ncol(.)) %>% 
  mutate(Counts = Counts +1) %>%
  spread(Species,Counts) %>%
  column_to_rownames("Library_ID")

mpa3_species_decontam_unfilt_mid_perio.pca <- mixOmics::pca(mpa3_species_decontam_unfilt_mid_perio, ncomp = 3, logratio = 'CLR')
mpa3_species_decontam_unfilt_mid_perio.euc <- vegdist(mpa3_species_decontam_unfilt_mid_perio.pca$X, method = "euclidean", na.rm = T)

# extract concentration ng/uL
metadata_mid_exconc <- metadata_mid %>% drop_na(`extract_conc_ng/uL`)  # %>% filter(!str_detect(Library_ID, "MID024|MID063|MID092"))

mpa3_species_decontam_unfilt_mid_exconc <- mpa3_species_decontam %>%
  select(-one_of(outliers_mpa3)) %>%
  select(-one_of(poor_samples_mpa3)) %>%
  select(matches("Species|MID")) %>%
  # no data for periodontal affect score
  select(-matches("MID025|MID029|MID030|MID031|MID033|MID052|MID056|MID065|MID068.A|MID076|MID078")) %>%
  adorn_totals(where = "col", name = "Sum_species.decontam") %>%
  mutate(Percent = Sum_species.decontam/sum(Sum_species.decontam)*100) %>%
  filter(Percent > 0.001) %>%
  select(-one_of(c("Sum_species.decontam","Percent"))) %>%
  gather("Library_ID","Counts",2:ncol(.)) %>% 
  mutate(Counts = Counts +1) %>%
  spread(Species,Counts) %>%
  column_to_rownames("Library_ID")

mpa3_species_decontam_unfilt_mid_exconc.pca <- mixOmics::pca(mpa3_species_decontam_unfilt_mid_exconc, ncomp = 3, logratio = 'CLR')
mpa3_species_decontam_unfilt_mid_exconc.euc <- vegdist(mpa3_species_decontam_unfilt_mid_exconc.pca$X, method = "euclidean", na.rm = T)


# Sanguinis and Anginosus group strep proportions
metadata_mid_strep <- metadata_mid %>% drop_na(Sanguinis)  # %>% filter(!str_detect(Library_ID, "MID024|MID063|MID092"))

mpa3_species_decontam_unfilt_mid_strep <- mpa3_species_decontam %>%
  select(-one_of(outliers_mpa3)) %>%
  select(-one_of(poor_samples_mpa3)) %>%
  select(matches("Species|MID")) %>%
  # no data for periodontal affect score
  select(-matches("MID046|MID075")) %>%
  adorn_totals(where = "col", name = "Sum_species.decontam") %>%
  mutate(Percent = Sum_species.decontam/sum(Sum_species.decontam)*100) %>%
  filter(Percent > 0.001) %>%
  select(-one_of(c("Sum_species.decontam","Percent"))) %>%
  gather("Library_ID","Counts",2:ncol(.)) %>% 
  mutate(Counts = Counts +1) %>%
  spread(Species,Counts) %>%
  column_to_rownames("Library_ID")

mpa3_species_decontam_unfilt_mid_strep.pca <- mixOmics::pca(mpa3_species_decontam_unfilt_mid_strep, ncomp = 3, logratio = 'CLR')
mpa3_species_decontam_unfilt_mid_strep.euc <- vegdist(mpa3_species_decontam_unfilt_mid_strep.pca$X, method = "euclidean", na.rm = T)


# mpa3_species_decontam_unfilt_mid %>%
#   rownames_to_column("Library_ID") %>%
#   select(Library_ID) %>%
#   anti_join(., metadata_mid %>% select(Library_ID, Anginosus), by  = "Library_ID")

```


Run a PERMANOVA with adonis2 on the abundance-filtered clr-transformed species matrix.
```{r adonis_mid_species}
# Euclidean distance, uses the clr-transformed matrix from the mixOmics PCA above
mpa3_species_decontam_unfilt_mid.euc <- vegdist(mpa3_species_decontam_unfilt_mid.pca$X, method = "euclidean", na.rm = T)
mpa3_species_decontam_unfilt_mid.euc.pcoa <- ape::pcoa(mpa3_species_decontam_unfilt_mid.euc)

mpa3_species_decontam_unfilt_mid.euc.pcoavectors <- as.data.frame(mpa3_species_decontam_unfilt_mid.euc.pcoa$vectors)
mpa3_species_decontam_unfilt_mid.euc.pcoavectors <- mpa3_species_decontam_unfilt_mid.euc.pcoavectors %>%
  rownames_to_column("Library_ID")

mpa3_species_decontam_unfilt_mid.euc.vecmet <- inner_join(mpa3_species_decontam_unfilt_mid.euc.pcoavectors, metadata_mid, by = "Library_ID")

mpa3_species_decontam_unfilt_mid.euc.vecmet %>%
  ggplot(., aes(Axis.1, Axis.2, colour = Site_code, shape = Site_code)) +
  geom_point(size = 3) +
  scale_colour_manual(values = Site_code_colors) +
  scale_shape_manual(values = Site_code_shapes) +
  stat_ellipse(aes(colour = Site_code, group = Site_code)) +
  theme_minimal(base_size = 7) +
  ggtitle("Site_code")

# mpa3_species_decontam_unfilt_mid %>%
#   rownames_to_column("Library_ID") %>%
#   select(Library_ID) %>%
#   anti_join(., metadata_mid %>% select(Library_ID, Anginosus), by  = "Library_ID")


# test for difference between different metadata categories 
mid_permanova <- adonis2(mpa3_species_decontam_unfilt_mid.euc ~ gc_avg, metadata_mid, permutations = 999, by = "margin") %>% rownames_to_column("Metadata")
mid_permanova  <- mid_permanova %>% bind_rows(., adonis2(mpa3_species_decontam_unfilt_mid.euc ~ Sex, metadata_mid, permutations = 999, by = "margin") %>% rownames_to_column("Metadata")) %>%
  bind_rows(adonis2(mpa3_species_decontam_unfilt_mid.euc ~ Pipenotch, metadata_mid, permutations = 999, by = "margin") %>% rownames_to_column("Metadata")) %>%
  bind_rows(adonis2(mpa3_species_decontam_unfilt_mid.euc ~ seq_len_avg, metadata_mid, permutations = 999, by = "margin") %>% rownames_to_column("Metadata")) %>%
  bind_rows(adonis2(mpa3_species_decontam_unfilt_mid.euc ~ `DNA_molecules_per_library_x10^6`, metadata_mid, permutations = 999, by = "margin") %>% rownames_to_column("Metadata")) %>%
  bind_rows(adonis2(mpa3_species_decontam_unfilt_mid.euc ~ calc_sup_SI_score, metadata_mid, permutations = 999, by = "margin") %>% rownames_to_column("Metadata")) %>%
  bind_rows(adonis2(mpa3_species_decontam_unfilt_mid.euc ~ Age, metadata_mid, permutations = 999, by = "margin") %>% rownames_to_column("Metadata")) %>%
  bind_rows(adonis2(mpa3_species_decontam_unfilt_mid.euc ~ `%AMTL`, metadata_mid, permutations = 999, by = "margin") %>% rownames_to_column("Metadata")) %>%
  bind_rows(adonis2(mpa3_species_decontam_unfilt_mid.euc ~ Max_Perio_Score, metadata_mid, permutations = 999, by = "margin") %>% rownames_to_column("Metadata")) %>%
  bind_rows(adonis2(mpa3_species_decontam_unfilt_mid.euc ~ Calc_sub_SI_score, metadata_mid, permutations = 999, by = "margin") %>% rownames_to_column("Metadata")) %>%
  bind_rows(adonis2(mpa3_species_decontam_unfilt_mid.euc ~ `extract_conc_after_storage_ng/uL`, metadata_mid, permutations = 999, by = "margin") %>% rownames_to_column("Metadata")) %>%
  bind_rows(adonis2(mpa3_species_decontam_unfilt_mid.euc ~ `extract_conc_ng/mg`, metadata_mid, permutations = 999, by = "margin") %>% rownames_to_column("Metadata")) %>%
  bind_rows(adonis2(mpa3_species_decontam_unfilt_mid.euc ~ No_reads_total, metadata_mid, permutations = 999, by = "margin") %>% rownames_to_column("Metadata")) %>%
  bind_rows(adonis2(mpa3_species_decontam_unfilt_mid.euc ~ Tooth_location, metadata_mid, permutations = 999, by = "margin") %>% rownames_to_column("Metadata")) %>%
  bind_rows(adonis2(mpa3_species_decontam_unfilt_mid_exmg.euc ~ Calculus_extracted_mg, metadata_mid_exmg, permutations = 999, by = "margin") %>% rownames_to_column("Metadata")) %>%
  bind_rows(adonis2(mpa3_species_decontam_unfilt_mid_exmg_pctc.euc ~ `%teeth_with_caries`, metadata_mid_pctc, permutations = 999, by = "margin") %>% rownames_to_column("Metadata")) %>%
  bind_rows(adonis2(mpa3_species_decontam_unfilt_mid_exmg_pctc.euc ~ `%_positions_perioapical`, metadata_mid_pctc, permutations = 999, by = "margin") %>% rownames_to_column("Metadata")) %>%
  bind_rows(adonis2(mpa3_species_decontam_unfilt_mid_pctcal.euc ~ `%teeth_with_calc`, metadata_mid_pctcal, permutations = 999, by = "margin") %>% rownames_to_column("Metadata")) %>%
  bind_rows(adonis2(mpa3_species_decontam_unfilt_mid_pas.euc ~ Peridontal_affect_score, metadata_mid_pas, permutations = 999, by = "margin") %>% rownames_to_column("Metadata")) %>%
  bind_rows(adonis2(mpa3_species_decontam_unfilt_mid_perio.euc ~ `%_positions_with_Periodontal`, metadata_mid_perio, permutations = 999, by = "margin") %>% rownames_to_column("Metadata")) %>%
  bind_rows(adonis2(mpa3_species_decontam_unfilt_mid_exconc.euc ~ `extract_conc_ng/uL`, metadata_mid_exconc, permutations = 999, by = "margin") %>% rownames_to_column("Metadata")) %>%
  bind_rows(adonis2(mpa3_species_decontam_unfilt_mid_strep.euc ~ Sanguinis, metadata_mid_strep, permutations = 999, by = "margin") %>% rownames_to_column("Metadata")) %>%
  bind_rows(adonis2(mpa3_species_decontam_unfilt_mid_strep.euc ~ Anginosus, metadata_mid_strep, permutations = 999, by = "margin") %>% rownames_to_column("Metadata"))


mid_permanova %>%
  as_tibble() %>%
  drop_na(`F`) %>%
  arrange(desc(`F`)) %>%
  bind_cols(., as_tibble(c("***","***","***","**","**","**","ns","*","ns","ns","ns","ns","ns","ns","ns","ns","ns","ns","ns","ns","ns","ns","ns"))) %>%
  arrange(desc(R2))
  # fwrite(., "./05-results.backup/mid_permanova.tsv", sep = "\t", quote =  F)

# controlled for tooth site
# adonis2(mpa3_species_decontam_unfilt_mid.euc ~ Pipenotch, metadata_mid, permutations = 999, by = "margin", strata = metadata_mid$Tooth_site)

```

## Filtered PCAs
```{r pca_plot_fxn}
# plot PCA with colored dots and the title including the # of species or genera
plot_pca <- function(df, pc1, pc2, color_group, shape_group, ncomps) {
    metadata_group_colors <- get(paste(color_group, "_colors", sep = ""))
    metadata_group_shapes <- get(paste(shape_group, "_shapes", sep = ""))

    pca.list <- mixOmics::pca(df, ncomp = ncomps, logratio = 'CLR')

    exp_var <- paste0(round(pca.list$prop_expl_var$X * 100, 2), "%")
    df_X <- pca.list$variates$X %>%
              as.data.frame() %>%
              rownames_to_column("Library_ID") %>%
              inner_join(full_metadata, by = "Library_ID") %>%
              inner_join(., strep_group_info)
              
    color_group = df_X[[color_group]]
    shape_group = df_X[[shape_group]]

    if (pc1 == 'PC1') {
        pc1 <- df_X$PC1
        exp_var_pc1 <- exp_var[1]
        xaxis <- c("PC1")
    }  else if (pc1 == 'PC2') {
        pc1 <- df_X$PC2
        exp_var_pc1 <- exp_var[2]
        xaxis <- c("PC2")
    } else if (pc1 == 'PC3') {
        pc1 <- df_X$PC3
        exp_var_pc1 <- exp_var[3]
        xaxis <- c("PC3")
    }

    if (pc2 == 'PC1') {
        pc2 <- df_X$PC1
        exp_var_pc2 <- exp_var[1]
        yaxis <- c("PC1")
    }  else if (pc2 == 'PC2') {
        pc2 <- df_X$PC2
        exp_var_pc2 <- exp_var[2]
        yaxis <- c("PC2")
    } else if (pc2 == 'PC3') {
        pc2 <- df_X$PC3
        exp_var_pc2 <- exp_var[3]
        yaxis <- c("PC3")
    }

    pca_plot <- ggplot(df_X, aes(pc1, pc2)) +
     geom_point(aes(, fill = color_group, shape = shape_group), size = 3.5, stroke = 0.3) +
     scale_fill_manual(values = metadata_group_colors) +
     scale_shape_manual(values = metadata_group_shapes) +
     # stat_ellipse() +
     xlab(paste(xaxis, " - ", exp_var_pc1)) +
     ylab(paste(yaxis, " - ", exp_var_pc2)) +
     theme_minimal(base_size = 16) +
     theme(text = element_text(size=16)) +
     theme(legend.title = element_blank(),
           legend.key.size = unit(2,"mm"),
           legend.text = element_text(size = 6)) +
     theme(legend.position = "top")

    return(pca_plot)
}


```

```{r pca_plot_mid_fxn}
# plot PCA with colored dots 
plot_pca_mid <- function(df, pc1, pc2, color_group, shape_group, ncomps, title_text) {

    pca.list <- mixOmics::pca(df, ncomp = ncomps, logratio = 'CLR')

    exp_var <- paste0(round(pca.list$prop_expl_var$X * 100, 2), "%")
    df_X <- pca.list$variates$X %>%
              as.data.frame() %>%
              rownames_to_column("Library_ID") %>%
              inner_join(full_metadata, by = "Library_ID")
    
    color_group = df_X[[color_group]]
    shape_group = df_X[[shape_group]]

    if (pc1 == 'PC1') {
        pc1 <- df_X$PC1
        exp_var_pc1 <- exp_var[1]
        xaxis <- c("PC1")
    }  else if (pc1 == 'PC2') {
        pc1 <- df_X$PC2
        exp_var_pc1 <- exp_var[2]
        xaxis <- c("PC2")
    } else if (pc1 == 'PC3') {
        pc1 <- df_X$PC3
        exp_var_pc1 <- exp_var[3]
        xaxis <- c("PC3")
    }

    if (pc2 == 'PC1') {
        pc2 <- df_X$PC1
        exp_var_pc2 <- exp_var[1]
        yaxis <- c("PC1")
    }  else if (pc2 == 'PC2') {
        pc2 <- df_X$PC2
        exp_var_pc2 <- exp_var[2]
        yaxis <- c("PC2")
    } else if (pc2 == 'PC3') {
        pc2 <- df_X$PC3
        exp_var_pc2 <- exp_var[3]
        yaxis <- c("PC3")
    }

    pca_plot <- ggplot(df_X, aes(pc1, pc2, fill = color_group, shape = shape_group)) +
     geom_point(size = 3) +
     scale_fill_manual(values = c("#5A167E","#C83E73","#E95562","#FED395","#252525","#311068","#FEA873")) + # 
     scale_shape_manual(values = c(24, 21)) +
     # stat_ellipse() +
     xlab(paste(xaxis, " - ", exp_var_pc1)) +
     ylab(paste(yaxis, " - ", exp_var_pc2)) +
     theme_minimal(base_size = 16) +
     theme(text = element_text(size=16)) +
     theme(legend.title = element_blank(),
           legend.key.size = unit(2,"mm"),
           legend.text = element_text(size = 6)) + # 
     theme(legend.position = "top") +
     ggtitle(title_text) + theme(plot.title = element_text(size = 10))

    return(pca_plot)
}

```

```{r}
mpa3_species_decontam_filt_mid <- mpa3_species_decontam %>%
  select(-one_of(outliers_mpa3)) %>%
  select(-one_of(poor_samples_mpa3)) %>%
  select(matches("Species|MID|CMB")) %>%
  adorn_totals(where = "col", name = "Sum_species.decontam") %>%
  mutate(Percent = Sum_species.decontam/sum(Sum_species.decontam)*100) %>%
  filter(Percent > 0.001) %>%
  select(-one_of(c("Sum_species.decontam","Percent"))) %>%
  gather("Library_ID","Counts",2:ncol(.)) %>% 
  mutate(Counts = Counts +1) %>%
  spread(Species,Counts) %>%
  column_to_rownames("Library_ID")

# prepare to run a PCA
# check the number of components to retain by tuning the PCA
mixOmics::tune.pca(mpa3_species_decontam_unfilt_mid, logratio = 'CLR')


malt_otu.pca <- mixOmics::pca(mpa3_species_decontam_unfilt_mid, ncomp = 2, logratio = 'CLR')
malt_pca_values <- malt_otu.pca$variates$X %>%
  as.data.frame() %>%
  rownames_to_column("Library_ID") %>%
  inner_join(., full_metadata, by = "Library_ID")

Pipenotch_colors = c("#311068","#C83E73")
Pipenotch_shapes = c(19,17)

# perform and plot a PCA to see how the data cluster
# by pipe notch presence
mid_pipe_pc12 <- plot_pca_mid(mpa3_species_decontam_unfilt_mid, "PC1", "PC2", "Pipenotch", "Pipenotch", 3, "Pipe notch")
mid_pipe_pc12
# ggsave("~/Desktop/MID_pca_mid_pipe_pc12.pdf", plot = mid_pipe_pc12, device = "pdf",
#        scale = 1, width = 9, height = 5, units = c("in"), dpi = 300)

mpa3_species_decontam_unfilt <- mpa3_species_decontam %>%
  select(-one_of(outliers_mpa3)) %>%
  select(-one_of(poor_samples_mpa3)) %>%
  select(-matches("EXB|LIB|CSN|CSD|ARS|CSS|CSL")) %>%
  adorn_totals(where = "col", name = "Sum_species.decontam") %>%
  mutate(Percent = Sum_species.decontam/sum(Sum_species.decontam)*100) %>%
  filter(Percent > 0.001) %>%
  select(-one_of(c("Sum_species.decontam","Percent"))) %>%
  gather("Library_ID","Counts",2:ncol(.)) %>%
  mutate(Counts = Counts +1) %>%
  spread(Species,Counts) %>%
  column_to_rownames("Library_ID")

# prepare to run a PCA
# check the number of components to retain by tuning the PCA
mixOmics::tune.pca(mpa3_species_decontam_unfilt, logratio = 'CLR')


mpa3_all_otu.pca <- mixOmics::pca(mpa3_species_decontam_unfilt, ncomp = 2, logratio = 'CLR')
mpa3_all_pca_values <- mpa3_all_otu.pca$variates$X %>%
  as.data.frame() %>%
  rownames_to_column("Library_ID") %>%
  inner_join(., full_metadata, by = "Library_ID") %>%
  left_join(., strep_group_info, by = "Library_ID")

site_code_beta <- plot_pca(mpa3_species_decontam_unfilt, "PC1", "PC2","Site_code","Site_code", 3) 
site_code_beta

# ggsave("./06-publication/supplemental_figures/SXX8/site_code_beta.pdf", plot = site_code_beta, device = "pdf",
#        scale = 1, width = 6, height = 5, units = c("in"), dpi = 300)


```

```{r}

library(patchwork)

filt_plot <-  site_code_beta + mid_pipe_pc12 +
  plot_layout(widths = c(1, 1.1))


filt_plot

# ggsave("./06-publication/supplemental_figures/SXX16/filt_plot.pdf", plot = filt_plot, device = "pdf",
#        scale = 1, width = 11, height = 5, units = c("in"), dpi = 300)


```


```{r palaeofeces_gc, eval = F}

palaeofeces <- fread("~/archgen/microbiome_calculus/smoking_calculus/04-analysis/gut_gc/copevo_gc_table.tsv") %>%
  mutate(Sample_type = "palaeofeces") %>%
  filter(str_detect(V1, "_1.gc.tsv")) %>%
  bind_rows(fread("~/archgen/microbiome_calculus/smoking_calculus/04-analysis/gut_gc/pub_pal_gc_list.tsv") %>%
  mutate(Sample_type = "pub_palaeofeces")) %>%
  bind_rows(fread("~/archgen/microbiome_calculus/smoking_calculus/04-analysis/gut_gc/pubData_gc_list.tsv") %>%
  mutate(Sample_type = "pub_mod")) %>%
  bind_rows(fread("~/archgen/microbiome_calculus/smoking_calculus/04-analysis/gut_gc/fta_gc_table.tsv") %>%
  mutate(Sample_type = "fta_cards")) %>%
  rename(Library = 1,
        Avg_GC = 2) %>%
  mutate(Library = str_replace_all(Library, ".gc.tsv", "")) %>%
  separate(Library, into = c("folder","Library_ID"), sep = "/") %>%
  select(-folder)

palaeofeces %>% group_by(Sample_type) %>% summarize(avg_GC = mean(Avg_GC), std_gc = sd(Avg_GC))

```

```{r palaeofeces_plot, eval =  F}

ggplot(palaeofeces, aes(x = Sample_type, y = Avg_GC, fill = Sample_type)) + 
  ## add half-violin from {ggdist} package
  ggdist::stat_halfeye(
    ## custom bandwidth
    adjust = .5, 
    ## adjust height
    width = .6, 
    ## move geom to the right
    justification = -.2, 
    ## remove slab interval
    .width = 0, 
    point_colour = NA
  ) + 
  geom_boxplot(
    width = .12, 
    ## remove outliers
    outlier.color = NA, ## `outlier.shape = NA` works as well
    color = "grey30"
  ) +
  ## add dot plots from {ggdist} package
  ggdist::stat_dots(
    ## orientation to the left
    side = "left", 
    ## move geom to the left
    justification = 1.1, 
    ## adjust grouping (binning) of observations 
    binwidth = 0.005
  ) + 
  ## remove white space on the left
  coord_cartesian(xlim = c(1.2, NA)) +
  theme_minimal(base_size = 14) +
  # scale_fill_manual(values = Time_period_colors) +
  theme(axis.text = element_text(size = 14)) +
  theme(legend.position = "none") +
  xlab("Sample type") +
  ylab("Avg. GC (%)") +
  theme(axis.title.x = element_text(vjust=-1.5)) +
  theme(axis.title.y = element_text(vjust=1.9))

```


