---
title: "MID/CMB, Radcliffe, Kilteasheen, Iberian medieval calculus MALT beta diversity"
author: "Irina Velsko"
date: "`r format(Sys.time(), '%b %d, %Y')`"
output:
    github_document:
        fig_width: 8
        fig_height: 6
        toc: true
---


```{r load_libraries, echo = F, message = F}
library(knitr)
library(data.table)
library(compositions)
library(janitor)
library(tidyverse)
library(gplots)
library(viridis)
library(cowplot)
opts_chunk$set(echo=F, warning=F, message=F, dpi=150)
```

Set the working directory of the notebook to the parent directory of the 
notebook itself.  

```{r setup}
knitr::opts_knit$set(root.dir = normalizePath(".."))
```

# SourceTracker table from MALT shotgun profile

```{r load_data}

# load the taxonomy tables that have been decontaminated and the column headers are cleaned up
load("./05-results.backup/Taxonomy_tables.RData")

# load the metadata file
load("./05-results.backup/Metadata_tables.RData")

```

Read the metadata file and order alphabetically
```{r}
mapfile <- fread("./00-documentation.backup/source_tracker_mappingfile.tsv") %>%
  arrange(`X.SampleID`) %>%
  filter(!str_detect(`X.SampleID`, "ARS006.A0101.SG1.2|MID025.A0101|MID033.A0101|MID056.A0101|MID065.A0101|MID068.A0101|MID076.A0101|MID078.A0101|SRR059454|SRR513449")) %>%
  mutate_all(na_if,"") %>%
  replace(is.na(.), "NA")

# fwrite(mapfile, "./00-documentation.backup/source_tracker_mappingfile.tsv", sep = "\t", quote = T)

```


Read in the species table of sources from the DeepEvo project
```{r}
source_table <- fread("./05-results.backup/Evolution-Comparison_MEGAN_20190410-ex_absolute_species_prokaryotes_summarised_refseq.txt") %>%
  rename(Species = `#Datasets`) %>%
  select(matches("Species|SRR|ERR|ARS|JAE|VLC")) %>%
  adorn_totals(where = "col") %>%
  filter(Total > 0) %>%
  select(-Total)

# fix the column headers
colnames(source_table) <- gsub("_S0_L000_R1_000.fastq.merged.prefixed.hg19unmapped","", colnames(source_table))
colnames(source_table) <- gsub("_S0_L001_R1_001.fastq.combined.fq.prefixed.extractunmapped.bam","", colnames(source_table))
colnames(source_table) <- gsub("_S0_L003_R1_001.fastq.combined.fq.prefixed.extractunmapped.bam","", colnames(source_table))
colnames(source_table) <- gsub("_S0_L001_R1_000.fastq.merged.prefixed.hg19unmapped","", colnames(source_table))

```

Combine the input data table and the source table. This needs to be normalized
by sequencing depth? and CLR-transformed?
```{r}

full_table_sourcetracker <- all_species_decontam %>%
  full_join(., source_table) %>%
  select(-matches("EXB|LIB|CSN|CSL")) %>%
  replace(is.na(.), 0) %>%
  gather("SampleID","Counts", 2:ncol(.)) %>%
  spread(SampleID, Counts)

# fwrite(full_table_sourcetracker, "./05-results.backup/shotgun_sourcetracker_table.txt", sep = "\t", quote = F)


```

log-transform the table for sourcetracker
```{r}
# clr-transform (has negative values)
full_table_sourcetracker_clr <- clr(full_table_sourcetracker %>%
                                      gather("SampleID","Counts", 2:ncol(.)) %>%
                                      mutate(Counts = Counts + 1) %>%
                                      spread(SampleID, Counts) %>%
                                      column_to_rownames("Species"))

full_table_sourcetracker_clr <- full_table_sourcetracker_clr %>%
  as.data.frame.matrix(.)

# log2-transform (no negative values)
full_table_sourcetracker_log <- full_table_sourcetracker %>%
  gather("SampleID","Counts", 2:ncol(.)) %>%
  mutate(Counts = log2(Counts + 1)) %>%
  spread(SampleID, Counts) %>%
  column_to_rownames("Species")

# fwrite(full_table_sourcetracker_log, "./05-results.backup/shotgun_sourcetracker_table_log.txt", sep = "\t", quote = F)

```

```{r}

full_table_sourcetracker %>%
  gather("SampleID","Counts",2:ncol(.)) %>%
  spread(Species, Counts) %>%
  select(SampleID) %>%
  anti_join(., mapfile %>%
              rename(SampleID = `X.SampleID`) %>%
              select(SampleID))
  
mapfile %>%
  rename(SampleID = `X.SampleID`) %>%
  select(SampleID) %>%
  anti_join(., full_table_sourcetracker %>%
    gather("SampleID","Counts",2:ncol(.)) %>%
    spread(Species, Counts) %>%
    select(SampleID))
    
```





































