---
title: "MID/CMB, Radcliffe, Kilteasheen, Iberian medieval calculus MALT beta diversity"
author: "Irina Velsko"
date: "`r format(Sys.time(), '%b %d, %Y')`"
output:
    github_document:
        fig_width: 8
        fig_height: 6
        toc: true
---


```{r load_libraries, echo = F, message = F}
library(knitr)
library(data.table)
library(compositions)
library(janitor)
library(tidyverse)
library(gplots)
library(viridis)
library(cowplot)
opts_chunk$set(echo=F, warning=F, message=F, dpi=150)
```

Set the working directory of the notebook to the parent directory of the 
notebook itself.  

```{r setup}
knitr::opts_knit$set(root.dir = normalizePath(".."))
```

# SourceTracker table from MALT shotgun profile

```{r load_data}

# load the taxonomy tables that have been decontaminated and the column headers are cleaned up
load("./05-results.backup/Taxonomy_tables.RData")

# load the metadata file
load("./05-results.backup/Metadata_tables.RData")

```

Read the metadata file and order alphabetically
```{r, eval = F}
mapfile <- fread("./00-documentation.backup/source_tracker_mappingfile.tsv") %>%
  arrange(`X.SampleID`) %>%
  filter(!str_detect(`X.SampleID`, "ARS006.A0101.SG1.2|MID025.A0101|MID033.A0101|MID056.A0101|MID065.A0101|MID068.A0101|MID076.A0101|MID078.A0101|SRR059454|SRR513449")) %>%
  mutate_all(na_if,"") %>%
  replace(is.na(.), "NA") %>%
  rename(`#SampleID` = `X.SampleID`)

fwrite(mapfile, "./00-documentation.backup/source_tracker_mappingfile.tsv", sep = "\t", quote = F)


# mapfile %>%
#   select(`#SampleID`, BarcodeSequence, LinkerPrimerSequence, Study, Env, SourceSink) %>%
#   fwrite(., "./00-documentation.backup/source_tracker_mappingfile_new.tsv", sep = "\t", quote = T)
  
```

Read in the formatted metadata table
```{r}
mapfile <- fread("./00-documentation.backup/source_tracker_mappingfile.tsv")
```


Read in the species table of sources from the DeepEvo project
```{r}
source_table <- fread("./05-results.backup/Evolution-Comparison_MEGAN_20190410-ex_absolute_species_prokaryotes_summarised_refseq.txt") %>%
  rename(Species = `#Datasets`) %>%
  select(matches("Species|SRR|ERR|ARS|JAE|VLC")) %>%
  adorn_totals(where = "col") %>%
  filter(Total > 0) %>%
  select(-Total)

# fix the column headers
colnames(source_table) <- gsub("_S0_L000_R1_000.fastq.merged.prefixed.hg19unmapped","", colnames(source_table))
colnames(source_table) <- gsub("_S0_L001_R1_001.fastq.combined.fq.prefixed.extractunmapped.bam","", colnames(source_table))
colnames(source_table) <- gsub("_S0_L003_R1_001.fastq.combined.fq.prefixed.extractunmapped.bam","", colnames(source_table))
colnames(source_table) <- gsub("_S0_L001_R1_000.fastq.merged.prefixed.hg19unmapped","", colnames(source_table))

```

Combine the input data table and the source table. This needs to be normalized
by sequencing depth
```{r}

full_table_sourcetracker <- all_species_decontam %>%
  full_join(., source_table) %>%
  select(-matches("EXB|LIB|CSN|CSL")) %>%
  replace(is.na(.), 0) %>%
  gather("SampleID","Counts", 2:ncol(.)) %>%
  spread(SampleID, Counts) %>%
  rename(`#OTU ID` = Species)

```

Create a rarefied table with phyloseq to use in SourceTracker
```{r}
# all_species_decontam is the OTU table
# what is the lowest sequencing depth?
full_table_sourcetracker %>%
  gather("Library_ID","Counts", 2:ncol(.)) %>%
  spread(`#OTU ID`, Counts) %>%
  adorn_totals(where = "col") %>%
  select(Library_ID, Total) %>%
  arrange(Total) %>%
  as_tibble()

# Make the table a matrix
st_table <- full_table_sourcetracker %>%
  column_to_rownames("#OTU ID") %>%
  as.matrix(.)

# make an accompanying taxonomy matrix
st_taxa <- full_table_sourcetracker %>%
  mutate(Species_2 = `#OTU ID`) %>%
  select(`#OTU ID`, Species_2) %>%
  column_to_rownames("#OTU ID") %>%
  rename(Species = Species_2) %>%
  as.matrix(.)

# now convert to a phyloseq object
library("phyloseq")
OTU = otu_table(st_table, taxa_are_rows = TRUE)
TAX = tax_table(st_taxa)

st_physeq = phyloseq(OTU, TAX)
st_physeq

rare_st_physeq <- rarefy_even_depth(st_physeq, rngseed = 14)

# save the rarefied table (from https://github.com/joey711/phyloseq/issues/613)
# Extract abundance matrix from the phyloseq object
OTU1 <- as(otu_table(rare_st_physeq), "matrix")
# transpose if necessary
if(taxa_are_rows(rare_st_physeq)){OTU1 <- t(OTU1)}
# Coerce to data.frame
st_table_rare <- as.data.frame(OTU1)

# check the depth of rarefaction
st_table_rare %>%
  rownames_to_column("#OTU ID") %>%
  adorn_totals(where = "col") %>%
  select(`#OTU ID`,Total) %>%
  as_tibble()

# make the rownames a column with the appropriate header
st_table_rare <- st_table_rare %>%
  rownames_to_column("SampleID") %>%
  gather("#OTU ID","Counts",2:ncol(.)) %>%
  spread(SampleID, Counts)


```


```{r}
# save the rarefied table for input to sourcetracker
fwrite(st_table_rare, "./05-results.backup/shotgun_sourcetracker_table_noblanks_rarefied.txt", sep = "\t", quote = F)

```

log-transform the table for sourcetracker
```{r, eval = F}
# clr-transform (has negative values)
full_table_sourcetracker_clr <- clr(full_table_sourcetracker %>%
                                      gather("SampleID","Counts", 2:ncol(.)) %>%
                                      mutate(Counts = Counts + 1) %>%
                                      spread(SampleID, Counts) %>%
                                      column_to_rownames("Species"))

full_table_sourcetracker_clr <- full_table_sourcetracker_clr %>%
  as.data.frame.matrix(.)

# log2-transform (no negative values)
full_table_sourcetracker_log <- full_table_sourcetracker %>%
  gather("SampleID","Counts", 2:ncol(.)) %>%
  mutate(Counts = log2(Counts + 1)) %>%
  spread(SampleID, Counts) %>%
  column_to_rownames("Species")

# fwrite(full_table_sourcetracker_log, "./05-results.backup/shotgun_sourcetracker_table_log.txt", sep = "\t", quote = F)

```

```{r}

st_table_rare %>%
  gather("SampleID","Counts",2:ncol(.)) %>%
  spread(Species, Counts) %>%
  select(SampleID) %>%
  anti_join(., mapfile %>%
              rename(SampleID = `#SampleID`) %>%
              select(SampleID))
  
mapfile %>%
  rename(SampleID = `#SampleID`) %>%
  select(SampleID) %>%
  anti_join(., st_table_rare %>%
    gather("SampleID","Counts",2:ncol(.)) %>%
    spread(Species, Counts) %>%
    select(SampleID))
    
```




































