---
title: "MID/CMB, Radcliffe, Kilteasheen, Iberian Medieval, calculus MetaPhlAn3 beta diversity"
author: "Irina Velsko"
date: "`r format(Sys.time(), '%b %d, %Y')`"
output:
    github_document:
        fig_width: 8
        fig_height: 6
        toc: true
---


```{r load_libraries, message = F}
library(knitr)
library(data.table)
library(psych)
library(ape)
library(mixOmics)
library(compositions)
library(vegan)
library(pairwiseAdonis)
library(janitor)
library(pander)
library(tidyverse)
library(gplots)
library(ggrepel)
library(viridis)
library(cowplot)
opts_chunk$set(echo=F, warning=F, message=F, dpi=150)
```

Set the working directory of the notebook to the parent directory of the 
notebook itself.  

```{r setup}
knitr::opts_knit$set(root.dir = normalizePath("../../../"))
```

# MetaPhlAn3

```{r load_data}
# read in the species table
mpa3_raw <- fread("./05-results.backup/mpa3_abundance_table.tsv") %>%
  select(-NCBI_tax_id)

# clean the file names
colnames(mpa3_raw) <- gsub(".metaphlan3","", colnames(mpa3_raw))
colnames(mpa3_raw) <- gsub(".SG1","", colnames(mpa3_raw))

# filter to have only species
mpa3_sp <- mpa3_raw %>% 
  filter(str_detect(clade_name, "s__")) %>%
  separate(clade_name, into = c("K","P","C","O","F","G","Species"), sep = "\\|") %>%
  select(-c("K","P","C","O","F","G")) %>%
  mutate(Species = str_replace_all(Species, "s__",""),
         Species = str_replace_all(Species, "_"," ")) %>%
  # these samples have no metadata
  select(-c("MID025.A0101","MID033.A0101","MID052.A0101","MID056.A0101","MID065.A0101","MID068.A0101","MID076.A0101","MID078.A0101")) %>%
  arrange(Species)

```


Now decontam the tables
```{r}
mpa3_contaminants <- fread("./05-results.backup/contaminant_species_mid_mpa3.tsv", sep = "\t") %>%
  bind_rows(., fread("./05-results.backup/contaminant_species_kilteasheen_mpa3.tsv", sep = "\t")) %>%
  bind_rows(., fread("./05-results.backup/contaminant_species_iberia_mpa3.tsv", sep = "\t")) %>%
  unique()

# remove contaminant species from nt table
mpa3_species_decontam <- mpa3_sp %>%
  anti_join(., mpa3_contaminants)

```

```{r load_strep_groups}
strep_groups <- fread("./00-documentation.backup/25-streptococcus_cladegroup_amylasegroup_database.tsv") %>%
  select(Taxon, Clade_Consensus) %>%
  rename(Species = Taxon) %>%
  mutate(Species = str_replace_all(Species, "_", " ")) %>%
  rename(Strep_group = Clade_Consensus) %>%
  mutate(Strep_group = fct_relevel(Strep_group, "sanguinis","mitis","salivarius","anginosus","bovis","pyogenic","mutans","unknown","NA"))

# get the proportion of sanguinis and anginosis for each sample
strep_group_info <- mpa3_species_decontam %>%
  filter(str_detect(Species, "Streptococcus")) %>%
  # join with the Strep grup table for plotting
  inner_join(., strep_groups) %>%
  select(-Species) %>%
  select(Strep_group, everything()) %>%
  # transpose
  gather("Library_ID","Counts", 2:ncol(.)) %>%
  group_by(Library_ID, Strep_group) %>%
  summarize(Reads = sum(Counts)) %>%
  spread(Strep_group, Reads) %>%
  # convert table to proportions
  adorn_percentages(denominator = "row") %>%
  # arrange by sanguinis and for the ones w/o sanguinis, by anginosis
  arrange(desc(sanguinis))

```

```{r load_metadata}

# load the metadata file
load("./05-results.backup/Metadata_tables.RData")

# read in the MALT mapped/unmapped gc and read length info from `MID_malt_map_unmap_rl_gc.Rmd`
malt_map_unmap_rl_gc <- fread("./05-results.backup/rl_gc_sub_data.tsv")
colnames(malt_map_unmap_rl_gc) <- gsub("_map","_malt_map", colnames(malt_map_unmap_rl_gc))
colnames(malt_map_unmap_rl_gc) <- gsub("_unmap","_malt_unmap", colnames(malt_map_unmap_rl_gc))

# full_metadata <- full_metadata %>%
#   mutate(`DNA_molecules_per_library_x10^6_log` = log10(`DNA_molecules_per_library_x10^6`)) %>%
#   full_join(., malt_map_unmap_rl_gc) %>%
#   full_join(., strep_groups, by = "Library_ID")

```


```{r pca_plot_fxn}
# plot PCA with colored dots and the title including the # of species or genera
plot_pca <- function(df, pc1, pc2, color_group, shape_group, ncomps) {
    metadata_group_colors <- get(paste(color_group, "_colors", sep = ""))
    metadata_group_shapes <- get(paste(shape_group, "_shapes", sep = ""))

    pca.list <- mixOmics::pca(df, ncomp = ncomps, logratio = 'CLR')

    exp_var <- paste0(round(pca.list$prop_expl_var$X * 100, 2), "%")
    df_X <- pca.list$variates$X %>%
              as.data.frame() %>%
              rownames_to_column("Library_ID") %>%
              inner_join(full_metadata, by = "Library_ID") %>%
              inner_join(., strep_group_info)
              
    color_group = df_X[[color_group]]
    shape_group = df_X[[shape_group]]

    if (pc1 == 'PC1') {
        pc1 <- df_X$PC1
        exp_var_pc1 <- exp_var[1]
        xaxis <- c("PC1")
    }  else if (pc1 == 'PC2') {
        pc1 <- df_X$PC2
        exp_var_pc1 <- exp_var[2]
        xaxis <- c("PC2")
    } else if (pc1 == 'PC3') {
        pc1 <- df_X$PC3
        exp_var_pc1 <- exp_var[3]
        xaxis <- c("PC3")
    }

    if (pc2 == 'PC1') {
        pc2 <- df_X$PC1
        exp_var_pc2 <- exp_var[1]
        yaxis <- c("PC1")
    }  else if (pc2 == 'PC2') {
        pc2 <- df_X$PC2
        exp_var_pc2 <- exp_var[2]
        yaxis <- c("PC2")
    } else if (pc2 == 'PC3') {
        pc2 <- df_X$PC3
        exp_var_pc2 <- exp_var[3]
        yaxis <- c("PC3")
    }

    pca_plot <- ggplot(df_X, aes(pc1, pc2)) +
     geom_point(aes(, fill = color_group, shape = shape_group), size = 2.5, stroke = 0.3) +
     scale_fill_manual(values = metadata_group_colors) +
     scale_shape_manual(values = metadata_group_shapes) +
     # stat_ellipse() +
     xlab(paste(xaxis, " - ", exp_var_pc1)) +
     ylab(paste(yaxis, " - ", exp_var_pc2)) +
     theme_minimal(base_size = 16) +
     theme(text = element_text(size=16)) +
     theme(legend.title = element_blank(),
           legend.key.size = unit(2,"mm"),
           legend.text = element_text(size = 6)) +
     theme(legend.position = "top")

    return(pca_plot)
}

```


```{r pca_plot_cont_fxn}
# for continuous data
plot_pca_cont <- function(df, pc1, pc2, color_group, shape_group, ncomps, title_text) {
    metadata_group_shapes <- get(paste(shape_group, "_shapes", sep = ""))

    pca.list <- mixOmics::pca(df, ncomp = ncomps, logratio = 'CLR')

    exp_var <- paste0(round(pca.list$prop_expl_var$X * 100, 2), "%")
    df_X <- pca.list$variates$X %>%
              as.data.frame() %>%
              rownames_to_column("Library_ID") %>%
              inner_join(full_metadata, by = "Library_ID") %>%
              inner_join(., strep_group_info, by = "Library_ID")
    # %>%
              # full_join(., full_source_results %>%
              #             rename(Library_ID = SampleID),  by = "Library_ID")
    
    color_group = df_X[[color_group]]
    shape_group = df_X[[shape_group]]

    if (pc1 == 'PC1') {
        pc1 <- df_X$PC1
        exp_var_pc1 <- exp_var[1]
        xaxis <- c("PC1")
    }  else if (pc1 == 'PC2') {
        pc1 <- df_X$PC2
        exp_var_pc1 <- exp_var[2]
        xaxis <- c("PC2")
    } else if (pc1 == 'PC3') {
        pc1 <- df_X$PC3
        exp_var_pc1 <- exp_var[3]
        xaxis <- c("PC3")
    }

    if (pc2 == 'PC1') {
        pc2 <- df_X$PC1
        exp_var_pc2 <- exp_var[1]
        yaxis <- c("PC1")
    }  else if (pc2 == 'PC2') {
        pc2 <- df_X$PC2
        exp_var_pc2 <- exp_var[2]
        yaxis <- c("PC2")
    } else if (pc2 == 'PC3') {
        pc2 <- df_X$PC3
        exp_var_pc2 <- exp_var[3]
        yaxis <- c("PC3")
    }

    pca_plot <- ggplot(df_X, aes(pc1, pc2, fill = color_group, shape = shape_group)) +
     geom_point(size = 2.5, stroke = 0.3) +
     scale_fill_viridis_c(option = "C") +
     scale_shape_manual(values = metadata_group_shapes) +
     # stat_ellipse() +
     xlab(paste(xaxis, " - ", exp_var_pc1)) +
     ylab(paste(yaxis, " - ", exp_var_pc2)) +
     theme_minimal(base_size = 16) +
     theme(text = element_text(size=16)) +
     theme(legend.title = element_blank(),
           legend.key.size = unit(2,"mm"),
           legend.text = element_text(size = 6)) + 
     theme(legend.position = "top") +
     ggtitle(title_text) + theme(plot.title = element_text(size = 10))

    return(pca_plot)
}


```

```{r pca_biplot_fxn}
plot_pca_bi <- function(df, pc1, pc2, metadata_group, columntitle) {
    metadata_group_colors <- get(paste(metadata_group, "_colors", sep = ""))
    metadata_group_shapes <- get(paste(metadata_group, "_shapes", sep = ""))
   
    arrow_pc <- enquo(columntitle)
    
    exp_var <- paste0(round(df$prop_expl_var$X * 100, 2), "%") # explained variance for x- and y-labels
    
    # select only the PCs from the PCA and add metadata
    df_X <- df$variates$X %>%
              as.data.frame() %>%
              rownames_to_column("Library_ID") %>%
              inner_join(full_metadata, by = "Library_ID")
    
    # metadata_group = df_X[[metadata_group]]
    
    corr_lam <- df$sdev[c("PC1", "PC2", "PC3")] * sqrt(nrow(df_X))

    df_X <- df_X %>%
      mutate(PC1 = PC1 / corr_lam[1],
             PC2 = PC2 / corr_lam[2], 
             PC3 = PC3 / corr_lam[3])

    df_Y <- df_X[, c(pc1,  pc2, metadata_group)]
    # select the correct PC column and explained variance for PC1
    xaxis <- pc1
    yaxis <- pc2
    exp_var_pc1 <- exp_var[match(pc1, c("PC1", "PC2", "PC3"))]
    exp_var_pc2 <- exp_var[match(pc2, c("PC1", "PC2", "PC3"))]
    
    # Identify the 10 pathways that have highest positive and negative loadings in the selected PC
    pws_10 <- df$loadings$X %>%
      as.data.frame(.) %>%
      rownames_to_column(var = "Taxon") %>%
      top_n(10, !!arrow_pc)

    neg_10 <- df$loadings$X %>%
      as.data.frame(.) %>%
      rownames_to_column(var = "Taxon") %>%
      top_n(-10, !!arrow_pc)


    pca_plot_bi <- ggplot(df_Y, aes(x = get(paste(pc1)), y = get(paste(pc2)))) +
      geom_point(size = 2.5, aes(shape = get(paste(metadata_group)), fill = get(paste(metadata_group)))) +
      geom_segment(data = pws_10,
                   aes(xend = get(paste(pc1)), yend = get(paste(pc2))),
                   x = 0, y = 0, colour = "black",
                   size = 0.5,
                   arrow = arrow(length = unit(0.03, "npc"))) +
      geom_label_repel(data = pws_10,
                   aes(x = get(paste(pc1)), y = get(paste(pc2)), label = Taxon),
                   size = 2.5, colour = "grey20", label.padding = 0.2, force = 5, max.overlaps = 14) +
      geom_segment(data = neg_10,
                   aes(xend = get(paste(pc1)), yend = get(paste(pc2))),
                   x = 0, y = 0, colour = "grey50",
                   size = 0.5,
                   arrow = arrow(length = unit(0.03, "npc"))) +
      geom_label_repel(data = neg_10,
                   aes(x = get(paste(pc1)), y = get(paste(pc2)), label = Taxon),
                   size = 2.5, colour = "grey20", label.padding = 0.2, max.overlaps = 14) +
      labs(x = paste(xaxis, " - ", exp_var_pc1),
           y = paste(yaxis, " - ", exp_var_pc2)) +
      scale_fill_manual(values = metadata_group_colors) +
      # scale_shape_manual(values = c(21, 22, 23, 24, 21, 22, 23, 24)) +
      scale_shape_manual(values = metadata_group_shapes) +
      theme_minimal() + theme(text = element_text(size = 16)) +
      theme(text = element_text(size=16)) +
      theme(legend.position = "top")

    return(pca_plot_bi)
}

# mpa3_species_decontam_filtered_nocontrols_biplot_122 <- plot_pca_bi(mpa3_species_decontam_unfilt_all.pca, "PC1", "PC2", "Site_code", PC1)
# mpa3_species_decontam_filtered_nocontrols_biplot_122


```


List the outliers from the plot in `MID_CMB_Rad_Kil_Iber_decontam.Rmd`. 
```{r outliers}
outliers_mpa3 <- c("EXB059.A2101","EXB059.A2501","EXB015.A3301","EXB034.A2701","EXB059.A2201","EXB059.A2301","EXB059.A2401","LIB058.A0103","LIB058.A0106","LIB058.A0104")
poor_samples_mpa3 <- fread("./05-results.backup/cuperdec_mpa3_poor_samples.tsv") %>%
  pull(Sample)

outliersF <- str_c(outliers_mpa3, collapse = "|")

```


#Sample Clustering with PCA

Step-by-step removal of taxa based on decontam and a filter threshold, followed
by PCA to see how contaminants/low abundance species affect the relationships of
samples to eachother between groups.
```{r pca_all_species}
# decontaminated only
mpa3_species_decontam_unfilt <- mpa3_species_decontam %>%
  select(-one_of(outliers_mpa3)) %>%
  select(-one_of(poor_samples_mpa3)) %>%
  select(-matches("ARS|CSS|CSL")) %>%
  gather("Library_ID","Counts",2:ncol(.)) %>%
  mutate(Counts = Counts +1) %>%
  spread(Species,Counts) %>%
  column_to_rownames("Library_ID")

# prepare to run a PCA
# check the number of components to retain by tuning the PCA
tune.pca(mpa3_species_decontam_unfilt, logratio = 'CLR')

# define Time Period shapes for plotting
Time_period_shapes <- c(Industrial = 21, Medieval = 24, Modern = 22, Extraction_blank = 23, Library_blank = 23)

```


# Species-level analyses
What is the relationship of the groups in a PCA without the blanks? Let's try
the PCA using only the species that are present in at least 10% of at least one
sample group (MID, CMB, Radcliffe, Kilteasheen, Iberia)

```{r}

# decontaminated only
mpa3_species_decontam_unfilt <- mpa3_species_decontam %>%
  select(-one_of(outliers_mpa3)) %>%
  select(-one_of(poor_samples_mpa3)) %>%
  select(-matches("EXB|LIB|CSN|CSD|ARS|CSS|CSL")) %>%
  gather("Library_ID","Counts",2:ncol(.)) %>%
  mutate(Counts = Counts +1) %>%
  spread(Species,Counts) %>%
  column_to_rownames("Library_ID")

# prepare to run a PCA
# check the number of components to retain by tuning the PCA
tune.pca(mpa3_species_decontam_unfilt, logratio = 'CLR')

# define Time Period shapes for plotting
Time_period_shapes <- c(Industrial = 21, Medieval = 24, Modern = 22, Extraction_blank = 23, Library_blank = 23)

```

```{r pca_time_period}
time_period <- plot_pca(mpa3_species_decontam_unfilt, "PC1", "PC2","Time_period","Time_period", 3) 
time_period

```

```{r pca_site}
plot_pca(mpa3_species_decontam_unfilt, "PC1", "PC2","Site_code","Time_period", 3) 

```

```{r pca_reads}
# by total reads into MetaPhlAn3 
seqdepth_pc12 <- plot_pca_cont(mpa3_species_decontam_unfilt, "PC1", "PC2", "No_reads_total", "Time_period", 3, "Total reads into MetaPhlAn3")
seqdepth_pc12

```

```{r pca_gc}
# by average GC content
gc <- plot_pca_cont(mpa3_species_decontam_unfilt, "PC1", "PC2", "gc_avg", "Time_period", 3, "Avg. GC content (%)")
gc

```

```{r pca_rl}
# by average read length
rl <- plot_pca_cont(mpa3_species_decontam_unfilt, "PC1", "PC2", "seq_len_avg", "Time_period", 3, "Avg. read length (bp)")
rl

```

```{r biplots_all_species}

mpa3_species_decontam_unfilt_all <- mpa3_species_decontam %>%
  select(-one_of(outliers_mpa3)) %>%
  select(-one_of(poor_samples_mpa3)) %>%
  select(-matches("EXB|LIB|CSN|CSD|ARS|CSS|CSL")) %>%
  gather("Library_ID","Counts",2:ncol(.)) %>% 
  mutate(Counts = Counts +1) %>%
  spread(Species,Counts) %>%
  column_to_rownames("Library_ID")

mpa3_species_decontam_unfilt_all.pca <- mixOmics::pca(mpa3_species_decontam_unfilt_all, ncomp = 3, logratio = 'CLR')
plot(mpa3_species_decontam_unfilt_all.pca)

# Define Time Period shapes
Time_period_shapes <- c(Industrial = 21, Medieval = 24, Modern = 22, Extraction_blank = 23, Library_blank = 23)

mpa3_species_decontam_filtered_nocontrols_biplot_121 <- plot_pca_bi(mpa3_species_decontam_unfilt_all.pca, "PC1", "PC2", "Time_period", PC1)
mpa3_species_decontam_filtered_nocontrols_biplot_121
# plot_pca_bi(mpa3_species_decontam_unfilt_no_Moralis.pca, "PC1", "PC2", "Time_period", PC1)

mpa3_species_decontam_unfilt_all.pca$variates$X %>%
  as.data.frame() %>%
  rownames_to_column("Library_ID") %>%
  inner_join(full_metadata, by = "Library_ID") %>% select(Library_ID, PC1, PC2, Site_code)



# Define site code shapes
Site_code_shapes <- c(MID = 21, CMB = 21, RAD = 21, KIL = 24, ELR = 24, IVE = 24, JAE = 22, VLC = 22)


mpa3_species_decontam_filtered_nocontrols_biplot_122 <- plot_pca_bi(mpa3_species_decontam_unfilt_all.pca, "PC1", "PC2", "Site_code", PC1)
mpa3_species_decontam_filtered_nocontrols_biplot_122

pandoc.table(mpa3_species_decontam_filtered_nocontrols_biplot_121$plot_env$pws_10 %>% arrange(desc(PC1)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC1 positive loadings")

pandoc.table(mpa3_species_decontam_filtered_nocontrols_biplot_121$plot_env$neg_10 %>% arrange(PC1),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC1 negative loadings")

mpa3_species_decontam_filtered_nocontrols_biplot_122 <- plot_pca_bi(mpa3_species_decontam_unfilt_all.pca, "PC1", "PC2", "Site_code", PC2)
mpa3_species_decontam_filtered_nocontrols_biplot_122

pandoc.table(mpa3_species_decontam_filtered_nocontrols_biplot_122$plot_env$pws_10 %>% arrange(desc(PC2)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC2 positive loadings")

pandoc.table(mpa3_species_decontam_filtered_nocontrols_biplot_122$plot_env$neg_10 %>% arrange(PC2),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC2 negative loadings")

```
```{r no_methanobrevibacter}
# remove Methanobrevibacter oralis from the species table
mpa3_species_decontam_unfilt_no_Moralis <- mpa3_species_decontam %>%
  filter(Species != "Methanobrevibacter oralis") %>%
  select(-one_of(outliers_mpa3)) %>%
  select(-one_of(poor_samples_mpa3)) %>%
  select(-matches("EXB|LIB|CSN|CSD|ARS|CSS|CSL")) %>%
  gather("Library_ID","Counts",2:ncol(.)) %>%
  mutate(Counts = Counts +1) %>%
  spread(Species,Counts) %>%
  column_to_rownames("Library_ID")


# perform and plot a PCA to see how the data cluster
nomethano_beta <- plot_pca(mpa3_species_decontam_unfilt_no_Moralis, "PC1", "PC2", "Site_code", "Time_period", 3)
nomethano_beta

# by average GC content
nomethano_gc <- plot_pca_cont(mpa3_species_decontam_unfilt_no_Moralis, "PC1", "PC2", "gc_avg", "Time_period", 3, "Avg. GC content (%)")
nomethano_gc

# by average read length
plot_pca_cont(mpa3_species_decontam_unfilt_no_Moralis, "PC1", "PC2", "seq_len_avg", "Time_period", 3, "Avg. read length (bp)")

# biplot to see strongest loading species
mpa3_species_decontam_unfilt_no_Moralis.pca <- mixOmics::pca(mpa3_species_decontam_unfilt_no_Moralis, ncomp = 3, logratio = 'CLR')
plot(mpa3_species_decontam_unfilt_no_Moralis.pca)

# define shapes
Site_code_shapes <- c(MID = 21, CMB = 21, RAD = 21, KIL = 24, ELR = 24, IVE = 24, JAE = 22, VLC = 22, EXB = 25, LIB = 25)

nomethano_bi <- plot_pca_bi(mpa3_species_decontam_unfilt_no_Moralis.pca, "PC1", "PC2", "Site_code", PC1)
nomethano_bi

no_methano <- plot_grid(nomethano_beta, nomethano_gc, nomethano_bi,
          ncol = 1, labels = c("A", "B","C"), rel_widths = c(1), scale = 0.95)
no_methano

# ggsave("./06-publication/supplemental_figures/SXX5/no_methano.pdf", plot = no_methano,
#        device = "pdf", scale = 1, width = 6, height = 15, units = c("in"), dpi = 300)


```

Check the GC content of the species w/strongest loadings in PC2+ and PC2- space
```{r}

# read in the file w/avg GC content of each species, taken from NCBI Genome
pc2_gc <- fread("./05-results.backup/pc2_gc_avg.tsv") %>%
  mutate(Order = c(rep(1:10, times = 2)))
  # mutate(Order = c(rep(1:10)), c(rep(1:10)))

# plot
ggplot(pc2_gc, aes(x = PC2, y = Avg_GC, color = Order)) +
  geom_point() +
  theme_minimal() +
  scale_color_viridis_c(option = "C") +
  xlab("PC2 direction") +
  ylab("Species avg. GC (%)")



rain <- ggplot(pc2_gc, aes(x = PC2, y = Avg_GC, fill = PC2)) + 
  ## add half-violin from {ggdist} package
  ggdist::stat_halfeye(
    ## custom bandwidth
    adjust = .5, 
    ## adjust height
    width = .6, 
    ## move geom to the right
    justification = -.2, 
    ## remove slab interval
    .width = 0, 
    point_colour = NA
  ) + 
  geom_boxplot(
    width = .12, 
    ## remove outliers
    outlier.color = NA, ## `outlier.shape = NA` works as well
    color = "grey30"
  ) +
  ## add dot plots from {ggdist} package
  ggdist::stat_dots(
    ## orientation to the left
    side = "left", 
    ## move geom to the left
    justification = 1.1, 
    ## adjust grouping (binning) of observations 
    binwidth = 1.0
  ) + 
  ## remove white space on the left
  coord_cartesian(xlim = c(1.2, NA)) +
  theme_minimal(base_size = 14) +
  # scale_fill_manual(values = Time_period_colors) +
  theme(axis.text = element_text(size = 14)) +
  theme(legend.position = "none") +
  xlab("PC2 direction") +
  ylab("Species avg. GC (%)") +
  theme(axis.title.x = element_text(vjust=-1.5)) +
  theme(axis.title.y = element_text(vjust=1.9)) +
  scale_fill_manual(values = c("#311068", "#C83E73"))

rain

```

```{r plot_together}

beta_div <- plot_grid(time_period, mpa3_species_decontam_filtered_nocontrols_biplot_121, 
                         gc, rain,
          nrow=2, labels = c("A", "B","C","D"), rel_widths = c(1, 1), scale = 0.95)
beta_div

ggsave("./06-publication/main_figures/Figure_XX5/beta_div_all.pdf", plot = beta_div,
       device = "pdf", scale = 1, width = 11, height = 8.5, units = c("in"), dpi = 300)

```

Run a permanova with function adonis to test if the clusters are significantly
different. First load the metadata again so the blanks and Yanomami samples can
be removed and then make them factors.
```{r metadata_betadisper}

metadata_noblanks <- fread("./00-documentation.backup/radcliffe_pfuT_metadata.tsv") %>%
  bind_rows(., fread("./00-documentation.backup/iberian_medieval_metadata.tsv")) %>%
  bind_rows(., fread("./00-documentation.backup/kilteasheen_metadata.tsv")) %>%
  bind_rows(., fread("./00-documentation.backup/mod_metadata.tsv")) %>%
  full_join(., fread("./00-documentation.backup/MID_analysis_metadata.tsv"))


metadata_noblanks <- metadata_noblanks %>%
  as_tibble() %>%
  filter(!Library_ID %in% outliers_mpa3) %>%
  filter(!Library_ID %in% poor_samples_mpa3) %>%
  filter(!str_detect(Library_ID, "CSN|CSL|CSS|CSD|EXB|LIB")) %>%
  # remove the MID samples that have no metadata
  filter(!str_detect(Library_ID, "MID025.A0101|MID033.A0101|MID056.A0101|MID065.A0101|MID068.A0101|MID076.A0101|MID078.A0101")) %>%
  mutate(Site = fct_relevel(Site_code, "MID", "CMB", "RAD", "KIL",
                                   "ELR", "IVE", "JAE","VLC"),
         Time_period = fct_relevel(Time_period, "Industrial","Medieval")) %>%
  select(Library_ID, Site_code, Time_period) %>%
  arrange(Library_ID) %>%
  unique()

```

Run a PERMANOVA with adonis2 on the abundance-filtered clr-transformed species matrix.
```{r adonis_noblanks_species}
# Euclidean distance, uses the clr-transformed matrix from the mixOmics PCA above
mpa3_species_decontam_unfilt_all.euc <- vegdist(mpa3_species_decontam_unfilt_all.pca$X, method = "euclidean", na.rm = T)
mpa3_species_decontam_unfilt_all.euc.pcoa <- ape::pcoa(mpa3_species_decontam_unfilt_all.euc)

mpa3_species_decontam_unfilt_all.euc.pcoavectors <- as.data.frame(mpa3_species_decontam_unfilt_all.euc.pcoa$vectors)
mpa3_species_decontam_unfilt_all.euc.pcoavectors <- mpa3_species_decontam_unfilt_all.euc.pcoavectors %>%
  rownames_to_column("Library_ID")

mpa3_species_decontam_unfilt_all.euc.vecmet <- inner_join(mpa3_species_decontam_unfilt_all.euc.pcoavectors, full_metadata, by = "Library_ID")

mpa3_species_decontam_unfilt_all.euc.vecmet %>%
  ggplot(., aes(Axis.1, Axis.2, colour = Time_period, shape = Time_period)) +
  geom_point(size = 3) +
  scale_colour_manual(values = Time_period_colors) +
  scale_shape_manual(values = Time_period_shapes) +
  stat_ellipse(aes(colour = Time_period, group = Time_period)) +
  theme_minimal(base_size = 7) +
  scale_y_reverse() +
  scale_x_reverse() +
  ggtitle("Time_period")

metadata_noblanks %>% select(Library_ID) %>% anti_join(., mpa3_species_decontam_unfilt_all.euc.pcoavectors %>% select(Library_ID))

# test for difference between different metadata categories 
adonis2(mpa3_species_decontam_unfilt_all.euc ~ Site_code + Time_period, metadata_noblanks, permutations = 999, by = "margin")
adonis2(mpa3_species_decontam_unfilt_all.euc ~ Time_period, metadata_noblanks, permutations = 999, by = "margin")


# pairwise adonis2 to see which groups are different
# pa2in <- mpa3_species_decontam_unfilt_all.pca$X %>% as.data.frame.matrix() 
# pairwise.adonis2(mpa3_species_decontam_unfilt_all.euc ~ Time_period, metadata_noblanks, strata = 'Time_period')

```

```{r pairwise_adonis}
library(funfuns)

x <- mpa3_species_decontam_unfilt_all  %>% rownames_to_column("Library_ID") %>% select(-Library_ID)
#   pivot_longer(!Library_ID, names_to = "Taxa", values_to = "Counts") %>% 
#   pivot_wider(names_from = "Library_ID", values_from = "Counts") %>%
#   column_to_rownames("Taxa")

factors <- metadata_noblanks %>% select(Time_period) %>% mutate(Time_period = fct_relevel(Time_period, "Medieval", "Industrial","Modern")) %>% pull()
 
pwa_time_period <- pairwise.adonis(
  x,
  factors,
  sim.method = "euclidean",
  p.adjust.m = "fdr",
  permutations = 999
)

factors <- metadata_noblanks %>% select(Site_code) %>% mutate(Site_code = fct_relevel(Site_code, "MID", "CMB","RAD","KIL","ELR","IVE","JAE","VLC"))  %>% pull()

pwa_site <- pairwise.adonis(
  x,
  factors,
  sim.method = "euclidean",
  p.adjust.m = "fdr",
  permutations = 999
)

pwa_time_period
pwa_site %>%
  arrange(p.adjusted)

```

```{r palaeofeces_gc}

palaeofeces <- fread("~/archgen/microbiome_calculus/smoking_calculus/04-analysis/gut_gc/copevo_gc_table.tsv") %>%
  mutate(Sample_type = "palaeofeces") %>%
  filter(str_detect(V1, "_1.gc.tsv")) %>%
  bind_rows(fread("~/archgen/microbiome_calculus/smoking_calculus/04-analysis/gut_gc/pub_pal_gc_list.tsv") %>%
  mutate(Sample_type = "pub_palaeofeces")) %>%
  bind_rows(fread("~/archgen/microbiome_calculus/smoking_calculus/04-analysis/gut_gc/pubData_gc_list.tsv") %>%
  mutate(Sample_type = "pub_mod")) %>%
  bind_rows(fread("~/archgen/microbiome_calculus/smoking_calculus/04-analysis/gut_gc/fta_gc_table.tsv") %>%
  mutate(Sample_type = "fta_cards")) %>%
  rename(Library = 1,
        Avg_GC = 2) %>%
  mutate(Library = str_replace_all(Library, ".gc.tsv", "")) %>%
  separate(Library, into = c("folder","Library_ID"), sep = "/") %>%
  select(-folder)

palaeofeces %>% group_by(Sample_type) %>% summarize(avg_GC = mean(Avg_GC), std_gc = sd(Avg_GC))

```

```{r palaeofeces_plot}

ggplot(palaeofeces, aes(x = Sample_type, y = Avg_GC, fill = Sample_type)) + 
  ## add half-violin from {ggdist} package
  ggdist::stat_halfeye(
    ## custom bandwidth
    adjust = .5, 
    ## adjust height
    width = .6, 
    ## move geom to the right
    justification = -.2, 
    ## remove slab interval
    .width = 0, 
    point_colour = NA
  ) + 
  geom_boxplot(
    width = .12, 
    ## remove outliers
    outlier.color = NA, ## `outlier.shape = NA` works as well
    color = "grey30"
  ) +
  ## add dot plots from {ggdist} package
  ggdist::stat_dots(
    ## orientation to the left
    side = "left", 
    ## move geom to the left
    justification = 1.1, 
    ## adjust grouping (binning) of observations 
    binwidth = 0.005
  ) + 
  ## remove white space on the left
  coord_cartesian(xlim = c(1.2, NA)) +
  theme_minimal(base_size = 14) +
  # scale_fill_manual(values = Time_period_colors) +
  theme(axis.text = element_text(size = 14)) +
  theme(legend.position = "none") +
  xlab("Sample type") +
  ylab("Avg. GC (%)") +
  theme(axis.title.x = element_text(vjust=-1.5)) +
  theme(axis.title.y = element_text(vjust=1.9))

```

