---
title: "MID/CMB, Radcliffe, Kilteasheen, Iberian Medieval, calculus MetaPhlAn3 beta diversity"
author: "Irina Velsko"
date: "`r format(Sys.time(), '%b %d, %Y')`"
output:
    github_document:
        fig_width: 8
        fig_height: 6
        toc: true
---


```{r load_libraries, message = F}
library(knitr)
library(data.table)
library(psych)
library(ape)
library(mixOmics)
library(compositions)
library(vegan)
library(pairwiseAdonis)
library(variancePartition)
library(janitor)
library(pander)
library(sf)
library(rnaturalearth)
library(rnaturalearthdata)
library(rgeos)
library(maps)
library(tidyverse)
library(gplots)
library(ggsignif)
library(ggrepel)
library(viridis)
library(cowplot)
library(RColorBrewer)
library(patchwork)
opts_chunk$set(echo=F, warning=F, message=F, dpi=150)
```

Set the working directory of the notebook to the parent directory of the 
notebook itself.  

```{r setup}
knitr::opts_knit$set(root.dir = normalizePath("../../../"))
```

# Site map
```{r}
# pull country data
world <- ne_countries(scale = "medium", returnclass = "sf")

# add coordinates to a dataframe
smoking_sites <- data.frame(longitude = c(4.9168030, -2.9804059,-1.257,-3.7892050,-0.3765000), latitude = c(52.5519900, 43.2805735,51.752,37.7769480,39.4695270), Site = c("MID","CMB","Radcliffe","JAE","VLC"), Data = c("New","New","Published","Published","Published"))

non_smoking_sites <- data.frame(longitude = c(-8.2023080,-0.8088890,-2.8622220), latitude = c(54.0068400,38.2486110,38.6963890), Site = c("Kilteasheen","ELR","IVE"), Data = c("Published","New","New"))


# plot the map
sampling_sites_europe <- ggplot(data = world) +
  geom_sf(colour = "#ffffff", fill = "#bbbbbb", linetype = "dotted") +
  #for grey map with white lines
  geom_sf(fill = "#f7fcf5") +
  geom_point(data = non_smoking_sites, aes(x = longitude, y = latitude, color = Data), size = 4,
        shape = 17) +
  geom_point(data = smoking_sites, aes(x = longitude, y = latitude, color = Data), size = 4,
        shape = 19) +
  coord_sf(xlim = c(-12, 20), ylim = c(35, 60), expand = FALSE) +
  theme_void() +
  scale_color_manual(values = c("black","grey50"))
sampling_sites_europe

ggsave("./06-publication/main_figures/Figure_4/sample_site_map.pdf", plot = sampling_sites_europe, device = "pdf",
       scale = 1, width = 5, height = 5, units = c("in"), dpi = 300)

```


# MetaPhlAn3

```{r load_data}
# read in the species table
mpa3_raw <- fread("./05-results.backup/mpa3_abundance_table.tsv") %>%
  select(-NCBI_tax_id)

# clean the file names
colnames(mpa3_raw) <- gsub(".metaphlan3","", colnames(mpa3_raw))
colnames(mpa3_raw) <- gsub(".SG1","", colnames(mpa3_raw))

# filter to have only species
mpa3_sp <- mpa3_raw %>% 
  filter(str_detect(clade_name, "s__")) %>%
  separate(clade_name, into = c("K","P","C","O","F","G","Species"), sep = "\\|") %>%
  select(-c("K","P","C","O","F","G")) %>%
  mutate(Species = str_replace_all(Species, "s__",""),
         Species = str_replace_all(Species, "_"," ")) %>%
  # these samples have no metadata
  select(-c("MID025.A0101","MID033.A0101","MID052.A0101","MID056.A0101","MID065.A0101","MID068.A0101","MID076.A0101","MID078.A0101")) %>%
  arrange(Species)

```


Now decontam the tables
```{r}
mpa3_contaminants <- fread("./05-results.backup/contaminant_species_mid_mpa3.tsv", sep = "\t") %>%
  bind_rows(., fread("./05-results.backup/contaminant_species_kilteasheen_mpa3.tsv", sep = "\t")) %>%
  bind_rows(., fread("./05-results.backup/contaminant_species_iberia_mpa3.tsv", sep = "\t")) %>%
  unique()

# remove contaminant species from nt table
mpa3_species_decontam <- mpa3_sp %>%
  anti_join(., mpa3_contaminants)

```

```{r load_strep_groups}
strep_groups <- fread("./00-documentation.backup/25-streptococcus_cladegroup_amylasegroup_database.tsv") %>%
  select(Taxon, Clade_Consensus) %>%
  rename(Species = Taxon) %>%
  mutate(Species = str_replace_all(Species, "_", " ")) %>%
  rename(Strep_group = Clade_Consensus) %>%
  mutate(Strep_group = fct_relevel(Strep_group, "sanguinis","mitis","salivarius","anginosus","bovis","pyogenic","mutans","unknown","NA"))

# get the proportion of sanguinis and anginosis for each sample
strep_group_info <- mpa3_species_decontam %>%
  filter(str_detect(Species, "Streptococcus")) %>%
  # join with the Strep grup table for plotting
  inner_join(., strep_groups) %>%
  select(-Species) %>%
  select(Strep_group, everything()) %>%
  # transpose
  gather("Library_ID","Counts", 2:ncol(.)) %>%
  group_by(Library_ID, Strep_group) %>%
  summarize(Reads = sum(Counts)) %>%
  spread(Strep_group, Reads) %>%
  # convert table to proportions
  adorn_percentages(denominator = "row") %>%
  # arrange by sanguinis and for the ones w/o sanguinis, by anginosis
  arrange(desc(sanguinis))

```

```{r strep_groups}
# load the MALT RefSeq taxonomy tables that have been decontaminated 
# and the column headers are cleaned up
# this is to get the Strep groups for coloring the PCA plot
load("./05-results.backup/Taxonomy_tables.RData")

# list poor samples from cuperdec
poor_samples_rs <- fread("./05-results.backup/cuperdec_poor_samples.tsv") %>%
  rename(Library_ID = Sample) %>%
  filter(str_detect(Library_ID, "MID")) %>%
  pull(Library_ID)

# we'll use the prevalence-filtered dataset
malt_rs <- all_species_decontam_prev %>%
  # select(matches("Species|MID|CMB")) %>%
  adorn_totals(where = "col")  %>%
  filter(Total >  0) %>%
  select(-Total) %>%
  select(-poor_samples_rs)

strep_groups <- fread("./00-documentation.backup/dRep_strep_groups.tsv") %>%
  # remove the extrac column with the old dRep
  select(-Clade_Consensus) %>%
  # make a factor for plotting
  mutate(dRep = fct_relevel(dRep, "Sanguinis","Mitis","Salivarius","Anginosus","Bovis","Pyogenic","Mutans","Other_oral","Other","Unknown"))

strep_rs <- malt_rs %>%
  filter(str_detect(Species, "Streptococcus")) %>%
  # join with the Strep grup table for plotting
  inner_join(., strep_groups) %>%
  select(Species, dRep, everything()) %>%
  # convert to long-format
  gather("Library_ID","Counts", 3:ncol(.)) %>%
  # mutate(Library_ID = fct_relevel(Library_ID, sample_order_rs)) %>%
  dplyr::group_by(Library_ID, dRep) %>%
  summarize(Count_sum = sum(Counts)) %>%
  ungroup() %>%
  group_by(Library_ID) %>%
  mutate(Percent = Count_sum / sum(Count_sum)) %>%
  select(Library_ID, dRep, Percent) %>%
  pivot_wider(names_from = dRep, values_from = Percent)

```

```{r load_metadata}

# load the metadata file
load("./05-results.backup/Metadata_tables.RData")

# read in the MALT mapped/unmapped gc and read length info from `MID_malt_map_unmap_rl_gc.Rmd`
malt_map_unmap_rl_gc <- fread("./05-results.backup/rl_gc_sub_data.tsv")
colnames(malt_map_unmap_rl_gc) <- gsub("_map","_malt_map", colnames(malt_map_unmap_rl_gc))
colnames(malt_map_unmap_rl_gc) <- gsub("_unmap","_malt_unmap", colnames(malt_map_unmap_rl_gc))

# full_metadata <- full_metadata %>%
#   mutate(`DNA_molecules_per_library_x10^6_log` = log10(`DNA_molecules_per_library_x10^6`)) %>%
#   full_join(., malt_map_unmap_rl_gc) %>%
#   full_join(., strep_groups, by = "Library_ID")

```


```{r pca_plot_fxn}
# plot PCA with colored dots and the title including the # of species or genera
plot_pca <- function(df, pc1, pc2, color_group, shape_group, ncomps) {
    metadata_group_colors <- get(paste(color_group, "_colors", sep = ""))
    metadata_group_shapes <- get(paste(shape_group, "_shapes", sep = ""))

    pca.list <- mixOmics::pca(df, ncomp = ncomps, logratio = 'CLR')

    exp_var <- paste0(round(pca.list$prop_expl_var$X * 100, 2), "%")
    df_X <- pca.list$variates$X %>%
              as.data.frame() %>%
              rownames_to_column("Library_ID") %>%
              inner_join(full_metadata, by = "Library_ID") %>%
              inner_join(., strep_group_info)
              
    color_group = df_X[[color_group]]
    shape_group = df_X[[shape_group]]

    if (pc1 == 'PC1') {
        pc1 <- df_X$PC1
        exp_var_pc1 <- exp_var[1]
        xaxis <- c("PC1")
    }  else if (pc1 == 'PC2') {
        pc1 <- df_X$PC2
        exp_var_pc1 <- exp_var[2]
        xaxis <- c("PC2")
    } else if (pc1 == 'PC3') {
        pc1 <- df_X$PC3
        exp_var_pc1 <- exp_var[3]
        xaxis <- c("PC3")
    }

    if (pc2 == 'PC1') {
        pc2 <- df_X$PC1
        exp_var_pc2 <- exp_var[1]
        yaxis <- c("PC1")
    }  else if (pc2 == 'PC2') {
        pc2 <- df_X$PC2
        exp_var_pc2 <- exp_var[2]
        yaxis <- c("PC2")
    } else if (pc2 == 'PC3') {
        pc2 <- df_X$PC3
        exp_var_pc2 <- exp_var[3]
        yaxis <- c("PC3")
    }

    pca_plot <- ggplot(df_X, aes(pc1, pc2)) +
     geom_point(aes(, fill = color_group, shape = shape_group), size = 2.5, stroke = 0.3) +
     scale_fill_manual(values = metadata_group_colors) +
     scale_shape_manual(values = metadata_group_shapes) +
     # stat_ellipse() +
     xlab(paste(xaxis, " - ", exp_var_pc1)) +
     ylab(paste(yaxis, " - ", exp_var_pc2)) +
     theme_minimal(base_size = 16) +
     theme(text = element_text(size=16)) +
     theme(legend.title = element_blank(),
           legend.key.size = unit(2,"mm"),
           legend.text = element_text(size = 6)) +
     theme(legend.position = "top")

    return(pca_plot)
}

```


```{r pca_plot_cont_fxn}
# for continuous data
plot_pca_cont <- function(df, pc1, pc2, color_group, shape_group, ncomps, title_text) {
    metadata_group_shapes <- get(paste(shape_group, "_shapes", sep = ""))

    pca.list <- mixOmics::pca(df, ncomp = ncomps, logratio = 'CLR')

    exp_var <- paste0(round(pca.list$prop_expl_var$X * 100, 2), "%")
    df_X <- pca.list$variates$X %>%
              as.data.frame() %>%
              rownames_to_column("Library_ID") %>%
              inner_join(full_metadata, by = "Library_ID") %>%
              inner_join(., strep_group_info, by = "Library_ID")
    # %>%
              # full_join(., full_source_results %>%
              #             rename(Library_ID = SampleID),  by = "Library_ID")
    
    color_group = df_X[[color_group]]
    shape_group = df_X[[shape_group]]

    if (pc1 == 'PC1') {
        pc1 <- df_X$PC1
        exp_var_pc1 <- exp_var[1]
        xaxis <- c("PC1")
    }  else if (pc1 == 'PC2') {
        pc1 <- df_X$PC2
        exp_var_pc1 <- exp_var[2]
        xaxis <- c("PC2")
    } else if (pc1 == 'PC3') {
        pc1 <- df_X$PC3
        exp_var_pc1 <- exp_var[3]
        xaxis <- c("PC3")
    }

    if (pc2 == 'PC1') {
        pc2 <- df_X$PC1
        exp_var_pc2 <- exp_var[1]
        yaxis <- c("PC1")
    }  else if (pc2 == 'PC2') {
        pc2 <- df_X$PC2
        exp_var_pc2 <- exp_var[2]
        yaxis <- c("PC2")
    } else if (pc2 == 'PC3') {
        pc2 <- df_X$PC3
        exp_var_pc2 <- exp_var[3]
        yaxis <- c("PC3")
    }

    pca_plot <- ggplot(df_X, aes(pc1, pc2, fill = color_group, shape = shape_group)) +
     geom_point(size = 2.5, stroke = 0.3) +
     scale_fill_viridis_c(option = "C") +
     scale_shape_manual(values = metadata_group_shapes) +
     # stat_ellipse() +
     xlab(paste(xaxis, " - ", exp_var_pc1)) +
     ylab(paste(yaxis, " - ", exp_var_pc2)) +
     theme_minimal(base_size = 16) +
     theme(text = element_text(size=16)) +
     theme(legend.title = element_blank(),
           legend.key.size = unit(2,"mm"),
           legend.text = element_text(size = 6)) + 
     theme(legend.position = "top") +
     ggtitle(title_text) + theme(plot.title = element_text(size = 10))

    return(pca_plot)
}


```

```{r pca_biplot_fxn}
plot_pca_bi <- function(df, pc1, pc2, metadata_group, columntitle) {
    metadata_group_colors <- get(paste(metadata_group, "_colors", sep = ""))
    metadata_group_shapes <- get(paste(metadata_group, "_shapes", sep = ""))
   
    arrow_pc <- enquo(columntitle)
    
    exp_var <- paste0(round(df$prop_expl_var$X * 100, 2), "%") # explained variance for x- and y-labels
    
    # select only the PCs from the PCA and add metadata
    df_X <- df$variates$X %>%
              as.data.frame() %>%
              rownames_to_column("Library_ID") %>%
              inner_join(full_metadata, by = "Library_ID")
    
    # metadata_group = df_X[[metadata_group]]
    
    corr_lam <- df$sdev[c("PC1", "PC2", "PC3")] * sqrt(nrow(df_X))

    df_X <- df_X %>%
      mutate(PC1 = PC1 / corr_lam[1],
             PC2 = PC2 / corr_lam[2], 
             PC3 = PC3 / corr_lam[3])

    df_Y <- df_X[, c(pc1,  pc2, metadata_group)]
    # select the correct PC column and explained variance for PC1
    xaxis <- pc1
    yaxis <- pc2
    exp_var_pc1 <- exp_var[match(pc1, c("PC1", "PC2", "PC3"))]
    exp_var_pc2 <- exp_var[match(pc2, c("PC1", "PC2", "PC3"))]
    
    # Identify the 10 pathways that have highest positive and negative loadings in the selected PC
    pws_10 <- df$loadings$X %>%
      as.data.frame(.) %>%
      rownames_to_column(var = "Taxon") %>%
      top_n(10, !!arrow_pc)

    neg_10 <- df$loadings$X %>%
      as.data.frame(.) %>%
      rownames_to_column(var = "Taxon") %>%
      top_n(-10, !!arrow_pc)


    pca_plot_bi <- ggplot(df_Y, aes(x = get(paste(pc1)), y = get(paste(pc2)))) +
      geom_point(size = 2.5, aes(shape = get(paste(metadata_group)), fill = get(paste(metadata_group)))) +
      geom_segment(data = pws_10,
                   aes(xend = get(paste(pc1)), yend = get(paste(pc2))),
                   x = 0, y = 0, colour = "black",
                   size = 0.5,
                   arrow = arrow(length = unit(0.03, "npc"))) +
      geom_label_repel(data = pws_10,
                   aes(x = get(paste(pc1)), y = get(paste(pc2)), label = Taxon),
                   size = 2.5, colour = "grey20", label.padding = 0.2, force = 5, max.overlaps = 14) +
      geom_segment(data = neg_10,
                   aes(xend = get(paste(pc1)), yend = get(paste(pc2))),
                   x = 0, y = 0, colour = "grey50",
                   size = 0.5,
                   arrow = arrow(length = unit(0.03, "npc"))) +
      geom_label_repel(data = neg_10,
                   aes(x = get(paste(pc1)), y = get(paste(pc2)), label = Taxon),
                   size = 2.5, colour = "grey20", label.padding = 0.2, max.overlaps = 14) +
      labs(x = paste(xaxis, " - ", exp_var_pc1),
           y = paste(yaxis, " - ", exp_var_pc2)) +
      scale_fill_manual(values = metadata_group_colors) +
      # scale_shape_manual(values = c(21, 22, 23, 24, 21, 22, 23, 24)) +
      scale_shape_manual(values = metadata_group_shapes) +
      theme_minimal() + theme(text = element_text(size = 16)) +
      theme(text = element_text(size=16)) +
      theme(legend.position = "top")

    return(pca_plot_bi)
}

# mpa3_species_decontam_filtered_nocontrols_biplot_122 <- plot_pca_bi(mpa3_species_decontam_unfilt_all.pca, "PC1", "PC2", "Site_code", PC1)
# mpa3_species_decontam_filtered_nocontrols_biplot_122


```


List the outliers from the plot in `MID_CMB_Rad_Kil_Iber_decontam.Rmd`. 
```{r outliers}
outliers_mpa3 <- c("EXB059.A2101","EXB059.A2501","EXB015.A3301","EXB034.A2701","EXB059.A2201","EXB059.A2301","EXB059.A2401","LIB058.A0103","LIB058.A0106","LIB058.A0104")
poor_samples_mpa3 <- fread("./05-results.backup/cuperdec_mpa3_poor_samples.tsv") %>%
  pull(Sample)

outliersF <- str_c(outliers_mpa3, collapse = "|")

```


#Sample Clustering with PCA

Step-by-step removal of taxa based on decontam and a filter threshold, followed
by PCA to see how contaminants/low abundance species affect the relationships of
samples to eachother between groups.
```{r pca_all_species}
# decontaminated only
mpa3_species_decontam_unfilt <- mpa3_species_decontam %>%
  select(-one_of(outliers_mpa3)) %>%
  select(-one_of(poor_samples_mpa3)) %>%
  select(-matches("ARS|CSS|CSL")) %>%
  gather("Library_ID","Counts",2:ncol(.)) %>%
  mutate(Counts = Counts +1) %>%
  spread(Species,Counts) %>%
  column_to_rownames("Library_ID")

# prepare to run a PCA
# check the number of components to retain by tuning the PCA
tune.pca(mpa3_species_decontam_unfilt, logratio = 'CLR')

# define Time Period shapes for plotting
Time_period_shapes <- c(Industrial = 21, Medieval = 24, Modern = 22, Extraction_blank = 23, Library_blank = 23)

```


# Species-level analyses
What is the relationship of the groups in a PCA without the blanks? Let's try
the PCA using only the species that are present in at least 10% of at least one
sample group (MID, CMB, Radcliffe, Kilteasheen, Iberia)

```{r}

# decontaminated only
mpa3_species_decontam_unfilt <- mpa3_species_decontam %>%
  select(-one_of(outliers_mpa3)) %>%
  select(-one_of(poor_samples_mpa3)) %>%
  select(-matches("EXB|LIB|CSN|CSD|ARS|CSS|CSL")) %>%
  gather("Library_ID","Counts",2:ncol(.)) %>%
  mutate(Counts = Counts +1) %>%
  spread(Species,Counts) %>%
  column_to_rownames("Library_ID")

# prepare to run a PCA
# check the number of components to retain by tuning the PCA
tune.pca(mpa3_species_decontam_unfilt, logratio = 'CLR')

# define Time Period shapes for plotting
Time_period_shapes <- c(Industrial = 21, Medieval = 24, Modern = 22, Extraction_blank = 23, Library_blank = 23)

```

```{r}
mpa3_species_decontam_unfilt %>%
  rownames_to_column("Library_ID") %>%
  select(Library_ID) %>%
  filter(str_detect(Library_ID, "MID"))
  
```

```{r pca_time_period}
time_period <- plot_pca(mpa3_species_decontam_unfilt, "PC1", "PC2","Time_period","Time_period", 3) 
time_period

```

```{r pca_site}
plot_pca(mpa3_species_decontam_unfilt, "PC1", "PC2","Site_code","Time_period", 3) 

```

```{r pca_reads}
# by total reads into MetaPhlAn3 
seqdepth_pc12 <- plot_pca_cont(mpa3_species_decontam_unfilt, "PC1", "PC2", "No_reads_total", "Time_period", 3, "Total reads into MetaPhlAn3")
seqdepth_pc12

```

```{r pca_gc}
# by average GC content
gc <- plot_pca_cont(mpa3_species_decontam_unfilt, "PC1", "PC2", "gc_avg", "Time_period", 3, "")
gc

```

```{r pca_rl}
# by average read length
rl <- plot_pca_cont(mpa3_species_decontam_unfilt, "PC1", "PC2", "seq_len_avg", "Time_period", 3, "Avg. read length (bp)")
rl

# ggsave("./06-publication/supplemental_figures/S11/read_length.pdf", plot = rl,
#        device = "pdf", scale = 1, width = 6, height = 5, units = c("in"), dpi = 300)

```

```{r}
# select the species with strongest loadings in PC1 + and - directions
pc1_mpa3 <- fread("./05-results.backup/mpa3_pc1_species_list.tsv") 

pc1neg <- pc1_mpa3 %>%
  filter(PC1 > 0) %>%
  pull(Taxon)

pc_species_neg <- mpa3_species_decontam %>%
  select(-one_of(outliers_mpa3)) %>%
  select(-one_of(poor_samples_mpa3)) %>%
  select(-matches("EXB|LIB|CSN|CSD|ARS|CSS|CSL")) %>%
  filter(Species %in% pc1neg) %>%
  # convert to long-format
  pivot_longer(!Species, names_to = "Library_ID", values_to = "Counts") %>%
  dplyr::group_by(Library_ID) %>%
  summarize(Count_sum = sum(Counts)) %>%
  ungroup() %>%
  full_join(., mpa3_species_decontam %>%
  select(-one_of(outliers_mpa3)) %>%
  select(-one_of(poor_samples_mpa3)) %>%
  select(-matches("EXB|LIB|CSN|CSD|ARS|CSS|CSL")) %>%
              pivot_longer(!Species, names_to = "Library_ID", values_to = "Counts") %>%
              dplyr::group_by(Library_ID) %>%
              summarize(Total = sum(Counts)) %>%
              select(Library_ID, Total)) %>%
  mutate(Percent_neg = Count_sum / Total * 100) %>%
  select(Library_ID, Percent_neg)

mpa3_all_otu.pca <- mixOmics::pca(mpa3_species_decontam_unfilt, ncomp = 2, logratio = 'CLR')

mpa3_all_pca_values_rs_pc <- mpa3_all_otu.pca$variates$X %>%
  as.data.frame() %>%
  rownames_to_column("Library_ID") %>%
  inner_join(., full_metadata, by = "Library_ID") %>%
  left_join(., pc_species_neg, by = "Library_ID")

pc1_neg_plot <- ggplot(mpa3_all_pca_values_rs_pc, aes(PC1, PC2, fill = Percent_neg, shape = Time_period)) +
  geom_point(size = 2, color = "black") +
     scale_fill_viridis_c(option = "C") +
     scale_shape_manual(values = c(21, 24, 22, 23, 23)) +
     # stat_ellipse() +
     # xlab(paste(xaxis, " - ", exp_var_pc1)) +
     # ylab(paste(yaxis, " - ", exp_var_pc2)) +
     theme_minimal(base_size = 16) +
     theme(text = element_text(size=16)) +
     theme(legend.title = element_blank(),
           legend.key.size = unit(2,"mm"),
           legend.text = element_text(size = 6)) + 
     theme(legend.position = "top") +
     ggtitle("") + theme(plot.title = element_text(size = 10))
pc1_neg_plot


```

```{r biplots_all_species, eval = F}

mpa3_species_decontam_unfilt_all <- mpa3_species_decontam %>%
  select(-one_of(outliers_mpa3)) %>%
  select(-one_of(poor_samples_mpa3)) %>%
  select(-matches("EXB|LIB|CSN|CSD|ARS|CSS|CSL")) %>%
  gather("Library_ID","Counts",2:ncol(.)) %>% 
  mutate(Counts = Counts +1) %>%
  spread(Species,Counts) %>%
  column_to_rownames("Library_ID")

mpa3_species_decontam_unfilt_all.pca <- mixOmics::pca(mpa3_species_decontam_unfilt_all, ncomp = 3, logratio = 'CLR')
plot(mpa3_species_decontam_unfilt_all.pca)

# Define Time Period shapes
Time_period_shapes <- c(Industrial = 21, Medieval = 24, Modern = 22, Extraction_blank = 23, Library_blank = 23)

mpa3_species_decontam_filtered_nocontrols_biplot_121 <- plot_pca_bi(mpa3_species_decontam_unfilt_all.pca, "PC1", "PC2", "Time_period", PC1)
mpa3_species_decontam_filtered_nocontrols_biplot_121
# plot_pca_bi(mpa3_species_decontam_unfilt_no_Moralis.pca, "PC1", "PC2", "Time_period", PC1)

mpa3_species_decontam_unfilt_all.pca$variates$X %>%
  as.data.frame() %>%
  rownames_to_column("Library_ID") %>%
  inner_join(full_metadata, by = "Library_ID") %>% select(Library_ID, PC1, PC2, Site_code)



# Define site code shapes
Site_code_shapes <- c(MID = 21, CMB = 21, RAD = 21, KIL = 24, ELR = 24, IVE = 24, JAE = 22, VLC = 22)


mpa3_species_decontam_filtered_nocontrols_biplot_122 <- plot_pca_bi(mpa3_species_decontam_unfilt_all.pca, "PC1", "PC2", "Site_code", PC1)
mpa3_species_decontam_filtered_nocontrols_biplot_122

pandoc.table(mpa3_species_decontam_filtered_nocontrols_biplot_121$plot_env$pws_10 %>% arrange(desc(PC1)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC1 positive loadings")

pandoc.table(mpa3_species_decontam_filtered_nocontrols_biplot_121$plot_env$neg_10 %>% arrange(PC1),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC1 negative loadings")

mpa3_species_decontam_filtered_nocontrols_biplot_122 <- plot_pca_bi(mpa3_species_decontam_unfilt_all.pca, "PC1", "PC2", "Site_code", PC2)
mpa3_species_decontam_filtered_nocontrols_biplot_122

pandoc.table(mpa3_species_decontam_filtered_nocontrols_biplot_122$plot_env$pws_10 %>% arrange(desc(PC2)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC2 positive loadings")

pandoc.table(mpa3_species_decontam_filtered_nocontrols_biplot_122$plot_env$neg_10 %>% arrange(PC2),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC2 negative loadings")

```

```{r no_methanobrevibacter, eval = F}
# remove Methanobrevibacter oralis from the species table
mpa3_species_decontam_unfilt_no_Moralis <- mpa3_species_decontam %>%
  filter(Species != "Methanobrevibacter oralis") %>%
  select(-one_of(outliers_mpa3)) %>%
  select(-one_of(poor_samples_mpa3)) %>%
  select(-matches("EXB|LIB|CSN|CSD|ARS|CSS|CSL")) %>%
  gather("Library_ID","Counts",2:ncol(.)) %>%
  mutate(Counts = Counts +1) %>%
  spread(Species,Counts) %>%
  column_to_rownames("Library_ID")


# perform and plot a PCA to see how the data cluster
nomethano_beta <- plot_pca(mpa3_species_decontam_unfilt_no_Moralis, "PC1", "PC2", "Site_code", "Time_period", 3)
nomethano_beta

# by average GC content
nomethano_gc <- plot_pca_cont(mpa3_species_decontam_unfilt_no_Moralis, "PC1", "PC2", "gc_avg", "Time_period", 3, "Avg. GC content (%)")
nomethano_gc

# by average read length
plot_pca_cont(mpa3_species_decontam_unfilt_no_Moralis, "PC1", "PC2", "seq_len_avg", "Time_period", 3, "Avg. read length (bp)")

# biplot to see strongest loading species
mpa3_species_decontam_unfilt_no_Moralis.pca <- mixOmics::pca(mpa3_species_decontam_unfilt_no_Moralis, ncomp = 3, logratio = 'CLR')
plot(mpa3_species_decontam_unfilt_no_Moralis.pca)

# define shapes
Site_code_shapes <- c(MID = 21, CMB = 21, RAD = 21, KIL = 24, ELR = 24, IVE = 24, JAE = 22, VLC = 22, EXB = 25, LIB = 25)

nomethano_bi <- plot_pca_bi(mpa3_species_decontam_unfilt_no_Moralis.pca, "PC1", "PC2", "Site_code", PC1)
nomethano_bi

no_methano <- plot_grid(nomethano_beta, nomethano_gc, nomethano_bi,
          ncol = 1, labels = c("A", "B","C"), rel_widths = c(1), scale = 0.95)
no_methano

# ggsave("./06-publication/supplemental_figures/S4/no_methano.pdf", plot = no_methano,
#        device = "pdf", scale = 1, width = 6, height = 15, units = c("in"), dpi = 300)


```


Check the GC content of the species w/strongest loadings in PC2+ and PC2- space
```{r}

# read in the file w/avg GC content of each species, taken from NCBI Genome
pc2_gc <- fread("./05-results.backup/pc2_gc_avg.tsv") %>%
  mutate(Order = c(rep(1:10, times = 4))) %>%
  # mutate(Order = c(rep(1:10)), c(rep(1:10)))
  drop_na(PC2_loading)

# plot
box <- ggplot(pc2_gc, aes(x = PC2, y = Avg_GC, fill = PC2)) +
  geom_boxplot() +
  geom_jitter(width = 0.1, color = "grey50") +
  geom_signif(
    comparisons = list(c("PWS", "NEG")),
    map_signif_level = TRUE
  ) +
  theme_minimal() +
  scale_fill_manual(values = c("#311068", "#C83E73")) +
  xlab("PC2 direction") +
  ylab("Species avg. GC (%)")
box


rain <- ggplot(pc2_gc, aes(x = PC2, y = Avg_GC, fill = PC2)) + 
  ## add half-violin from {ggdist} package
  ggdist::stat_halfeye(
    ## custom bandwidth
    adjust = .5, 
    ## adjust height
    width = .6, 
    ## move geom to the right
    justification = -.2, 
    ## remove slab interval
    .width = 0, 
    point_colour = NA
  ) + 
  geom_boxplot(
    width = .12, 
    ## remove outliers
    outlier.color = NA, ## `outlier.shape = NA` works as well
    color = "grey30"
  ) +
  ## add dot plots from {ggdist} package
  ggdist::stat_dots(
    ## orientation to the left
    side = "left", 
    ## move geom to the left
    justification = 1.1, 
    ## adjust grouping (binning) of observations 
    binwidth = 1.0
  ) + 
  ## remove white space on the left
  coord_cartesian(xlim = c(1.2, NA)) +
  theme_minimal(base_size = 14) +
  # scale_fill_manual(values = Time_period_colors) +
  theme(axis.text = element_text(size = 14)) +
  theme(legend.position = "none") +
  xlab("PC2 direction") +
  ylab("Species avg. GC (%)") +
  theme(axis.title.x = element_text(vjust=-1.5)) +
  theme(axis.title.y = element_text(vjust=1.9)) +
  scale_fill_manual(values = c("#311068", "#C83E73"))

rain

```

```{r plot_together}
beta_div <- time_period + pc1_neg_plot + gc + box
beta_div

# ggsave("./06-publication/main_figures/Figure_4/beta_div_all.pdf", plot = beta_div,
#        device = "pdf", scale = 1, width = 11, height = 8.5, units = c("in"), dpi = 300)

```


## Canonical Correlation Analysis

```{r pca_species_all}
mpa3_species_decontam_unfilt <- mpa3_species_decontam %>%
  select(-one_of(outliers_mpa3)) %>%
  select(-one_of(poor_samples_mpa3)) %>%
  select(-matches("EXB|LIB|CSN|CSD|ARS|CSS|CSL")) %>%
  gather("Library_ID","Counts",2:ncol(.)) %>%
  mutate(Counts = Counts +1) %>%
  spread(Species,Counts) %>%
  column_to_rownames("Library_ID")

# combine metadata
comb_metadata <- full_metadata %>%
  mutate(`DNA_molecules_per_library_x10^6_log` = log10(`DNA_molecules_per_library_x10^6`)) %>%
  full_join(., malt_map_unmap_rl_gc) %>%
  full_join(., strep_rs, by = "Library_ID")


# prepare to run a PCA
# check the number of components to retain by tuning the PCA
tune.pca(mpa3_species_decontam_unfilt, logratio = 'CLR')

pca.list <- mixOmics::pca(mpa3_species_decontam_unfilt, ncomp = 5, logratio = 'CLR')

exp_var_all <- paste0(round(pca.list$prop_expl_var$X * 100, 2), "%")
all_pca_values <- pca.list$variates$X %>%
          as.data.frame() %>%
          rownames_to_column("Library_ID") %>%
          inner_join(comb_metadata, by = "Library_ID") %>%
          inner_join(., strep_rs) %>%
  drop_na(Site_code) %>%
  mutate(Time_period_n = ifelse(Time_period == "Medieval",0,
                                ifelse(Time_period == "Industrial",1,2))) %>%
  mutate(Site_code_n = ifelse(Site_code == "MID",0,
                              ifelse(Site_code == "CMB",1,
                                     ifelse(Site_code == "RAD",2,
                                            ifelse(Site_code == "KIL",3,
                                                   ifelse(Site_code == "ELR",4,
                                                          ifelse(Site_code == "IVE",5,
                                                                 ifelse(Site_code == "JAE",6,7))))))))


```

```{r all_corrs}
# rename variables to look nice in figures
all_pca_values_select <- all_pca_values %>%
  select(Library_ID,PC1, PC2, seq_len_avg, gc_avg, No_reads_total, Sanguinis, Anginosus, Time_period_n, Site_code_n) %>%
  rename(`Seq length` = seq_len_avg,
         GC = gc_avg,
         `Total reads` = No_reads_total,
         `Time period` = Time_period_n,
         Site = Site_code_n)

form <- ~ PC1 + PC2 + `Seq length` + GC + `Total reads` + Sanguinis + Anginosus + `Time period` + Site
C_all = canCorPairs(form, all_pca_values_select)
plotCorrMatrix( C_all )


```


```{r pearson_corr_mid}
# test for significant differences in diversity between the metadata categories

pearson_all <- all_pca_values %>%
rstatix::cor_test(., PC1, PC2, PC3, seq_len_avg, gc_avg, No_reads_total, Sanguinis, Anginosus, Time_period_n, Site_code_n, method="pearson")

pearson_all

pearson_all %>%
  mutate(same = var1 == var2) %>%
  filter(same == FALSE) %>%
  select(-same) %>%
  filter(p <= 0.001) %>%
  filter(cor > 0.40 | cor < -0.40) %>%
  distinct(statistic, .keep_all = TRUE)
  
```

```{r}

C_all %>%
  as_data_frame(.) %>%
  mutate(names = rownames(C_all)) %>%
  select(names, everything()) %>%
  pivot_longer(!names, names_to = "Corrs", values_to = "values") %>%
  filter(values != 1) %>%
  filter(values >= 0.4) %>%
  distinct(values, .keep_all = TRUE)


C_all %>%
  as_data_frame(.) %>%
  mutate(names = rownames(C_all)) %>%
  select(names, everything()) %>%
  pivot_longer(!names, names_to = "Corrs", values_to = "values") %>%
  filter(names == "GC") %>%
  arrange(desc(values))

```


```{r}
# colorset <- rev(colorspace::sequential_hcl(10, palette = "BuPu"))

colorset <- c("#ffffff","#e8e2ee","#d1c6dc","#baaacb","#a48fba","#8d74aa","#775b99","#604289","#492978","#311068")

 pvals_all <- all_pca_values_select %>%
  column_to_rownames("Library_ID") %>%
  as.matrix(.) %>%
  corrplot::cor.mtest(., method="pearson")

C_all %>%
  corrplot::corrplot(., p.mat = pvals_all$p, order = "hclust", type = "upper", col = colorset, is.corr = FALSE, cl.cex = 1.5, diag = FALSE, tl.cex = 1.7,
                     tl.col = 'black', sig.level = c(0.001), pch.cex = 2.2, insig = 'label_sig', pch.col = 'grey20')

```

```{r, eval = F}
svg(file = "./06-publication/main_figures/Figure_4/all_corr_plot.svg") 

C_all %>%
  corrplot::corrplot(., p.mat = pvals_all$p, order = "hclust", type = "upper", col = colorset, is.corr = FALSE, cl.cex = 1.5, diag = FALSE, tl.cex = 1.7,
                     tl.col = 'black', sig.level = c(0.001), pch.cex = 2.2, insig = 'label_sig', pch.col = 'grey20')

dev.off()

```

